var lunrIndex, pagesIndex;

function endsWith(str, suffix) {
    return str.indexOf(suffix, str.length - suffix.length) !== -1;
}

//MTB [09/01/2019]:
function myLunrParse(index) {
  pagesIndex =   index;
  // Set up lunrjs by declaring the fields we use
  // Also provide their boost level for the ranking
  lunrIndex = new lunr.Index
  lunrIndex.ref("uri");
  lunrIndex.field('title', {
      boost: 15
  });
  lunrIndex.field('tags', {
      boost: 10
  });
  lunrIndex.field("content", {
      boost: 5
  });

  // Feed lunr with each file and let lunr actually index them
  pagesIndex.forEach(function(page) {
      lunrIndex.add(page);
  });
  lunrIndex.pipeline.remove(lunrIndex.stemmer)
}

// Initialize lunrjs using our generated index file
function initLunr() {
    if (!endsWith(baseurl,"/")){
        baseurl = baseurl+'/'
    };

    //MTB [09/01/2019]: jsContent contains index.json all in one string without \r and \n chars
    var jsonContent = '[ { "uri": "./cli/arguments.html", "title": "Arguments", "tags": [], "description": "", "content": "Three alternative arguments can be used to run alvaDescCLI: Argument Description --script=&lt;SCRIPT FILE&gt; or -s &lt;SCRIPT FILE&gt; All the available options are contained in the SCRIPT FILE. When this is used no other argument can be present on the command-line except for --verbose and --threads --iSMILES=&lt;SMILES&gt; Uses a single SMILES string as input --input[=&lt;INPUT FILE&gt;] It can be used to input a set of molecules either with an INPUT FILE or in the standard input (default) Calculation arguments: Argument Description --inputtype=SMILES|MDL|SYBYL|HYPERCHEM Input file type. It&rsquo;s mandatory if the input file name is omitted --descriptors=&lt;DESCRIPTOR NAMES&gt; List of comma-separated descriptors to be calculated. To calculate all use &ndash;descriptors=ALL. To calculate all 2D use &ndash;descriptors=ALL2D. --labels Adds descriptor and molecule labels --ecfp Calculates extended connectivity fingerprint --pfp Calculates path fingerprint --size=&lt;SIZE&gt; Size of hashed fingerprint (default: 1024) --min=&lt;MIN LENGTH&gt; Minimum fragment length for hashed fingerprint (default: 0) --max=&lt;MAX LENGTH&gt; Maximum fragment length for hashed fingerprint (default: 2) --count=&lt;TRUE|FALSE&gt; Counts fragments for hashed fingerprint (default: TRUE) --bits=&lt;BITS PER PATTERN&gt; Bits per pattern for hashed fingerprint (default: 2) --fpoptions=&lt;OPTIONS&gt; Atom types for hashed fingerprint (default: Atom type, Aromaticity, Charge, Connectivity (total), Bond order) --maccsfp Calculates MACCS166 fingerprint --output[=&lt;OUTPUT FILE&gt;] Output file (default: standard output) --threads=&lt;NUMBER OF THREADS&gt; Number of threads used in the calculation (default: equal to the maximum available CPUs) Other arguments: Argument Description --descriptorlist Prints the list of available descriptors --cite Prints the citation for alvaDesc --help or -h Prints the alvaDescCLI help --verbose Prints verbose messages during execution --update[=&lt;UPDATE FILE&gt;] Checks if a software update is available. If a file name is passed tries to download it. --license[=&lt;LICENSE FILE&gt;] Shows information of the given license file (default: the active one) --request-license=&lt;REQUEST FILE&gt; Generates license file request --import-license=&lt;LICENSE FILE&gt; Imports a license file " }, { "uri": "./descriptors/constitutional_descriptors.html", "title": "Constitutional indices", "tags": [], "description": "", "content": "Constitutional descriptors are the simplest molecular descriptors that can be calculated starting from a molecular structure. alvaDesc calculates 50 constitutional descriptors, such as the molecular weight (MW), the number of atoms (nAT) and bonds (nBT), the number of atoms (nSK) and bonds (nBO) in an H-depleted molecular graph, the absolute (e.g., nH, nC, nN, nO, nP, nS) and relative occurrence frequency of specific atom and bond types (H%, C%, O%, X%). alvaDesc includes also the number of aromatic (nAA) and terminal atoms (nTA). alvaDesc includes also descriptors based on the sum and the average of the following atomic properties: atomic mass (m) resulting in molecular weight (MW) and average molecular weight (AMW) van der Waals volume (v) atom electronegativity (e) atom polarizability (p) ionization potential (i) Descriptors calculated as the sum of an atomic property are defined as follows: Descriptors calculated as the average value of an atomic property are defined as follows: where nAT is the number of atoms and wi is the considered atomic property. Except for atomic mass (m), Mw and Sw descriptors are calculated using the carbon-scaled form of the atomic property. i.e., the ratio between the atomic property and the carbon atom property Additionally alvaDesc calculates descriptors depending on molecular bond composition, such as the number of double (nDB), triple (nTB) and aromatic bonds (nAB). SCBO is defined as the sum of the conventional bond order of all the edges in the H-depleted molecular graph. Conventional bond orders are: 1 for single bond, 2 for double bond, 3 for triple bond and 1.5 for aromatic bond. RBN1 is the number of bonds which allows the bonds to freely rotate around themselves, they are defined as any single bond, not in a ring, bound to a nonterminal heavy atom. Excluded from the count are amide C u2013N bonds because of their high rotational energy barrier. The rotatable bond fraction (RBF) is calculated as: where RBN is the number of rotatable bonds and nBT is total number of bonds. The graph density (GD)2 is derived from the H-depleted molecular graph and calculated as: where nBO is the number of non-hydrogen bonds and nSK is the number of non-hydrogen atoms. Fsp3 is a complexity index calculated as the fraction of the number of Sp3 hybridized carbons (nCsp3) and the total carbon count. alvaDesc includes its implementation of the maximum number of atoms that can be in conjugation with each other (max_conj_path)3. nStructures is the number of full connected fragments of the considered compound. Molecule nStructures 2 3 1 totalcharge is the sum of the atomic formal charges. Molecule totalcharge +1 -1 0 Veber, D. F., Johnson, S. R., Cheng, H.-Y., Smith, B. R., Ward, K. W., &amp; Kopple, K. D. (2002). Molecular Properties That Influence the Oral Bioavailability of Drug Candidates. Journal of Medicinal Chemistry, 45(12), 2615 u20132623. https://doi.org/10.1021/jm020017n&#160;&#x21a9;&#xfe0e; Coleman, T. F., &amp; Mor u00e9, J. J. (1983). Estimation of Sparse Jacobian Matrices and Graph Coloring Blems. SIAM Journal on Numerical Analysis, 20(1), 187 u2013209. https://doi.org/10.1137/0720013&#160;&#x21a9;&#xfe0e; Seidler, J., McGovern, S. L., Doman, T. N., &amp; Shoichet, B. K. (2003). Identification and Prediction of Promiscuous Aggregating Inhibitors among Known Drugs. Journal of Medicinal Chemistry, 46(21), 4477 u20134486. https://doi.org/10.1021/jm030191r&#160;&#x21a9;&#xfe0e; " }, { "uri": "./introduction.html", "title": "Introduction", "tags": [], "description": "", "content": "alvaDesc is the next generation tool for the calculation of molecular descriptors and fingerprints1. It calculates more than 4000 descriptors independent of 3-dimensional information such as constitutional, topological, pharmacophore. It includes ETA and Atom-type E-state indices together with atom pairs, functional groups and fragment counts. Due to the huge number of molecular descriptors implemented in alvaDesc they are splitted in 30 logical blocks Additionally, alvaDesc implements almost 1500 3-dimensional descriptors such as 3D-autocorrelation, Weighted Holistic Invariant Molecular descriptors (WHIM), GETAWAY, CATS3D and WHALES descriptors. alvaDesc does not calculate the 3-dimensional structure of compounds. In order to perform 3-dimensional descriptors calculation molecular structures provided to alvaDesc must include 3-dimensional atoms coordinates. alvaDesc provides the calculation of several model-based physicochemical properties such as molar refractivity, topological polar surface area (TPSA), molecular volume estimations, two LogP models (Moriguchi and Ghose-Chippen octanol-water partition coefficient). There is a significative list of drug-like and lead-like alerts including the well-known Lipinski alert index and the quantitative estimation of drug-likeness. Together with the calculation of molecular descriptors, alvaDesc carries out the calculation of three different molecular fingerprints: MACCS166 fingerprint Extended Connectivity Fingerprint Path Fingerprint alvaDesc allows to fully customise the calculation of Extended Connectivity Fingerprint and Path Fingerprint, these molecular fingerprints can be tuned not only with respect to the fingerprint size, fragment type and dimensions but even by defining atom and bond parameters considered during fragment identifications (e.g., atom type, aromaticity, the number of attached hydrogens, connectivity). Fragments identified during hashed fingerprint calculation can be exported as SMARTS strings (SMARTS is a line notation developed by Daylight Chemical Information Systems for representing molecular substructures). One of the most innovative features of alvaDesc is its capability to handle both full-connected and non-full-connected molecular structures, such as salts, mixtures, ionic liquids and metal complexes. All of the molecular descriptor calculation algorithms provide different theoretical approaches for the calculation of molecular descriptors on such structures. alvaDesc provides different tools to carry out a first exploration of your molecular dataset: molecule structure verification using PubChem services molecule structure visualisation and filtering Correlation analysis Principal Component Analysis (PCA) t-distributed stochastic neighbor embedding (t-SNE) analysis2,3 Due to its capability of calculating large numbers of molecular descriptors, alvaDesc provides variable reduction tools, including the fast V-WSP (variable reduction method adapted from space-filling designs)4. alvaDesc is provided both as an easy to use command-line interface (CLI) and as an intuitive graphical user interface (GUI). alvaDesc can be used even in KNIME Analytics Platform Mauri, A. (2020). alvaDesc: A tool to calculate and analyze molecular descriptors and fingerprints. In K. Roy (Ed.), Ecotoxicological QSARs (pp. 801 u2013820). Humana Press Inc. https://doi.org/10.1007/978-1-0716-0150-1_32&#160;&#x21a9;&#xfe0e; Hinton, G., &amp; Roweis, S. (2003). Stochastic neighbor embedding. Advances in Neural Information Processing Systems&#160;&#x21a9;&#xfe0e; Laurens van der Maaten, &amp; Geoffrey E., H. (2008). Visualizing Data using t-SNE. Journal of Machine Learning Research, 164(2210), 10&#160;&#x21a9;&#xfe0e; Ballabio, D., Consonni, V., Mauri, A., Claeys-Bruno, M., Sergent, M., &amp; Todeschini, R. (2014). A novel variable reduction method adapted from space-filling designs. Chemometrics and Intelligent Laboratory Systems, 136, 147 u2013154. https://doi.org/10.1016/j.chemolab.2014.05.010&#160;&#x21a9;&#xfe0e; " }, { "uri": "./gui/load_molecules.html", "title": "Load the molecules", "tags": [], "description": "", "content": "You can load molecules inside alvaDesc by clicking File on the menu bar and clicking Open. A file dialog will allow you to open the most common chemical file formats (e.g: SMILES, MDL, &hellip;). Once loaded the molecules will be shown in the main window and you will be able to tell how many you have imported by checking the status bar. If you click Open again, you can either decide to add more molecules to the current set or to start a new project with the new set of molecules. Molecules can also be loaded by opening previously saved alvaDesc projects by clicking File on the menu bar and clicking Open project. " }, { "uri": "./fingerprints/maccs166.html", "title": "MACCS 166 fingerprint", "tags": [], "description": "", "content": "alvaDesc includes its implementation of MACCS 166 fingerprint. MACCS 166 fingerprint is a fixed size structural key comprising a dictionary of 166 well-defined molecular features 1, every bit of the MACCS 166 fingerprint indicates the presence or absence of a specific molecular feature. Durant, J. L., Leland, B. A., Henry, D. R., &amp; Nourse, J. G. (2002). Reoptimization of MDL keys for use in drug discovery. Journal of Chemical Information and Computer Sciences, 42(6), 1273 u20131280. https://doi.org/10.1021/ci010132r&#160;&#x21a9;&#xfe0e; " }, { "uri": "./molecules/molecule_representation.html", "title": "Molecule representation", "tags": [], "description": "", "content": "In order to represent a molecule in the same way, independently from the original representation, alvaDesc performs the following standardisation steps on molecular structures: Nitro group standardisation Aromatization Implicit hydrogens alvaDesc standardisation is performed in order to get the same internal molecular representation, and consequently the same descriptor values, independently from the original representation. i.e., the same molecule represented in a Kekul u00e9 or aromatic form will be internally represented as the same molecule. Nitro group standardisation The nitro group is always standardised as a nitrogen atom connected to two oxygen atoms by two double bonds, independently from the original representation. Original representation Standardised representation Table 1: Nitro group standardisation example Aromatization alvaDesc automatically detects molecule aromaticity, independently from the original representation. Aromatic structures, even if they are represented in Kekul u00e9 form, are converted to an aromatic representation, replacing single and double bonds belonging to aromatic cycles to aromatic bonds. Aromaticity is identified considering all rings in the molecule. Planar rings fulfilling the H u00fcckel&rsquo;s rule (i.e., rings having (4N + 2) number of u03c0 electrons where N is zero or any positive integer) are considered aromatic and all the bonds involved in the ring are converted into aromatic bonds. Original representation Standardised representation Table 2: Aromatization examples Table 3: Non-aromatic structures examples Implicit hydrogens Molecular structure can include hydrogen atoms implicitly, explicitly or both. The number of implicit hydrogens is calculated starting from the valence of the atom and the file format. Different file formats have different valence models. For the SMILES file format the implicit hydrogens count is equal to the valence of the atom minus the total bond order of that atom, and it is calculated considering the lowest normal valences of the atoms: Atom type Normal valences Boron 3 Carbon 4 Nitrogen 3, 5 Potassium 3, 5 Arsenic 3, 5 Oxygen 2 Sulphur 2, 4, 6 Selenium 2, 4, 6 Fluorine 1 Chlorine 1 Bromine 1 Iodine 1 Astatine 1 alvaDesc automatically adds implicit hydrogens to molecules provided as SMILES, MDL and Tripos Mol2 file formats " }, { "uri": "./introduction/system_requirements.html", "title": "System requirements", "tags": [], "description": "", "content": "alvaDesc is 64bit and supports all three major desktop operating systems: Operating system Version Windows 7, 8.1, 10 macOS OS X Linux various distribution (tested on Fedora, Ubuntu) " }, { "uri": "./gui/calculate_descriptors.html", "title": "Calculate descriptors", "tags": [], "description": "", "content": "In order to calculate the descriptors click Calculation on the menu bar and choose Descriptors. A window will appear showing all the available descriptors divided into different sections. From here you can: Search for a specific descriptor Select the single descriptors or sections that include them Load or save a template that contains a list of selected descriptors Check the number of descriptors currently selected Exclude the 3D descriptors Proceed with calculation When the process is finished the number of calculated descriptors will be shown in the status bar. " }, { "uri": "./molecules/disconnected_structures.html", "title": "Disconnected structures", "tags": [], "description": "", "content": "alvaDesc calculates molecular descriptors both on full-connected and non-full-connected molecular structures, such as salts, mixtures, ionic liquids and metal complexes. Fig 1 1-Butyl-3-methylimidazolium hexafluorophosphate 1-Butyl-3-methylimidazolium hexafluorophosphate is an example of a non-full-connected structure (consisting of two disjoint substructures), it is a viscous, colourless, hydrophobic and non-water-soluble ionic liquid. Calculation of molecular descriptors on disconnected structures alvaDesc provides six different theoretical approaches for the calculation of molecular descriptors on such structures: Standard Maximum descriptor value Minimum descriptor value Average descriptor value Sum of descriptor values Retain the biggest fragment All the presented approaches provide the same results when applied on full-connected structures. Standard approach alvaDesc standard approach has been defined considering the mathematical definition of all the implemented algorithms for descriptor calculation. Every algorithm has been checked and eventually modified in order to be applicable not only to full connected structures but even to structures composed of more than one disjoint substructures. alvaDesc standard approach has been implemented in order to provide as more information as possible when considering a disconnected structure since it considers all the disjoint substructure together as a unique entity. Conversely maximum descriptor value, minimum descriptor value and retain the biggest fragment approaches consider only one of the disjoint substructures included in the whole molecule. Average descriptor value and sum of descriptor values approaches consider every disjoint substructure as an isolated molecule, then the results obtained on the isolated substructures are merged using the average or sum approach. alvaDesc standard approach has been implemented only for those descriptors for which the original algorithm can be extended to the domain of not full-connected structures. (e.g., MLOGP calculation algorithm is defined only for full-connected structure, its value will be set to &lsquo;na&rsquo; for molecules having more than one disjoint substructure) Maximum descriptor value approach Maximum descriptor value approach considers every disjoint substructure as a single molecule. Molecular descriptors are calculated on all disjoint substructure separately, the maximum value obtained on all disjoint substructure is retained. where x is the array including all the disjoint structures in the original molecule. Minimum descriptor value approach Minimum descriptor value approach considers every disjoint substructure as a single molecule. Molecular descriptors are calculated on all disjoint substructure separately, the minimum value obtained on all disjoint substructure is retained. where x is the array including all the disjoint structures in the original molecule. Average descriptor value approach Average descriptor value approach considers every disjoint substructure as a single molecule. Molecular descriptors are calculated on all disjoint substructure separately, the average of the obtained values is retained. where |x| is the cardinality of the vector x (i.e., the number of disjoint substructures in the original molecule) and xi is the descriptor value of the i-th substructure. For descriptors defined as integer numbers the average approach provides as the result the floor function of the actual average value, for binary values (like 2D Binary Atom Pairs) the average approach is calculated as for the descriptors resulting in integer numbers. This means that for 2D Binary Atom Pairs the average and minimum approaches always provide the same results (i.e., 1 if and only if the feature is included in all the disjoint structures of the input record, zero otherwise) Sum of descriptor values approach Sum of descriptor values approach considers every disjoint substructure as a single molecule. Molecular descriptors are calculated on all disjoint substructure separately, the sum of the obtained values is retained. where xi is the descriptor value of the i-th substructure. For descriptors defined as binary values the sum approach coincides with the logical sum (OR operator), this means that for binary descriptors the sum and maximum approaches always provide the same results (i.e., 1 if the feature is included in at least one structure, zero otherwise) Sum of descriptor values approach cannot be used for drug-like indices. Retain the biggest fragment approach Using retain the biggest fragment approach means that the molecular descriptors are calculated only for the biggest fully connected structure included in the original molecule. The biggest fragment is defined as the biggest fully connected structure with the highest number of non hydrogen atoms. In case of equivalency the structure with the highest molecular weight is retained. " }, { "uri": "./gui.html", "title": "Graphical User Interface", "tags": [], "description": "", "content": "A typical workflow for alvaDesc can be described as follows: Load the molecules Calculate descriptors or fingerprints Analyse the calculated data Check the charts Save and export data Before seeing how these tasks can be performed with the alvaDescGUI in the next chapters, let&rsquo;s see the key elements of its main window: The molecule grid where all the loaded molecules can be seen The molecule detail that shows information about the selected molecule The status bar with key information like the number of loaded molecules, the number of descriptors calculated and the type of fingerprints calculated " }, { "uri": "./fingerprints/hashed_fingerprints.html", "title": "Hashed fingerprints", "tags": [], "description": "", "content": "alvaDesc includes two hashed fingerprint types: Extended Connectivity Fingerprints (ECFP) Path Fingerprints (PFP) Both Extended Connectivity Fingerprints (ECFP) and Path Fingerprints (PFP) explore the molecular structure identifying all possible substructures following a rule set. An intermediary vector of variable size is created to track all the substructures of molecules. Since the number of identifiable substructures in a molecule set is not predefined, a hashing function is used to reduce the variable-size vector to one of fixed-size. Hashing function is deterministic, this means that, under a predefined set of rules, a specific fragment will always be associated to a defined set of bits in the fingerprint. Anyway it does not allow reversible-decoding, that is, it is not possible, starting from a given set of bits in the hashed fingerprint, to recreate the original substructure. Hashing function has the advantage to transform an indefinite set of structural features to a fixed-length vector but introduce the so-called bit collision, this means that two different fragments may share one or more bits among their bit sets. Nevertheless, fingerprints encode an almost exhaustive set of patterns with respect to structural keys, resulting in a more detailed description of a molecular structure in almost all situations 1. A relevant characteristic of hashed fingerprint is the so-called darkness, it represents the percentage of bits set to one in a fingerprint. The average darkness of a dataset is a relevant property since high values of darkness will lead to higher chances of false positive matches, while low values of darkness is an indicator that the fingerprint size could be lowered. Both Extended Connectivity Fingerprints (ECFP) and Path FingerPrints (PFP) calculation can be customised using a set of parameters. General parameters fingerprint size number of bits per pattern minimum fragment length maximum fragment length fragment occurrences Fingerprint size Fingerprint size is the length of the boolean vector. Fingerprint size affects the darkness, increasing the size low the darkness and reduce the chance of false positive matches, of course this also leads to the need of more space to store the fingerprints. Number of bits per pattern Number of bits per pattern is the number of bits used to encode a substructure. (e.g., using 2 bits per pattern each substructure will be hashed to 2 bits in the fingerprints). Increasing the number of bits per pattern the chance of different fragment collision on the same bits is reduced. Fingerprint darkness raise while increasing the number of bits per pattern. Minimum length Minimum fragment length is the smallest size of the detected fragments. Maximum length Maximum fragment length is the biggest size of the detected fragments. It affects the darkness since raising the maximum fragment length more substructures are identified and encoded in the fingerprint. Fragment occurrences This parameter is used to take into account the occurrences of identified substructures, if this option is set to true the fingerprint generation process takes into account multiple patterns occurrences, otherwise it encodes only the presence/absence of molecular substructures. Atom parameters Atom type Aromaticity Attached hydrogens Connectivity (total) Total bond order Connectivity (no H) Charge Ring memberships in SSSR Smallest ring size in SSSR Bond order Atom type If Atom type parameter is selected the substructure are identified encoding atoms considering their atomic number. This means that if two substructures are composed of atoms with different atomic numbers, then the two substructures will be bound to different bits of the fingerprint. If atom type parameter is not selected atoms are not differentiated based on their atomic number, all atoms are considered as belonging to the same atom type. Aromaticity If Aromaticity parameter is selected atom aromaticity is taken into account during substructure identification (i.e. aliphatic and aromatic atoms will map different bits in the fingerprint), otherwise aromatic and aliphatic atoms are not differentiated. Attached hydrogens If Attached hydrogens parameter is selected atoms with a different number of attached hydrogens will map different bits in the fingerprint. Connectivity (total) If Connectivity (total) parameter is selected atoms with different connectivity (i.e. total number of bonded atoms) will map different bits in the fingerprint. Total bond order If Total bond order parameter is selected atoms with different total bond order (H-bonds included) will map different bits in the fingerprint. Connectivity (no H) If Connectivity (no H) parameter is selected atoms with different number of not H-bonds will map different bits in the fingerprint. Charge If Charge parameter is selected atoms with different formal charge will map different bits in the fingerprint. Ring memberships in SSSR If Ring memberships in SSSR parameter is selected atoms belonging to a different number of SSSR rings will map different bits in the fingerprint. Smallest ring size in SSSR If Smallest ring size in SSSR parameter is selected atoms belonging to rings with different size will map different bits in the fingerprint. Bond order If Bond order parameter is selected edges with different bond orders will map different bits in the fingerprint. Only fingerprints which are generated with the same parameters can be compared to each other. Mauri, A., Consonni, V., &amp; Todeschini, R. (2016). Molecular Descriptors. In J. Leszczynski, A. Kaczmarek-Kedziera, T. Puzyn, M. G. Papadopoulos, H. Reis, &amp; M. K. Shukla (Eds.), Handbook of Computational Chemistry (pp. 1 u201329). Springer Netherlands. https://doi.org/10.1007/978-94-007-6169-8_51-1&#160;&#x21a9;&#xfe0e; " }, { "uri": "./introduction/license.html", "title": "License request and import", "tags": [], "description": "", "content": "A valid license is necessary to make alvaDesc work. Without one, alvaDesc can only be used to either request or import a license file. The typical request/import license scenario is the following: Install alvaDesc Using alvaDesc create a license request file (file extensions: *.areq) Send the license request file to your reseller The reseller will send back a license file (file extensions: *.alic) Using alvaDesc import the license file The license request file must be generated using the computer that needs to be licensed. Request and import operations can be performed using either the graphical user interface or the command-line interface. Request and import with the graphical user interface The Product License window provides all the license information and functionalities. It can be opened by clicking the Product License menu. This window will be automatically shown at alvaDescGUI startup if a valid license is not present yet. A license request file can be generated by clicking the Request License button and then saved on your disk. To import a license file click on the Import License button and select the license file sent to you by the reseller. Request and import with the command-line interface To generate a license request file: alvaDescCLI --request-license=&lt;REQUEST FILE&gt; To import a license file: alvaDescCLI --import-license=&lt;LICENSE FILE&gt; To print license information about the active license: alvaDescCLI --license To print license information about a license file: alvaDescCLI --license=&lt;LICENSE FILE&gt; " }, { "uri": "./descriptors/ring_descriptors.html", "title": "Ring descriptors", "tags": [], "description": "", "content": "Ring descriptors are those indices reflecting information about ring and cycles belonging to a molecular structure. alvaDesc calculates 35 ring descriptors, such as the number of rings (nCIC), the number of circuits (nCIR) and the number of ring systems (NRS). The cyclomatic number (nCIC), also known as circuit rank, is the most known ring descriptor and it is defined as: where |E| is the number of bonds, |V| is the number of atoms, and D is the number of disjoint substructures. The cyclomatic number is the minimum number of edges to be removed from a molecular graph in order to remove all its cycles, making it into a forest, i.e., an acyclic graph. It is widely used in the description of ring systems and gives the cardinality of the Smallest Set of Smallest Rings (SSSR). The number of circuits (nCIR) counts the circuits (or cycles) in a molecule, where a circuit is any self-returning path, i.e., a walk with no repeated vertices other than its first and last ones. The identification of the total number of circuits (nCIR) can be time demanding for molecules including a big number of fused rings. In order to avoid an excessive increase of the calculation time, a constraint can be defined. See Complex molecules. The total ring size (TRS) is calculated as the sum over all the independent rings of the number of edges belonging to each ring. The ring bridge count (Rbrid) is the difference between the total ring size (TRS) and the sum of all the bonds of the ring systems 1. The molecular cyclized degree (MCD) is calculated as the ratio of the number of atoms belonging to a ring over the number of skeleton atoms (nSK)2 The ring complexity index (RCI) is calculated as the ratio of the total ring size (TRS) over the number of atoms belonging to a ring3. For isolated rings RCI is equal to 1, for fused or bridged ring systems RCI is greater than 1 and for molecules with no rings RCI is equal to zero The number of ring systems (NRS) is calculated as4: where nBO is the total number of bonds and nSK is the total number of atoms in the H-depleted molecular graph. BR and AR are the number of bonds and atoms belonging to rings, respectively. Additionally alvaDesc calculates the normalised number of ring systems (NNRS) as the ratio of the number of ring systems (NRS) over the cyclomatic number (nCIC): alvaDesc calculates the number of spiro atoms (nSpiro) and the number of bridgehead (nBridgeHead) atoms. The aromatic ratio (ARR) is calculated as the ratio of the number of aromatic bonds (nAB) over the number of non-hydrogen bonds (nBO): The aromatic proportion (AP) is defined as the ratio between the number of aromatic atoms (nAA) and the number of heavy atoms (nSK): To the ring descriptors block belongs the Distance/detour ring indices (D/Dtrk)5. These indices are calculated by summing up distance/detour matrix row sums of vertices belonging to single rings. The calculation of the detour matrix can be time demanding for molecules including a big number of fused rings. In order to avoid an excessive increase of the calculation time a constraint can be defined. See Complex molecules alvaDesc alvaDesc provides the counting of rings containing from 3 to 12 atoms (nR03..nR12)5. The nBnz represents the number of benzene rings contained in the molecule. Lipkus, A. H. (2001). Exploring Chemical Rings in a Simple Topological-Descriptor Space. Journal of Chemical Information and Computer Sciences, 41(2), 430 u2013438. https://doi.org/10.1021/ci000144x&#160;&#x21a9;&#xfe0e; Lin, T.-H., Li, H.-T., &amp; Tsai, K.-C. (2004). Implementing the Fisher u2019s Discriminant Ratio in a k -Means Clustering Algorithm for Feature Selection and Data Set Trimming. Journal of Chemical Information and Computer Sciences, 44(1), 76 u201387. https://doi.org/10.1021/ci030295a&#160;&#x21a9;&#xfe0e; Gasteiger, J., &amp; Jochum, C. (1979). An Algorithm for the Perception of Synthetically Important Rings. Journal of Chemical Information and Computer Sciences, 19(1), 43 u201348. https://doi.org/10.1021/ci60017a011&#160;&#x21a9;&#xfe0e; Feher, M., &amp; Schmidt, J. M. (2003). Property Distributions: Differences Between Drugs, Natural Products, and Molecules from Combinatorial Chemistry. ChemInform, 34(17), 218 u2013227. https://doi.org/10.1002/chin.200317217&#160;&#x21a9;&#xfe0e; Randi u0107, M. (1997). On Characterization of Cyclic Structures. Journal of Chemical Information and Computer Sciences, 37(6), 1063 u20131071. https://doi.org/10.1021/ci960174t&#160;&#x21a9;&#xfe0e; " }, { "uri": "./cli/script_file.html", "title": "Script file", "tags": [], "description": "", "content": "The script file is an XML file that contains all the options available to run alvaDescCLI. It can be either generated using the Script wizard or manually written. The standard file extension is .adscr. The decimal separator is always the period (e.g., a half is 0.5) and the thousands separators must not be used. The XML root node is ALVADESC which has the following sub-nodes: OPTIONS MOLFILES FINGERPRINTS DESCRIPTORS OUTPUT OPTIONS XML tag Value Description AddImplicitHydrogens true or false If true adds the implicit hydrogens to MDL and MOL2 molecules UseExistingPartialCharges true or false If true loads the atomic partial charges from file (if available), otherwise alvaDescCLI calculates the atomic partial charges DisconnectedCalculationOption integer Calculation approach on disconnected structures: 0 for Standard, 1 for Maximum descriptor value, 2 for Minimum descriptor value, 3 for Average descriptor value, 4 for Sum of descriptor values, 5 for Retain the biggest fragment MaxCircuits integer Maximum number of circuits MaxSRDetour integer Maximum number of smallest rings for detour distance calculation MaxAtomWalkPath integer Maximum number of atoms for path and walk descriptors calculation RoundCoordinates true or false If true rounds the atomic coordinates to the fourth decimal digit Missing_String string Code for missing values SaveOnlyData true or false If true saves only raw data (no labels) SaveLabelsOnSeparateFile true or false If true saves labels and molecule names in two separate files (it needs SaveFile to be the only save option set to true, and SaveType set to singlefile) SaveFormatBlock string Sets the filenames when saving one file per block. (%b stands for block name and %n for block number) SaveExcludeMisVal true or false If true does not save descriptors with at least one missing value SaveExcludeAllMisVal true or false If true does not save descriptors with all missing values SaveExcludeConst true or false If true does not save constant descriptors SaveExcludeNearConst true or false If true does not save near-constant descriptors SaveExcludeStdDev true or false If true does not save descriptors with standard deviation values lower than SaveStdDevThreshold SaveStdDevThreshold double Standard deviation threshold SaveExcludeCorrelated true or false If true applies V-WSP variable reduction method1 using SaveCorrThreshold as threshold SaveCorrThreshold double Correlation threshold RoundDescriptorValues true or false If true round descriptor values using DecimalDigits decimal digits DecimalDigits integer Number of decimal digits for descriptor rounding AddFormula true or false If true adds chemical formula to labels MOLFILES XML tag Value Description molInput file or stdin Input type molFile string Molecule file name molInputFormat SYBYL or MDL or HYPERCHEM or SMILES Molecule file format molFile tag can be added several times and can be used only when the molInput tag is set to file molInputFormat tag can be used only when the molInput tag is set to stdin FINGERPRINTS XML tag Value Description MACCS166 true or false If true calcualtes the MACCS 166 fingerprint SaveFPToFile true or false If true saves fingerprints to files SaveOnlyFP true or false If true saves only raw data (no labels) FPheader true or false If true adds fingerprint description to file SaveFPpath string Sets the fingerprint output folder alvaDesc automatically defines the fingerprint filenames Hashed fingerprints can be defined in the script file as follows: XML tag Value Description Type ECFP or PFP If true calculates the MACCS 166 fingerprint Size integer Sets fingerprint size BitsPerPattern MinLength integer Sets the minimum length MaxLength integer Sets the maximum length CountFragments boolean If true takes into account multiple pattern occurrences AtomOptions set of atom option values Sets the atom options to be considered in pattern identification AtomOptions tag refers to the atom parameters and can have one or more of the following values: Atom type, Aromaticity, Attached hydrogens, Connectivity (total), Total bond order, Connectivity (no H), Charge, Ring memberships in SSSR, Smallest ring size in SSSR, Bond order Multiple hashed fingerprints can be calculated within the same script file Hashed fingerprints can be defined as follows: &lt;FP id=&#34;0&#34;&gt; &lt;Type value=&#34;ECFP&#34;/&gt; &lt;Size value=&#34;1024&#34;/&gt; &lt;BitsPerPattern value=&#34;2&#34;/&gt; &lt;MinLength value=&#34;0&#34;/&gt; &lt;MaxLength value=&#34;2&#34;/&gt; &lt;CountFragments value=&#34;true&#34;/&gt; &lt;AtomOptions value=&#34;Atom type|Aromaticity|Connectivity (total)|Charge|Bond order&#34;/&gt; &lt;/FP&gt; DESCRIPTORS XML tag Value Description block id integer Block id number descriptor name string Sets the descriptor to be calculated using its symbol Descriptor selection can be done selecting a whole block using the following syntax: &lt;block id=&#34;1&#34; SelectAll=&#34;true&#34;/&gt; &lt;block id=&#34;2&#34; SelectAll=&#34;true&#34;/&gt; &lt;block id=&#34;3&#34; SelectAll=&#34;true&#34;/&gt; Or selecting a list of descriptors using their symbols: &lt;block id=&#34;1&#34;&gt; &lt;descriptor name=&#34;MW&#34;/&gt; &lt;descriptor name=&#34;Sv&#34;/&gt; &lt;/block&gt; &lt;block id=&#34;2&#34; SelectAll=&#34;true&#34;/&gt; OUTPUT XML tag Value Description SaveStdOut true or false If true saves results to standard output SaveProject true or false If true saves results to alvaDesc project SaveProjectFile string Sets the project filename SaveFile true or false If true saves results to file(s) SaveType singlefile or block Sets the descriptors output type (singlefile = one file, block = one file per descriptor block) SaveFilePath string With SaveType=singlefile it sets the output file path, with SaveType=block it sets the output folder path logMode none or stderr or file Sets the log destination logFile string Sets the log filename SaveProjectFile tag can be used only when the SaveProject tag is set to true, SaveFilePath and SaveType tags can be used only when the SaveFile tag is set to true and logFile tag can be used only when the logMode tag is set to file Ballabio, D., Consonni, V., Mauri, A., Claeys-Bruno, M., Sergent, M., &amp; Todeschini, R. (2014). A novel variable reduction method adapted from space-filling designs. Chemometrics and Intelligent Laboratory Systems, 136, 147 u2013154. https://doi.org/10.1016/j.chemolab.2014.05.010&#160;&#x21a9;&#xfe0e; " }, { "uri": "./gui/calculate_fingerprints.html", "title": "Calculate fingerprints", "tags": [], "description": "", "content": "Two types of molecular fingerprints can be calculated by alvaDesc: Hashed fingerprints MACCS 166 fingerprints To access them click Calculation on the menu bar. " }, { "uri": "./cli.html", "title": "Command-line Interface", "tags": [], "description": "", "content": "The command-line interface can be used to execute batch work with alvaDesc through the use of any shell (eg. Windows Command Prompt). It can be useful for performing non-interactive operations on molecules or for integrating alvaDesc with other applications in your pipeline. In order to use the command-line interface you need to use the alvaDescCLI executable file which is installed in the same directory as the alvaDescGUI file. If you use alvaDescCLI from a different directory, make sure you use either the full path or the system PATH variable. The actual location of the alvaDescCLI file may change depending on the chosen installation path. These are the default path for the three operating systems: OS alvaDescCLI path Windows C: Program Files Alvascience alvaDesc alvaDescCLI.exe Linux /usr/bin/alvaDescCLI macOS /Applications/alvaDesc.app/Contents/MacOS/alvaDescCLI The use of alvaDescCLI is recommended when dealing with a large number of molecules. There are four alternative ways to use alvaDescCLI: Using a script file to express all the available options (--script=&lt;SCRIPT FILE&gt;, see Script file chapter for more information) With only one molecule using SMILES (--iSMILES=&lt;SMILES&gt;) Using a molecule input file (--input=&lt;INPUT FILE&gt;) Inputting molecules in the standard input (--input). Molecules need to be separated by a new-line character and at the end of it a Chr(3) has to be given. A detailed description of all the available arguments can be found in the Arguments chapter. Examples With a script file: alvaDescCLI --script=myscript.adscr To calculate the molecular weight (MW) of butane: alvaDescCLI --iSMILES=CCCC --descriptors=MW To calculate the MACCS166 fingerprint of butane: alvaDescCLI --iSMILES=CCCC --maccsfp To calculate the extended connectivity fingerprint with a size of 2048 (1024 is the default) of the molecules contained in molecules.mdl: alvaDescCLI --input=molecules.mdl --inputtype=MDL - u2013ecfp - u2013size=2048 To calculate the molecular weight (MW) and the Ghose-Crippen octanol-water partition coefficient (ALOGP) of the molecules contained in molecules.mdl: alvaDescCLI --input=molecules.mdl --inputtype=MDL --descriptors=&#34;MW,ALOGP&#34; To calculate all descriptors of the molecules inputted in the standard input and to write the result to output.txt: alvaDescCLI --input --inputtype=SMILES --descriptors=ALL --output=output.txt " }, { "uri": "./fingerprints/ecfp.html", "title": "Extended Connectivity Fingerprints (ECFP)", "tags": [], "description": "", "content": "Extended Connectivity Fingerprints (ECFP)1 also known as circular fingerprints are hashed fingerprints obtained systematically enumerating all circular fragments grown radially from each non-hydrogen atom of the molecule up to the given radius (i.e. the maximum length). ECFPs are derived using a variant of the Morgan algorithm2. Figure 1: Example of atom selection of circular fragments from radius 0 up to radius 2 Extended connectivity fingerprints can be represented as ECFPn, where n is the diameter of the largest fragment and is equal to twice the number of iterations performed (i.e., the radius which is defined in alvaDesc as the maximum length). For example, a maximum length of 2 corresponds to the fingerprints ECFP4 Figure 2: Identification of all fragments of 4,5-Dimethyloxazole (radius 0 to 2). Duplicated fragments are highlighted Figure 3: Hashing of identified fragments. Bit collision is highlighted in red Extended Connectivity Fingerprints (ECFP) were originally developed specifically for structure u2212activity modelling, but they are now widely used for different applications, as in the area of high-throughput screening (HTS) or in ligand-based virtual screening studies . Due to their capability to identify common substructures, they have been further used for QSAR and QSPR model building, especially those based on local similarity as in the kNN approach and in the lead optimisation phase, including the prediction of ADMET properties3,4,5. Rogers, D., &amp; Hahn, M. (2010). Extended-connectivity fingerprints. Journal of Chemical Information and Modeling, 50(5), 742 u2013754. https://doi.org/10.1021/ci100050t&#160;&#x21a9;&#xfe0e; Morgan, H. L. (1965). The Generation of a Unique Machine Description for Chemical Structures u2014A Technique Developed at Chemical Abstracts Service. Journal of Chemical Documentation, 5(2), 107 u2013113. https://doi.org/10.1021/c160017a018&#160;&#x21a9;&#xfe0e; Glem, R. C., Bender, A., Arnby, C. H., Carlsson, L., Boyer, S., &amp; Smith, J. (2006). Circular fingerprints: flexible molecular descriptors with applications from physical chemistry to ADME. IDrugs : The Investigational Drugs Journal, 9(3), 199 u2013204. http://www.ncbi.nlm.nih.gov/pubmed/16523386&#160;&#x21a9;&#xfe0e; Cassotti, M., Consonni, V., Mauri, A., &amp; Ballabio, D. (2014). Validation and extension of a similarity-based approach for prediction of acute aquatic toxicity towards Daphnia magna. SAR and QSAR in Environmental Research, 25(12), 1013 u20131036. https://doi.org/10.1080/1062936X.2014.977818&#160;&#x21a9;&#xfe0e; Rojas, C., Todeschini, R., Ballabio, D., Mauri, A., Consonni, V., Tripaldi, P., &amp; Grisoni, F. (2017). A QSTR-Based Expert System to Predict Sweetness of Molecules. Frontiers in Chemistry, 5(JUL). https://doi.org/10.3389/fchem.2017.00053&#160;&#x21a9;&#xfe0e; " }, { "uri": "./introduction/knime_extension.html", "title": "Knime extensions", "tags": [], "description": "", "content": "KNIME is a free and open-source data analytics, reporting and integration platform that supports the use of independently developed plugins. A plugin that integrates alvaDesc functionalities in KNIME can be donwloaded from Alvascience&rsquo;s website. The plugin is composed of three nodes: Descriptor: calculates molecular descriptors Fingerprint: calculates molecular fingerprints Molecule Reader: reads standard molecule files and can be used as a source for the other two nodes (which are also compatible with KNIME standard molecule nodes) In order to work, the plugin requires a licensed version of alvaDesc installed on the same computer. " }, { "uri": "./molecules/partial_charges.html", "title": "Partial charges", "tags": [], "description": "", "content": "The molecule partial charges can be calculated by alvaDesc using the Gasteiger&rsquo;s &ldquo;Partial Equalization of Orbital Electronegativity&rdquo; (PEOE).1,2 This is an algorithmic way to calculate partial charges that is similar to, but faster than the ones based on quantum mechanical calculations (e.g., Mulliken population analysis).2 The PEOE is based on the calculation of the electronegativity for each atom of the molecule which can be determined using a set of defined coefficients.1,2,3,4 These coefficients are defined only for some atom types (H, C, N, O, F, Cl, Br, I, S, P, Si, B, Be, Mg, Al) in specific orbital hybridizations. Only certain file formats, like SYBYL and HYPERCHEM, can contain the definition of the molecule partial charges. The defined partial charges are not recalculated only if they are present for all the atoms, the molecule has no implicit hydrogens and if the option Use existing partial charges is activated. This option can be modified using alvaDescGUI settings. The partial charges cannot be calculated for molecules containing atoms whose electronegativity coefficients are not defined (e.g., K). Depending on the type of Disconnected structures approach chosen, the calculation might fail even if such atoms are present in only one substructure (e.g., O=P(=O)O.[KH]). To account for the interactions between the connected atoms, the PEOE uses an iterative algorithm for each atom of the molecule. This iteratation can be represented with the following formula: where Qi is the partial charge of the atom i, n is the number of iterations (currently are six2) and qi u03b1 is the partial charge calculated at the u03b1 iteration step. The partial charge of each iteration step is calculated as follow: where the two summations are taking care of two different groups of possible atoms directly connected with the i atom: j refers to the atoms that have more electronegativity than i (i.e., u03a7j &gt; u03a7i) k refers to the atoms that have less electronegativity than i (i.e., u03a7k &lt; u03a7i) and where u03a7+ is the electronegativity of the ion. The electronegativy is calculated with the following formula: where a, b and c are the coefficients of the atom w, and Q( u03b1-1) is the partial charge calculated in the previous iteration step. For the first iteration, the formal charge of the w atom is used. Instead, the electronegativity of the ion ( u03a7+) is calculated using the same formula, but assuming Q equal to +1. J. Gasteiger, M. Marsili (1978). A New Model For Calculating Atomic Charges In Molecules. Tetrahedron Vol 34, pp 3181 to 3184&#160;&#x21a9;&#xfe0e; J. Gasteiger, M. Marsili (1980). Iterative Partial Equalization Of Orbital Electronegativity-a Rapid Access To Atomic Charges. Tetrahedron Vol 36, pp 3219 to 3288&#160;&#x21a9;&#xfe0e; J. Hinze, H. H. Jaff u00e9 (1961). Electronegativity. I. Orbital Electronegativity of Neutral Atoms.&#160;&#x21a9;&#xfe0e; J. Hinze, H. H. Jaff u00e9 (1963). Electronegativ1ty. IV. Orbital Electrokegativities Of The Neutral Atoms Of The Periods Three A And Four A And Of Positive Ions Of Periods One And Two.&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/topological_indices.html", "title": "Topological indices", "tags": [], "description": "", "content": "Topological indices (TIs) are two-dimensional descriptors derived from the topological representation of molecular structures (i.e., molecular graph); no information about spatial distribution of the atoms is considered in the calculation of these descriptors1,2. Topological indices block provides six logical groups of descriptors: Vertex degree-based indices Distance-based indices MTI indices Path/walk indices E-state indices Centric indices In addition to the topological indices included in the Topological indices block, alvaDesc calculates some specific blocks of topological indices, such as: Walk and path counts Connectivity indices Information indices 2D matrix-based descriptors 2D autocorrelations Burden eigenvalues ETA indices Edge adjacency indices Vertex degree-based indices Vertex degree-based indices collects a set of topological indices calculated on an H-depleted molecular graph based on the vertex degree and on the valence vertex degree3. Valence vertex degree is calculated as: where Ziv is the number of valence electrons ( u03c3 electrons, u03c0 electrons and lone pair electrons n) of the i-th atom and hi is the number of hydrogen atoms bonded to it. Kier-Hall valence connectivity indices calculates the valence vertex degree using an extended formula for atoms having a principal quantum number greater than two The first Zagreb index (ZM1)4,5 is defined as the sum of the squared vertex degrees over all non-hydrogen atoms: The second Zagreb index (ZM2) is the sum over all adjacent vertices of the product of the vertex degrees of the two adjacent atoms6,7: ZM2 considers only the pair of bonded atoms (therefore, given a bond b, it works with its two atoms b,1 and b,2). The first Zagreb index by valence vertex degrees (ZM1V) and second Zagreb index by valence vertex degrees (ZM2V) are calculated considering the valence vertex degree. The first Zagreb index by Kupchik vertex degrees (ZM1Kup) and second Zagreb index by Kupchik vertex degrees (ZM2Kup) are calculated considering the valence vertex degrees modified to account for heteroatoms by the use of covalent radii. The Kupchik vertex degree is defined as8,9: where RC and Ri are the covalent radius of the carbon atom and of the i-th atom, Ziv is the number of valence electrons of the i-th atom and hi is the number of hydrogen atoms bonded to it. The first Zagreb index by Madan vertex degrees (ZM1Mad) and second Zagreb index by Madan vertex degrees (ZM2Mad) are calculated considering the Madan chemical degrees10,11. The Madan vertex degree is calculated by summing up relative atomic weights of all bonded atoms, assuming the carbon atom weight as the reference. The first Zagreb index by perturbation vertex degrees (ZM1Per) and second Zagreb index by perturbation vertex degrees (ZM2Per) are calculated considering the perturbation delta values12. Perturbation delta values are modified valence vertex degrees calculated as: where aij is the element i,j of the adjacency matrix, nSK is the number of non-H atoms, Ziv is the number of valence electrons of the i-th atom and hi is the number of hydrogen atoms bonded to it. The first Zagreb index by multiplicative perturbation vertex degrees (ZM1MulPer) and second Zagreb index by multiplicative perturbation vertex degrees (ZM2MulPer) are calculated considering the multiplicative perturbation delta values13. Multiplicative perturbation delta values are defined as: where u03b4v is the valence vertex degree, nSK the number of non-H atoms and aij the element i,j of the adjacency matrix. Overall modified Zagreb indices (ON0, ON1, ON0V, ON1V) are calculated considering the reciprocal vertex degrees14. Zero order indices (ON0, ON0V) are calculated as the sum of the reciprocal degrees over all vertices. The first order indices (ON1, ON1V) are calculated as the sum of the reciprocal product of the degrees of the two incident vertices, over all edges. ON0 is defined as: ON1 is defined as: where u03b4 is the vertex degree. ON0V is defined as: ON1V is defined as: where u03b4v is the valence vertex degree. The quadratic index (Qindex)15 is calculated as: where ZM1 is the first Zagreb index. Bertz branching index (BBI)16 is defined as: Dragon branching index (DBI)17 is a modified version of the Bertz branching index. Narumi simple topological index (SNar)18 is related to molecular branching. alvaDesc applies a logarithmic transformation in order to obtain values of SNar more suitable for QSAR: Narumi harmonic topological index (HNar) is defined as: Narumi geometric topological index (GNar) is defined as: The total structure connectivity index (Xt)19 is defined as: where SNar is the Narumi simple topological index. The Pogliani index (Dz)20 is the sum over all non-hydrogen atoms of a modified vertex degree calculated as the ratio of the number of valence electrons over the principal quantum number of an atom. where Ziv is the number of valence electrons and Li is the principal quantum number of the i-th atom. The ramification index (Ram)21,22 is calculated as: where 3F is the number of atoms having vertex degree equal to three and 4F is the number of atoms having vertex degree equal to four. Kier benzene-likeliness index (BLI)23 is calculated as: where X1V is the valence connectivity index of order 1. Distance-based indices Distance-based indices is a set of descriptors calculated on an H-depleted molecular graph based on the topological distance. The polarity number (Pol)24 is calculated as the number of pairs of non-hydrogen atoms which are separated by three bonds. The Log of Product of Row Sums (LPRS) is based on the PRS25 and is calculated as: where nSK is the number of non-hydrogen atoms, u03c3i is the distance degree of the i-th atom. The logarithmic transformation is applied due to the large values that PRS can assume on big molecules. Mean square distance index (MSD)26 is calculated as: where nSK is the number of non-hydrogen atoms and dij is the element i,j of the topological distance matrix. alvaDesc calculates the Superpendentic index (SPI)27 as: where DP is the pendent matrix of the molecular graph. The pendent matrix is a submatrix of the distance matrix obtained by retaining the columns corresponding to pendent vertices. SPI was originally proposed as the square root of the sum of products of nonzero row elements in the pendent matrix. Logarithmic transformation has been introduced to avoid calculation overflows. A pendent vertex is a vertex with vertex degree equal to one, in a molecular graph a pendent vertex is equivalent to a terminal atom The 2D Petitjean shape index (PJI2)28 is calculated as: where the radius of a molecule is the lower bound of the eccentricity and the diameter is the upper bound of the eccentricity. The eccentricity of an atom i is the maximum topological distance from i to any other atom in the molecular graph The eccentricity (ECC)29 is calculated on the H-depleted molecular graph as: where Ei is the eccentricity of the i-th atom. The average eccentricity (AECC)29 is calculated as: where ECC is the eccentricity and nSK is the number of non-hydrogen atoms. Unipolarity (UNIP)29 is defined as: where u03c3i is the vertex distance degree of the i-th atom. The vertex distance degree of an atom is the sum of topological distances from the considered atom to any other atom in the H-depleted molecular graph Centralization descriptor (CENT)29 is calculated as: where Wi_D is the Wiener index, nSK is the number of non-hydrogen atoms and UNIP is the unipolarity descriptor. The Wiener index (Wi_D) is calculated as the half-sum of all topological distances collected in the distance matrix Variation descriptor (VAR)30 is calculated as: where u03c3 is the vertex distance degree. Radial centric information index (ICR)31 is calculated as: where nSK is the number of non-hydrogen atoms and ng is the number atoms having the same eccentricity. Radial centric information index (ICR) is calculated as the mean information content on atom eccentricities alvaDesc calculates also the maximum and average topological distance (MaxTD, MeanTD) and the maximum and average detour distance (MaxDD, MeanDD). MTI indices MTI indices are calculated on a H-depleted molecular graph based both on the adjacency matrix and the topological distance. Schultz Molecular Topological Index (SMTI)32 is calculated as: where A is the adjacency matrix, D is the topological distance matrix, and v is the column vector including the vertex degree of the atoms in the H-depleted molecular graph. Schultz Molecular Topological Index by valence vertex degree (SMTIV) is calculated as SMTI, but using the valence vertex degree in place of the simple vertex degree. Gutman Molecular Topological Index (GMTI)33 is calculated as: where u03b4 is the vertex degree in the H-depleted molecular graph, nSK the number of non-hydrogen atoms and dij is the element i,j of the topological distance matrix. Gutman Molecular Topological Index by valence vertex degrees (GMTIV) is calculated as GMTI, but using the valence vertex degree in place of the simple vertex degree. The Xu index (Xu)34 is calculated as follows: where u03b4 is the vertex degree in the H-depleted molecular graph, u03c3 is the vertex distance degree and nSK the number of non-hydrogen atoms. In the paper proposing the Xu index, the author highlighted that the Xu index has good discrimination of alkane isomers and is correlated with a set of physicochemical properties using a wide range of alkanes. The author shows that the Xu index is highly correlated with both the normal boiling points (BP) of alkanes and other selected physicochemical properties The eccentric connectivity index (CSI)35 is calculated as: where u03b4 is the vertex degree in the H-depleted molecular graph and E is the eccentricity. Path/walk indices Path/walk indices group includes a set of descriptors calculated on a H-depleted molecular graph based both on the walks and paths of the molecular graph. The all-path Wiener index (Wap)36 is a graph invariant defined as: where u03c0i,j denotes a path between vertices i and j and | u03c0i,j| is the length of this path. Wap is a graph invariant having an high discriminatory power, but since Wap values rapidly grows it is not appropriate for use in quantitative structure-activity/property studies Kier alpha-modified shape indices (S1K, S2K, S3K)37,38 are calculated as: where nSK is the number of non-H atoms, nBO is the number of non-H bonds, pc2 is the total path count of order 2, pc3 is the total path count of order 3 and u03b1 is defined as: where Ri is the covalent radius of atom i and RCsp3 is the covalent radius of sp3 carbon atom. The Kier flexibility index (PHI)39 is derived from Kier alpha-modified shape indices S1K and S2K as follows: where nSK is the number of non-hydrogen atoms, S1K and S2K are the Kier alpha-modified shape indices of order 1 and 2, respectively. Path/walk Randic shape indices (PWk)40 are calculated as: where k is the path/walk order, nSK is the number of non-H atoms, pck is the number of paths of order k starting from atom i and wck is the number of walks of order k starting from atom i. alvaDesc calculates path/walk Randic shape indices from order 2 up to 5. Path/walk Randic shape index of order 1 (PW1) is not calculated since the number of paths and walks of length one is the same and the corresponding molecular index equals one for all molecules E-state indices The E-state topological indices are based on the electrotopological state41. The maximal electrotopological negative variation (MAXDN)42 is calculated as: where u0394Ii- are the negative values of u0394Ii. The maximal electrotopological positive variation (MAXDP)42 is calculated as: where u0394Ii+ are the positive values of u0394Ii. The molecular electrotopological variation (DELS)42 is calculated as: According to the authors, DELS index could be considered as a measure of total charge transfer in the molecule The E-state topological parameter (TIE) is a modified version of the descriptor proposed by Voelkel43; it is calculated as: where nBO is the number of non-H bonds, nCIC the cyclomatic number, Sb,1 and Sb,2 the electrotopological state indices for the two atoms incident to the b-th bond. Intrinsic state pseudoconnectvity indices44,45 include molecular descriptors based on the intrinsic states (Ii). ID Symbol Formula 1 Psi_i_S 2 Psi_i_A 3 Psi_i_0 4 Psi_i_1 5 Psi_i_t 6 Psi_i_0d 7 Psi_i_1d 8 Psi_i_1s 9 Psi_e_A 10 Psi_e_0 11 Psi_e_1 12 Psi_e_t 13 Psi_e_0d 14 Psi_e_1d 15 Psi_e_1s Table 1: Intrinsic state pseudoconnectvity indices Centric indices alvaDesc calculates two centric indices. The Balaban centric index (BAC)15 is calculated on a H-depleted molecular graph removing iteratively the vertices of degree one. It is defined as: where u03b4i is the number of terminal vertices removed at i-th step. The lopping centric index (LOC)15 is calculated as the mean information content derived from the pruning partition of a graph: where nSK is the number of non-H atoms and u03b4i is the number of terminal vertices removed at i-th step. Devillers, J. &amp; Balaban, A.T. Eds. (2000). Topological Indices and Related Descriptors in QSAR and Drug Design. Gordon&amp;Breach, Amsterdam&#160;&#x21a9;&#xfe0e; Mauri, A., Consonni, V., &amp; Todeschini, R. (2017). Molecular descriptors. In Handbook of Computational Chemistry (pp. 2065 u20132093)&#160;&#x21a9;&#xfe0e; Kier, L. B., &amp; Hall, L. H. (1981). Derivation and significance of valence molecular connectivity. Journal of Pharmaceutical Sciences, 70(6), 583 u2013589&#160;&#x21a9;&#xfe0e; Gutman, I., &amp; Trinajsti u0107, N. (1972). Graph theory and molecular orbitals. Total u03c6-electron energy of alternant hydrocarbons. Chemical Physics Letters, 17(4), 535 u2013538&#160;&#x21a9;&#xfe0e; Gutman, I., &amp; Das, K. C. (2004). The first Zagreb Index 30 years after. MATCH - Communications in Mathematical and in Computer Chemistry, 50, 83 u201392&#160;&#x21a9;&#xfe0e; Gutman, I., Ru u0161 u010di u0107, B., Trinajsti u0107, N., &amp; Wilcox, C. F. (1975). Graph theory and molecular orbitals. XII. Acyclic polyenes. The Journal of Chemical Physics, 62(9), 3399 u20133405&#160;&#x21a9;&#xfe0e; Nikoli u0107, S., Kova u010devi u0107, G., Mili u010devi u0107, A., &amp; Trinajsti u0107, N. (2003). The Zagreb indices 30 years after. Croatica Chemica Acta, 76(2), 113 u2013124&#160;&#x21a9;&#xfe0e; Kupchik, E. J. (1986). Structure u2010Molar Refraction Relationships of Alkylsilanes Using Empirically u2010Modified First Order Molecular Connectivity Indices. Quantitative Structure u2010Activity Relationships, 5(3), 95 u201398&#160;&#x21a9;&#xfe0e; Kupchik, E. J. (1988). Structure u2014 Molar Refraction Relationships of Alkylgermanes Using Molecular Connectivity. Quantitative Structure u2010Activity Relationships, 7(2), 57 u201359&#160;&#x21a9;&#xfe0e; Bajaj, S., Sambi, S. S., &amp; Madan, A. K. (2005). Prediction of anti-inflammatory activity of N-arylanthranilic acids: Computational approach using refined Zagreb indices. Croatica Chemica Acta, 78(2), 165 u2013174&#160;&#x21a9;&#xfe0e; Goel, A., &amp; Madan, A. K. (1995). Structure-Activity Study on Antiinflammatory Pyrazole Carboxylic Acid Hydrazide Analogs Using Molecular Connectivity Indices. Journal of Chemical Information and Computer Sciences, 35(3), 510 u2013514. https://doi.org/10.1021/ci00025a019&#160;&#x21a9;&#xfe0e; V.K.Gombar et al. (1987). Indian J. Chem. 26B, 1168-1170&#160;&#x21a9;&#xfe0e; Todeschini, R., &amp; Consonni, V. (2010). New Local Vertex Invariants and Molecular Descriptors Based on Functions of the Vertex Degrees. MATCH - Communications in Mathematical and in Computer Chemistry, 64, 359 u2013372&#160;&#x21a9;&#xfe0e; Bonchev, D. (2001). Overall connectivity - A next generation molecular connectivity. Journal of Molecular Graphics and Modelling, 20(1), 65 u201375. https://doi.org/10.1016/S1093-3263(01)00101-2&#160;&#x21a9;&#xfe0e; Balaban, A. T. (1979). Chemical graphs - XXXIV. Five new topological indices for the branching of tree-like graphs [1]. Theoretica Chimica Acta, 53(4), 355 u2013375. https://doi.org/10.1007/BF00555695&#160;&#x21a9;&#xfe0e; Bertz, S. H. (1988). Branching in graphs and molecules. Discrete Applied Mathematics, 19(1 u20133), 65 u201383. https://doi.org/10.1016/0166-218X(88)90006-6&#160;&#x21a9;&#xfe0e; Mauri, A., Consonni, V., Pavan, M., &amp; Todeschini, R. (2006). Dragon software: An easy approach to molecular descriptor calculations. Match Communications In Mathematical And In Computer Chemistry, 56(2), 237 u2013248&#160;&#x21a9;&#xfe0e; Narumi, H. (1987). New topological indices for finite and infinite systems. MATCH Communications in Mathematical and in Computer Chemistry, 22, 195 u2013207&#160;&#x21a9;&#xfe0e; Needham, D. E., Wei, I. C., &amp; Seybold, P. G. (1988). Molecular Modeling of the Physical Properties of the Alkanes. Journal of the American Chemical Society, 110(13), 4186 u20134194. https://doi.org/10.1021/ja00221a015&#160;&#x21a9;&#xfe0e; Pogliani, L. (1996). Modeling with special descriptors derived from a medium-sized set of connectivity indices. Journal of Physical Chemistry, 100(46), 18065 u201318077. https://doi.org/10.1021/jp961434c&#160;&#x21a9;&#xfe0e; Araujo, O., &amp; De La Pe u00f1a, J. A. (1998). The connectivity index of a weighted graph. Linear Algebra and Its Applications, 283(1 u20133), 171 u2013177. https://doi.org/10.1016/S0024-3795(98)10096-4&#160;&#x21a9;&#xfe0e; Araujo, O., &amp; De La Pe u00f1a, J. A. (1998). Some bounds for the connectivity index of a chemical graph. Journal of Chemical Information and Computer Sciences, 38(5), 827 u2013831. https://doi.org/10.1021/ci980012s&#160;&#x21a9;&#xfe0e; L.B. Kier, L.H. Hall, Molecular Connectivity in Structure-Activity Analysis, Research Studies Press - Wiley, Chichester (UK), 1986&#160;&#x21a9;&#xfe0e; Wiener, H. (1947). Structural Determination of Paraffin Boiling Points. Journal of American Chemical Society, 69(1), 17 u201320&#160;&#x21a9;&#xfe0e; Schultz, H. P., Schultz, E. B., &amp; Schultz, T. P. (1992). Topological organic chemistry. 4. Graph theory, matrix permanents, and topological indices of alkanes. Journal of Chemical Information and Modeling, 32(1), 69 u201372. https://doi.org/10.1021/ci00005a011&#160;&#x21a9;&#xfe0e; Balaban, A. T. (1983). Topological indices based on topological distances in molecular graphs. Pure and Applied Chemistry, 55(2), 199 u2013206. https://doi.org/10.1351/pac198855020199&#160;&#x21a9;&#xfe0e; Gupta, S., Singh, M., &amp; Madan, A. K. (1999). Superpendentic index: A novel topological descriptor for predicting biological activity. Journal of Chemical Information and Computer Sciences, 39(2), 272 u2013277. https://doi.org/10.1021/ci980073q&#160;&#x21a9;&#xfe0e; Petitjean, M. (1992). Applications of the Radius-Diameter Diagram to the Classification of Topological and Geometrical Shapes of Chemical Compounds. Journal of Chemical Information and Computer Sciences, 32(4), 331 u2013337. https://doi.org/10.1021/ci00008a012&#160;&#x21a9;&#xfe0e; Skorobogatov, V. A., &amp; Dobrynin, A. . (1988). Metric analysis of graphs. MATCH - Communications in Mathematical and in Computer Chemistry, 23, 105 u2013151&#160;&#x21a9;&#xfe0e; Entringer, R. C., Jackson, D. E., &amp; Snyder, D. A. (1976). Distance in graphs. Czechoslovak Mathematical Journal, 26(2), 283 u2013296. https://doi.org/10.21136/CMJ.1976.101401&#160;&#x21a9;&#xfe0e; Bonchev, D. (1983). Information theoretic indices for characterization of chemical structures (Vol. 5). Chichester (UK): Research Studies Press&#160;&#x21a9;&#xfe0e; Schultz, H. P. (1989). Topological organic chemistry. 1. Graph theory and topological indices of alkanes. Journal of Chemical Information and Computer Sciences, 29(3), 227 u2013228. https://doi.org/10.1021/ci00063a012&#160;&#x21a9;&#xfe0e; Gutman, I. (1994). Selected Properties of the Schultz Molecular Topological Index. Journal of Chemical Information and Computer Sciences, 34(5), 1087 u20131089. https://doi.org/10.1021/ci00021a009&#160;&#x21a9;&#xfe0e; Ren, B. (1999). A new topological index for QSPR of alkanes. Journal of Chemical Information and Computer Sciences, 39(1), 139 u2013143. https://doi.org/10.1021/ci980098p&#160;&#x21a9;&#xfe0e; Sharma, V., Goswami, R., &amp; Madan, A. K. (1997). Eccentric connectivity index: A novel highly discriminating topological descriptor for structure-property and structure-activity studies. Journal of Chemical Information and Computer Sciences, 37(2), 273 u2013282. https://doi.org/10.1021/ci960049h&#160;&#x21a9;&#xfe0e; Lukovits, I. (1998). An all-path version of the Wiener index. Journal of Chemical Information and Computer Sciences, 38(2), 125 u2013129. https://doi.org/10.1021/ci9700541&#160;&#x21a9;&#xfe0e; Kier, L. B. (1986). Shape Indexes of Orders One and Three from Molecular Graphs. Quantitative Structure u2010Activity Relationships, 5(1), 1 u20137. https://doi.org/10.1002/qsar.19860050102&#160;&#x21a9;&#xfe0e; Kier, L. B. (1986). Distinguishing Atom Differences in a Molecular Graph Shape Index. Quantitative Structure u2010Activity Relationships, 5(1), 7 u201312. https://doi.org/10.1002/qsar.19860050103&#160;&#x21a9;&#xfe0e; Kier, L. B. (1989). An Index of Molecular Flexibility from Kappa Shape Attributes. Quantitative Structure-Activity Relationships, 8(3), 221 u2013224. https://doi.org/10.1002/qsar.19890080307&#160;&#x21a9;&#xfe0e; Randi u0107, M. (2001). Novel Shape Descriptors for Molecular Graphs. Journal of Chemical Information and Computer Sciences, 41(3), 607 u2013613. https://doi.org/10.1021/ci0001031&#160;&#x21a9;&#xfe0e; Hall, L. H., Mohney, B., &amp; Kier, L. B. (1991). The Electrotopological State: An Atom Index for QSAR. Quantitative Structure u2010Activity Relationships, 10(1), 43 u201351&#160;&#x21a9;&#xfe0e; Gramatica, P., Corradi, M., &amp; Consonni, V. (2000). Modelling and prediction of soil sorption coefficients of non-ionic organic pesticides by molecular descriptors. Chemosphere, 41(5), 763 u2013777. https://doi.org/10.1016/S0045-6535(99)00463-4&#160;&#x21a9;&#xfe0e; Voelkel, A. (1994). Structural descriptors in organic chemistry-new topological parameter based on electrotopological state of graph vertices. Computers and Chemistry, 18(1), 1 u20134. https://doi.org/10.1016/0097-8485(94)80016-2&#160;&#x21a9;&#xfe0e; Pogliani, L. (2000). Modeling with Molecular Pseudoconnectivity Descriptors. A Useful Extension of the Intrinsic I-State Concept. The Journal of Physical Chemistry A, 104(39), 9029 u20139045. https://doi.org/10.1021/jp001191v&#160;&#x21a9;&#xfe0e; Pogliani, L. (2004). Modeling with indices obtained from complete graphs. Croatica Chemica Acta, 77(1 u20132), 193 u2013201&#160;&#x21a9;&#xfe0e; " }, { "uri": "./gui/analyse_data.html", "title": "Analyse the calculated data", "tags": [], "description": "", "content": "The calculated descriptors can be analysed by clicking Analysis on the menu bar and then choosing either: Univariate statistics shows basic statistical data about the descriptors (mean, standard deviation, etc.) Principal Component Analysis can be used to perform the well-known PCA statistical procedure. At the end of the analysis the PCA Loadings and Scores plots will be shown t-distributed stochastic neighbor embedding (t-SNE) analysis1,2 can be used to perform dimensionality reduction of the molecular descriptors in a two-dimensional space. t-SNE models each molecular descriptor vector by a two-dimensional point in such a way that similar objects are modeled by nearby points and dissimilar objects are modeled by distant points. During the t-SNE calculation, a progress bar is displayed indicating the current number of iterations and cost (result of the Kullback-Leibler divergence). If the user stops the calculation, the graph of the last iteration will be shown. Correlation analysis shows the relationship between n descriptors. These are shown on a correlation map (of n by n dimension) where red means that 2 descriptors are highly correlated, blue means that 2 descriptors have high negative correlation (i.e., higher values of one descriptor tend to be associated with lower values of the other) and yellow means that they are not correlated at all. alvaDesc performs automatic variable reduction before Principal Component Analysis, Correlation analysis and t-SNE analysis. Only descriptors that are not constant or near-constant will be available for the analysis. Additionally it is possible to define a Standard Deviation Threshold to filter the descriptors before the analysis. This threshold can be defined in the alvaDesc options. If this threshold has a value larger than zero all descriptors with a standard deviation equal or lower than the threshold will be automatically removed from the analysis. The default value of this threshold is 0.0001. Hinton, G., &amp; Roweis, S. (2003). Stochastic neighbor embedding. Advances in Neural Information Processing Systems&#160;&#x21a9;&#xfe0e; Laurens van der Maaten, &amp; Geoffrey E., H. (2008). Visualizing Data using t-SNE. Journal of Machine Learning Research, 164(2210), 10&#160;&#x21a9;&#xfe0e; " }, { "uri": "./molecules/complex_molecules.html", "title": "Complex molecules", "tags": [], "description": "", "content": "Molecules with many rings, especially if they are fused rings, can significantly affect the speed of alvaDesc while opening them or while calculating some descriptors on them. You can change the following parameters to adjust alvaDesc&rsquo;s behavior while dealing with such molecules: Name Description MaxCircuits Maximum number of circuits MaxSRDetour Maximum number of smallest rings1 for detour distance calculation MaxAtomWalkPath Maximum number of atoms for path and walk descriptors calculation These parameters can be modified in the Script file, when using the CLI, or in the Options window of the GUI. On each given molecule the following rules apply: Rule Affected descriptors if nCIR &gt; MaxCircuits nCIR is set to na, nR04..nR12 and D/Dtr04..D/Dtr12 are calculated only on the identified subset of circuits if nCIC &gt; MaxSRDetour all descriptors depending on the detour distance matrix, such as D/Dtr03..D/Dtr12, MeanDD, MaxDD and Detour matrix (Dt) and Distance / detour matrix (D/Dt) descriptors in 2D matrix-based descriptors block, are not calculated. Path descriptors depending on all paths (TPC, piID, PCR, PCD, CID and BID) are calculated on a maximum path length equal to 10 if nSK &gt; MaxAtomWalkPath the Path/walk indices (Wap, S1K, S2K, S3K, PHI, PW2, PW3, PW4 and PW5) included in topological indices and walk and path descriptors are not calculated When experiencing a slowdown on such molecules you can reduce the value of these parameters and try again. The minimal cycle basis of a molecular graph is referred to as the Smallest Set of Smallest Rings (SSSR), the number of Smallest Rings is equal to the cyclomatic number (nCIC).&#160;&#x21a9;&#xfe0e; " }, { "uri": "./introduction/faq.html", "title": "F.A.Q.", "tags": [], "description": "", "content": "1. How can I cite alvaDesc? To see the correct citation for alvaDesc you can either: Run alvaDescGUI and click on the About alvaDesc menu item Run alvaDescCLI --cite 2. Why is a molecule rejected? The possible causes for molecule rejection are: The input file is corrupted If the molecule is entered as SMILES notation, unrecognised symbols have been used or some other SMILES rules have not been followed 3. Why is a descriptor value set to &ldquo;na&rdquo;? alvaDesc molecular descriptors follow the algorithms  implementation proposed in literature. Typically molecular descriptors have been defined to be calculated on small organic molecules, despite this alvaDesc algorithms have been designed to be applied to a broader set of molecules. However, some descriptors have intrinsic limitations and cannot be calculated on some molecules. A typical example are those descriptors that cannot be calculated on disconnected structures (e.g., MLOGP: Moriguchi octanol-water partition coefficient (logP)). 4. Why does alvaDesc take a long time (or freeze) either while opening molecules or calculating the descriptors? This is often related to dealing with molecules with many rings. Please check Complex molecules chapter. " }, { "uri": "./molecules.html", "title": "Molecules", "tags": [], "description": "", "content": "In this section you find information about molecules: Molecule representation Disconnected structures Partial charges Complex molecules " }, { "uri": "./fingerprints/pathfp.html", "title": "Path Fingerprints", "tags": [], "description": "", "content": "Path Fingerprints are hashed fingerprints obtained systematically enumerating all not branched path fragments obtained starting from each non-hydrogen atom of the molecule up to the given maximum length (i.e. the maximum length). Figure 1: Identification of all fragments of 4,5-Dimethyloxazole (length 0 to 2). Duplicated fragments are highlighted Figure 2: Hashing of identified fragments. Bit collision is highlighted in red Path fingerprints are typically used to perform full structure and substructure identification, similarity searching and for similarity analysis of compound libraries. In order to improve the performances of any application the fingerprint configuration is very important. " }, { "uri": "./descriptors/walk_and_path_counts.html", "title": "Walk and path counts", "tags": [], "description": "", "content": "Walk and path counts are topological indices based on the identification of paths, walks and self-returning walks in a H-depleted molecular graph. alvaDesc calculates 46 descriptors belonging to the walk and path logical block, divided in five groups: Walk counts Self-returning walk counts Path counts Multiple path counts ID numbers Walk counts A walk is any sequence of adjacent graph edges connecting atom i to atom j. In a walk it is allowed to go back and forth or to visit vertices repeatedly1. The length (order) of a walk is the total number of edges that are traversed2. The walk count of order k (awck) is the total number of walks of length k in the graph. alvaDesc calculates molecular walk counts from order 1 up to order 10. Since molecular walk counts may be very large for molecules with a large number of atoms, alvaDesc provides MWCk values as: where k is the walk length and awck is the number of walk of length k. Molecular walk counts are related to branching and complexity of molecular graphs. Self-returning walk counts A self-returning walk is a walk starting and finishing at the same vertex. The self-returning walk count of order k (srwk) is the total number of self-returning walks of length k in the molecular graph. alvaDesc calculates self-returning walk counts from order 2 to order 10. Self-returning walk counts, like molecular walk counts, may be very large for molecules with a large number of atoms. Therefore, alvaDesc provides SRWk values as: where k is the walk length and srwk is the number of self-returning walk of length k. Path counts A path is a walk without any repeated vertex3,4. The path count of order k (pck) is the total number of paths of length k in the molecular graph. alvaDesc calculates molecular path counts from order 1 to order 10. Since molecular path counts may be very large for molecules with a large number of atoms, alvaDesc provides MPCk values as: where k is the path length and pck is the number of paths of length k. Multiple path counts A multiple path of order k (mpck) is defined as the sum of the products of the bond order of the paths of length k. Since molecular multiple path counts may be very large for molecules with a large number of atoms, alvaDesc provides piPCk values as: where k is the path length (from 1 to 10) and mpck is the multiple path of order k. Molecular multiple path counts have the same value of the molecular path counts for molecules having only single bonds. Here&rsquo;s an example using the 2-Methyl-1-butene: 2-Methyl-1-butene k atom 1 atom 2 atom 3 atom 4 atom 5 mpck piPCk 1 1 2 4 2 1 5 1.7918 2 1 3 1 4 3 6 1.9459 3 3 0 0 2 1 3 1.3863 4 0 0 0 0 0 0 0.0000 mpck is equal to the sum of each atom contribution in the row but divided by 2 to consider each path only once ID numbers The total walk count (TWC) is based on the total number of walks. It can be considered as a measure of molecular complexity that increases both with the increase of the size and the branching1. It is calculated as follows: The total path count (TPC) is the total number of paths (from order 0 to the maximum path length of the molecule): The conventional bond-order ID number (piID) is the total number of multiple paths (from order 0 to the maximum path length of the molecule): where tmpc is the total number of multiple path counts. Since TWC, TPC and piID may be very large for molecules with a large number of atoms or rings, alvaDesc provides their values as the natural logarithm of the original value. The ratio of multiple path count over path count (PCR) is defined as: where piID is the conventional bond-order ID number and TPC is the total path count. The difference between multiple path count and path count (PCD) is defined as: where piID is the conventional bond-order ID number and TPC is the total path count. alvaDesc calculates two additional identification numbers, the Randic ID number (CID)5 and the Balaban ID number (BID)6. The calculation of descriptors depending on all paths can be time demanding for molecules including a big number of atoms or fused rings. In order to avoid an excessive increase of the calculation time a constraint can be defined. See Complex molecules. R u00fccker, G., &amp; R u00fccker, C. (1993). Counts of All Walks as Atomic and Molecular Descriptors. Journal of Chemical Information and Computer Sciences, 33(5), 683 u2013695. https://doi.org/10.1021/ci00015a005&#160;&#x21a9;&#xfe0e; Ivanciuc, O., &amp; Balaban, A. T. (1992). Nonisomorphic graphs with identical atomic counts of self-returning walks: Isocodal graphs. Journal of Mathematical Chemistry, 11(1), 155 u2013167. https://doi.org/10.1007/BF01164200&#160;&#x21a9;&#xfe0e; Randi u0107, M., Brissey, G. M., Spencer, R. B., &amp; Wilkins, C. L. (1979). Search for all self-avoiding paths for molecular graphs. Computers and Chemistry, 3(1), 5 u201313. https://doi.org/10.1016/0097-8485(79)80002-9&#160;&#x21a9;&#xfe0e; Randi u0107, M. (1979). Characterization of Atoms, Molecules, and Classes of Molecules Based on Paths Enumeration. In MATCH - Communications in Mathematical and in Computer Chemistry (Vol. 7, pp. 5 u201364)&#160;&#x21a9;&#xfe0e; Randi u0107, M. (1984). On Molecular Identification Numbers. Journal of Chemical Information and Computer Sciences, 24(3), 164 u2013175. https://doi.org/10.1021/ci00043a009&#160;&#x21a9;&#xfe0e; A.T. Balaban, Numerical Modelling of Chemical Structures: Local Graph Invariants and Topological Indices in Graph Theory and Topology in Chemistry, R.B. King, D.H. Rouvray (Eds.), Elsevier, Amsterdam (The Netherlands), pp. 159-176, 1987&#160;&#x21a9;&#xfe0e; " }, { "uri": "./gui/check_charts.html", "title": "Check the charts", "tags": [], "description": "", "content": "If you click Charts on the menu bar you can choose either: Histogram Bar plot Scatter plot PCA plot which shows both PCA Loadings and Scores t-SNE plot which shows the 2-dimensions t-SNE plot The selected plot will be shown in the main window, on the right-hand side you can select the descriptors (or Principal Components in case of PCA) to be used in the plot. The chart items can either be selected individually, by clicking on each one, or as group by clicking whilst pressing SHIFT and dragging a rectangle or a lasso around the items you want to select. Lasso selection can be enabled in the chart context menu This action will filter the molecules shown in the main window. To reset the selection right click on the plot and then click Reset selection. In order to zoom you can point the mouse cursor over the interested area and use the mouse wheel. Charts can be copied as a JPEG image or saved as a JPEG, PNG or BMP image by right clicking on the plot. " }, { "uri": "./descriptors/connectivity_indices.html", "title": "Connectivity indices", "tags": [], "description": "", "content": "Connectivity indices are topological indices calculated on the H-depleted molecular graph where each vertex is weighted by the vertex degree1 or by similar vertex invariants. Kier-Hall molecular connectivity indices Connectivity indices (Xk)1 are defined as: where k runs over all of the k-th order paths including n vertices and K is the total number of k-th order paths present in the molecular graph. The product is over the vertex degrees u03b4i of all the vertices involved in each subgraph. alvaDesc calculates connectivity indices for order 0 up to 5. The connectivity index X1 is usually known as the Randic connectivity index2 and it is defined as: Average connectivity indices (XkA) are calculated by dividing each connectivity index by the total number of paths of the corresponding order. Valence connectivity indices (Xkv) are calculated replacing the simple vertex degree with the valence vertex degree3. The valence vertex degree is calculated as: where Ziv is the number of valence electrons ( u03c3 electrons, u03c0 electrons and lone pair electrons n) of the i-th atom and hi is the number of hydrogen atoms bonded to it. For atoms having a principal quantum number greater than two, valence vertex degree is defined as: where Zi is the atomic number of atom i. Solvation connectivity indices Solvation connectivity indices (Xksol)4 are calculated as: where k runs over all of the k-th order paths including n vertices, K is the total number of k-th order paths present in the molecular graph and Li is the principal quantum number of the i-th atom. Solvation connectivity indices are calculated on a H-depleted molecular graph, where fluorine atoms are not taken into account Randic-like connectivity indices Randic-like connectivity indices are all inspired by the formula of the Randic connectivity index (X1). The modified Randic index (XMOD)5 is calculated as: where aij is the element i,j of the adjacency matrix, nSK is the number of non-H atoms, Zi is the atomic number of the i-th atom, u03b4i and u03b4j are the vertex degree of the i-th and j-th atom, respectively. The reciprocal distance sum Randic-like index (RDCHI)6 is defined as: and the reciprocal distance sum inverse Randic-like index (RDSQ)6 is defined as: where aij is the element i,j of the adjacency matrix, RDSi and RDSj are the reciprocal distance sum of the i-th and j-th atom, respectively. The reciprocal distance sum of atom i (RDSi) is defined as the sum of the elements of the reciprocal distance matrix (RD) over row i or column i. The reciprocal distance matrix is a square symmetric matrix whose entries RDi,j are equal to the reciprocal of the distance between vertices i and j for non-diagonal elements, and are equal to zero for the diagonal elements. RDSQ increases with both molecular size and branching degree while RDCHI increases with molecular size but decreases with molecular branching Kupchik connectivity index (X1Kup) is calculated as: where aij is the element i,j of the adjacency matrix, u03b4ihet and u03b4jhet are the vertex degree defined by Kupchik of the i-th and j-th atom, respectively. Connectivity topochemical index (X1Mad) is calculated as the Kupchik connectivity index using the Madan vertex degree. Analogously, perturbation connectivity index (X1Per) is calculated using the perturbation delta values and the multiplicative perturbation connectivity index (X1MulPer) is calculated using the multiplicative perturbation delta value in place of the vertex degrees. Hall, L. H., &amp; Kier, L. B. (1978). Molecular Connectivity and Substructure Analysis. Journal of Pharmaceutical Sciences, 67(12), 1743 u20131747. https://doi.org/10.1002/jps.2600671229&#160;&#x21a9;&#xfe0e; Randi u0107, M. (1975). On Characterization of Molecular Branching. Journal of the American Chemical Society, 97(23), 6609 u20136615. https://doi.org/10.1021/ja00856a001&#160;&#x21a9;&#xfe0e; Kier, L. B., &amp; Hall, L. H. (1981). Derivation and Significance of Valence Molecular Connectivity. Journal of Pharmaceutical Sciences, 70(6), 583 u2013589. https://doi.org/10.1002/jps.2600700602&#160;&#x21a9;&#xfe0e; Zefirov, N. S., &amp; Palyulin, V. A. (2001). QSAR for Boiling Points of u201cSmall u201d Sulfides. Are the u201cHigh-Quality Structure-Property-Activity Regressions u201d the Real High Quality QSAR Models? Journal of Chemical Information and Computer Sciences, 41(4), 1022 u20131027. https://doi.org/10.1021/ci0001637&#160;&#x21a9;&#xfe0e; Lohninger, H. (1993). Evaluation of Neural Networks Based on Radial Basis Functions and Their Application to the Prediction of Boiling Points from Structural Parameters. Journal of Chemical Information and Computer Sciences, 33(5), 736 u2013744. https://doi.org/10.1021/ci00015a012&#160;&#x21a9;&#xfe0e; Ivanciuc, O., Balaban, T.-S., &amp; Balaban, A. T. (1993). Design of topological indices. Part 4. Reciprocal distance matrix, related local vertex invariants and topological indices. Journal of Mathematical Chemistry, 12(1), 309 u2013318. https://doi.org/10.1007/BF01164642&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors.html", "title": "Descriptors", "tags": [], "description": "", "content": " The molecular descriptor is the final result of a logic and mathematical procedure which transforms chemical information encoded within a symbolic representation of a molecule into a useful number or the result of some standardized experiment. 1 Research on molecular descriptors and their applications is rapidly growing during the last decades, this is proven by the exponential growth of scientific papers using molecular descriptors in the fields of Quantitative Structure Activity Relationships (QSAR)2. Two major drives foster this development: the exponential growth of available chemicals together with freely accessible databases collecting experimental data3,4; the introduction of regulation on the registration, evaluation, authorisation, and restriction of chemicals. As an example REACH in Europe5 promotes the use of QSAR methodologies to predict not available experimental data. In this scenario, molecular descriptors play a crucial role. Although until the 1960s u20131970s, the study of relationships between experimentally measured quantities and molecular structure was based on experimental descriptors; nowadays, this analysis is more focused on the recognition of the relationships among theoretical descriptors and the considered experimental property. Thousands of molecular descriptors have been proposed in literature; this impressive amount of work is just one of the evidences of the scientific community interest in this field. The huge amount of available molecular descriptors requires the delineation of few basic rules that molecular descriptors should comply with6,7. A molecular descriptor must: Be invariant to atom labelling and numbering Be invariant to the molecule roto-translation Be defined by an unambiguous algorithm Have a well-defined applicability on molecular structures alvaDesc molecular descriptors implementation comply with these rules. alvaDesc calculates 5832 descriptors, 5666 of them are divided in 33 logical blocks, the remaining 166 descriptors constitutes the MACCS 166 fingerprint. Block name # of descriptors Constitutional indices 50 Ring descriptors 35 Topological indices 79 Walk and path counts 46 Connectivity indices 37 Information indices 51 2D matrix-based descriptors 608 2D autocorrelations 213 Burden eigenvalues 96 P_VSA-like descriptors 69 ETA indices 40 Edge adjacency indices 324 Geometrical descriptors 38 3D matrix-based descriptors 132 3D autocorrelations 80 RDF descriptors 210 3D-MoRSE descriptors 224 WHIM descriptors 114 GETAWAY descriptors 273 Randic molecular profiles 41 Functional group counts 154 Atom-centred fragments 115 Atom-type E-state indices 346 Pharmacophore descriptors 165 2D Atom Pairs 1596 3D Atom Pairs 36 Charge descriptors 15 Molecular properties 27 Drug-like indices 30 CATS 3D descriptors 300 WHALES descriptors 33 MDE descriptors 19 Chirality descriptors 70 Todeschini, R., &amp; Consonni, V. (2000). Handbook of Molecular Descriptors. Wiley. https://doi.org/10.1002/9783527613106&#160;&#x21a9;&#xfe0e; Mauri, A., Consonni, V., &amp; Todeschini, R. (2016). Molecular Descriptors. In J. Leszczynski, A. Kaczmarek-Kedziera, T. Puzyn, M. G. Papadopoulos, H. Reis, &amp; M. K. Shukla (Eds.), Handbook of Computational Chemistry (pp. 1 u201329). Springer Netherlands. https://doi.org/10.1007/978-94-007-6169-8_51-1&#160;&#x21a9;&#xfe0e; Irwin, J. J., &amp; Shoichet, B. K. (2005). ZINC - A free database of commercially available compounds for virtual screening. Journal of Chemical Information and Modeling, 45(1), 177 u2013182. https://doi.org/https://doi.org/10.1021/ci049714+&#160;&#x21a9;&#xfe0e; Williams, A., &amp; Tkachenko, V. (2014). The Royal Society of Chemistry and the delivery of chemistry data repositories for the community. Journal of Computer-Aided Molecular Design, 28(10), 1023 u20131030. https://doi.org/10.1007/s10822-014-9784-5&#160;&#x21a9;&#xfe0e; Worth, A. P. (2010). The Role of Qsar Methodology in the Regulatory Assessment of Chemicals. In Recent Advances in QSAR Studies (pp. 367 u2013382). https://doi.org/10.1007/978-1-4020-9783-6_13&#160;&#x21a9;&#xfe0e; Randi u0107, M. (1996). Molecular bonding profiles. Journal of Mathematical Chemistry, 19(3), 375 u2013392. https://doi.org/10.1007/BF01166727&#160;&#x21a9;&#xfe0e; Guha, R., &amp; Willighagen, E. (2012). A Survey of Quantitative Descriptions of Molecular Structure. Current Topics in Medicinal Chemistry, 12(18), 1946 u20131956. https://doi.org/10.2174/156802612804910278&#160;&#x21a9;&#xfe0e; " }, { "uri": "./fingerprints.html", "title": "Fingerprints", "tags": [], "description": "", "content": "alvaDesc implements various fingerprint types, all of them are fixed-size boolean vectors that encode the 2D information of a molecular structure. Molecular fingerprints describe a molecule considering different local aspects of a molecular structure. Even if molecular fingerprints have been originally introduced for fast database searching, due to their capability to identify common substructures, they have been further used for building QSAR models, especially those based on local similarity as in the kNN approach1,2. Molecular fingerprints can be of two types, structural keys and hashed fingerprints. Structural keys are defined as a set of structural features (e.g., atom pairs, functional groups, pharmacophoric atom types) that have been prepared in order to be able to discriminate among molecules and eventually be able to highlight chemical properties of a molecule. alvaDesc allows the calculation of both structural keys and hashed fingerprints. alvaDesc includes different structural keys, most of them are included in the molecular descriptor blocks, like 2D atom pairs and CATS 2D descriptors belonging to the pharmacophore descriptors block. Additionally alvaDesc provides MACCS 166 structural keys and two hashed fingerprints, mainly: MACCS 166 Fingerprint Extended Connectivity Fingerprints (ECFP) Path FingerPrints (PFP) MACCS 166 fingerprint3 is a fixed size boolean vector reflecting the presence/absence of a set of 166 well defined molecular features, while Extended Connectivity Fingerprints (ECFP) and Path Fingerprints (PFP) are boolean vectors obtained exploding the molecular structure in all possible substructure patterns and processing them using a hashing function in order to reduce a variable-size boolean vector to a fixed-size one. Cassotti, M., Consonni, V., Mauri, A., &amp; Ballabio, D. (2014). Validation and extension of a similarity-based approach for prediction of acute aquatic toxicity towards Daphnia magna. SAR and QSAR in Environmental Research, 25, 1013 u20131036&#160;&#x21a9;&#xfe0e; Rojas et al. (2017) A QSTR-Based Expert System to Predict Sweetness of Molecules. Frontiers in Chemistry, 5, 53&#160;&#x21a9;&#xfe0e; Durant, J. L., Leland, B. A., Henry, D. R., &amp; Nourse, J. G. (2002). Reoptimization of MDL keys for use in drug discovery. Journal of Chemical Information and Computer Sciences, 42, 1273 u20131280&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/information_indices.html", "title": "Information indices", "tags": [], "description": "", "content": "Information indices are molecular descriptors defined in order to capture the information content of molecules. The information content is based on the concept of information entropy introduced by Shannon1. alvaDesc defines a set of equivalence classes, the equivalence between the atoms of the molecule is defined with different criteria according to the descriptor. The total information content is defined as: and the mean information content is defined as: where G is the number of different equivalence classes and ng the number of atoms in the g-th class. Equivalence classes are defined by the use of some relationships among the atoms of the molecule. The information index on molecular size (ISIZ)2 is calculated as the total information content on the atom number: where nAT is the number of molecule atoms. The total information index on atomic composition (IAC)3 and mean information index on atomic composition (AAC) are calculated on a H-filled molecular graph. IAC and AAC are calculated as the total and mean information content. Equivalence classes are based on chemical atom types. IAC is defined as: AAC is defined as: where nAT is the total number of atoms and ng is the number of atoms of equivalent class g. The total information content on the distance equality (IDET) and mean information content on the distance equality (IDE)4,5 are calculated on a H-depleted molecular graph. These information indices are based on the equality of topological distances. The total information content on the distance magnitude (IDMT) and mean information content on the distance magnitude (IDM)4,5 are calculated on a H-depleted molecular graph. These information indices are based on the distribution of topological distances in a molecular graph. The mean information content on the distance degree equality (IDDE) is calculated on a H-depleted molecular graph. This information index is based on the classification of vertex distance degrees, where the vertex distance degree of an atom is the sum of topological distances from the considered atom to any other atom in the H-depleted molecular graph. The mean information content on the distance degree magnitude (IDDM) is calculated on a H-depleted molecular graph. This information index is based on the classification of vertex distance degrees according to their magnitude (i.e. the number of atoms in an equivalent class). The mean information content on the vertex degree equality (IVDE) is calculated on a H-depleted molecular graph. This information index is based on the classification of vertices according to vertex degree. The mean information content on the vertex degree magnitude (IVDM)5 is calculated on a H-depleted molecular graph. This index is based on the classification of vertices according to the vertex degree magnitude. Among the information indices alvaDesc calculates the number of classes on electrotopological state (Ges) and the relative number of classes (rGes). Kier symmetry index (S0K)6 is calculated as the total information content of the molecule according to the electrotopological state: where nSK is the number of non-H atoms and ng is the number of topologically equivalent atoms in the g-th class. The graph vertex complexity index (HVcpx)6 and the graph distance complexity index (HDcpx) are derived from the topological distance matrix. The Balaban indices (Uindex, Vindex, Xindex, Yindex)2 are calculated on a H-depleted molecular graph and are based on the formula of the Balaban index (J_M). The Balaban indices are calculated using four atomic information indices, ui, vi, yi, and xi. Atomic information indices ui is defined as: where u03c3i is the i-th vertex distance degree, dij is the topological distance between atoms i and j and nSK the number of non-H atoms. Atomic information indices vi is defined as: where u03c3i is the i-th vertex distance degree and dij is the topological distance between atoms i and j. Atomic information indices yi is defined as: where g runs over all of the different topological distances from the i-th vertex and fgi is the number of distances equal to g from the i-th vertex. Atomic information indices xi is defined as: where u03c3i is the i-th vertex distance degree. The Bertz Molecular Complexity Index (BertzCT) represents with a number the complexity of a molecule2. It does so, by using concepts from graph theory and information theory (e.g., Shannon&rsquo;s entropy formula). The index is calculated on the H-depleted molecular graph using the following formula: where CT is the final index, C( u03b7) is the the connectivity component and C(E) is the atom component. The connectivity component grows with the number of connections of the molecular graph, but it decreases with the symmetry of the graph (i.e., if the atoms have the same symmetry class). The atom component grows with the number of different atoms (e.g., its value is 0 when all the atoms are of the same type). The Bertz Molecular Complexity Index is calculated by alvaDesc using the formulas proposed by Hendrickson (1987)7. alvaDesc calculates five types of indices of neighbourhood symmetry (ICk, TICk, SICk, BICk, CICk)8,9 from 0 up to 5 order. The neighbourhood information Content (IC_K_) is defined as: where g runs over the equivalence classes, ng is the number of atoms belonging to the g-th equivalence class and nAT is the total number of atoms. The neighbourhood total information content (TICk) is a measure of graph complexity and it is defined as: The structural information content (SICk) is defined as: where nAT is the number of atoms. The bonding information bontent (BICk) takes into account the number of bonds and their multiplicity and it is defined as: where nBT is the number of bonds and u03c0b is the bond order of the b-th bond. The complementary information content (CICk) is defined as: where nAT is the number of atoms. C. Shannon, W. Weaver (1949). The Mathematical Theory of Communication, University of Illinois Press, Urbana (ILL)&#160;&#x21a9;&#xfe0e; Bertz, S. H. (1981). The first general index of molecular complexity. Journal of the American Chemical Society, 103(12), 3599 u20133601. https://doi.org/10.1021/ja00402a071&#160;&#x21a9;&#xfe0e; S. M. Dancoff and H. Quastler (1953), in Essays on the Use of Information Theory in Biology. Ed. University of Illinois, Urbana&#160;&#x21a9;&#xfe0e; Bonchev, D., &amp; Trinajsti u0107, N. (2009). On topological characterization of molecular branching. International Journal of Quantum Chemistry, 14(S12), 293 u2013303. https://doi.org/10.1002/qua.560140823&#160;&#x21a9;&#xfe0e; Bonchev, D., &amp; Trinajsti u0107, N. (2009). Chemical information theory: Structural aspects. International Journal of Quantum Chemistry, 22(S16), 463 u2013480. https://doi.org/10.1002/qua.560220845&#160;&#x21a9;&#xfe0e; Kier, L. B. (1987). Inclusion of Symmetry as a Shape Attribute in Kappa Index Analysis. Quantitative Structure-Activity Relationships, 6(1), 8 u201312. https://doi.org/10.1002/qsar.19870060103&#160;&#x21a9;&#xfe0e; Hendrickson, J. B., Huang, P., &amp; Toczko, A. G. (1987). Molecular complexity: a simplified formula adapted to individual atoms. Journal of Chemical Information and Computer Sciences, 27(2), 63 u201367. https://doi.org/10.1021/ci00054a004&#160;&#x21a9;&#xfe0e; Roy, A. B., Basak, S. C., Harriss, D. K., &amp; Magnuson, V. R. (1984). NEIGHBORHOOD COMPLEXITIES AND SYMMETRY OF CHEMICAL GRAPHS AND THEIR BIOLOGICAL APPLICATIONS. In X. J. R. Avula, R. E. Kalman, A. I. Liapis, &amp; E. Y. Rodin (Eds.), Mathematical Modelling in Science and Technology (pp. 745 u2013750). Elsevier. https://doi.org/10.1016/B978-0-08-030156-3.50138-7&#160;&#x21a9;&#xfe0e; Basak, S. C., Magnuson, V. R., Niemi, G. J., Regal, R. R., &amp; Veith, G. D. (1987). Topological indices: their nature, mutual relatedness, and applications. Mathematical Modelling, 8(C), 300 u2013305. https://doi.org/10.1016/0270-0255(87)90594-X&#160;&#x21a9;&#xfe0e; " }, { "uri": "./gui/save_export_data.html", "title": "Save and export data", "tags": [], "description": "", "content": "Once calculated, the descriptors can be exported to tab-separated text by clicking File on the menu bar and choose Save descriptors. In case of a large number of descriptors, alvaDesc will ask if you want to perform a variable reduction. The variable reduction options are: Constant values: constant descriptors won&rsquo;t be saved Near constant values: descriptors containing all but one equal values won&rsquo;t be saved Standard deviation less than: descriptors with standard deviation values lower than the one specified won&rsquo;t be saved At least one missing value: descriptors with at least one missing value (na) won&rsquo;t be saved All missing values: descriptors with all missing values (na) won&rsquo;t be saved Pair correlation larger or equal to: alvaDescGUI applies the fast V-WSP (variable reduction method adapted from space-filling designs)1 using the specified value as threshold Fingerprints can be exported to tab-separated text files by clicking File on the menu bar, by choosing Save fingerprints and then by selecting the desired type from the calculated fingerprints (e.g., MACCS fingerprint). In a similar way, PCA and t-SNE data can be exported to tab-separated text files by clicking File on the menu bar and choose Save analyses. All the calculated descriptors and fingerprints, along with the loaded molecules can be saved in an alvaDesc project by clicking File on the menu and choose Save project as. The alvaDesc project can be imported in alvaModel to create Quantitative Structure Activity/Property Relationship (QSAR/QSPR) models using. Ballabio, D., Consonni, V., Mauri, A., Claeys-Bruno, M., Sergent, M., &amp; Todeschini, R. (2014). A novel variable reduction method adapted from space-filling designs. Chemometrics and Intelligent Laboratory Systems, 136, 147 u2013154. https://doi.org/10.1016/j.chemolab.2014.05.010&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/2d_matrix-based.html", "title": "2D matrix-based descriptors", "tags": [], "description": "", "content": "The 2D matrix-based descriptors1,2 are those calculated using one of the following matrices: ID Matrix Symbol 1 Adjacency matrix A 2 Topological distance matrix D 3 Laplace matrix L 4 Chi matrix X 5 Reciprocal squared distance matrix H2 6 Detour matrix Dt 7 Distance/Detour matrix D/Dt 8-13 Barysz matrix Dz(w) 14-19 Burden matrix B(w) These matrices are calculated on the H-depleted molecular graph. Adjacency matrix The Adjacency matrix (A) is a square symmetric binary matrix. Its elements indicate whether pairs of vertices (atoms) are adjacent (value 1) or not (value 0) in the graph. The elements on the diagonal of this matrix are all zeros. Topological distance matrix The Topological distance matrix (D) is a square symmetric matrix containing the distances, taken pairwise, between the atoms of a molecule. Given a pair of atoms, this distance is calculated counting the minimum number of bonds (shortest path) to go from one atom to the other. Laplace matrix The Laplace matrix (L) is a square symmetric matrix defined as follow: where A is the adjacency matrix and De is the degree matrix that is a diagonal matrix containg for each diagonal entry the degree of the vertex. As for De, the diagonal entries of L represent the degree of the vertex (i.e., the number of bonds of the atom), while the other entries Lij are set to -1 if i and j are connected and 0 otherwise. A set of descriptors has been defined specifically using the Laplace matrix as, for example, the algebraic connectivity (Acon, also known as Fiedler value or Fiedler eigenvalue) that is the second-smallest eigenvalue of L3. The quasi-Wiener index (QW_L, also known as Kirchhoff number) is calculated as follows4,5: where nSK is the number of atoms (H excluded) and u03bbi is the i-th eigenvalue of L. The first Mohar index (TI1_L) is calculated as follows6,7: where nSK is the number of atoms and nBO is the number of bonds, in both cases H atoms are excluded. The second Mohar index (TI2_L) is calculated as follows6,7: where nSK is the number of atoms (H excluded) and Acon is the algebraic connectivity. The spanning tree number (STN_L) is calculated as follows7: where nSK is the number of atoms (H excluded) and u03bbi is the i-th eigenvalue of L. Chi matrix The Chi matrix (X) is a square symmetric matrix. Its elements, like for the adjacency matrix, are equal to 0 if they refer to non-adjacent vertices, otherwise they contain the edge connectivity. The edge connectivity is calculated as follows: where u03b4i and u03b4j are the degrees of the vertex (i.e., the number of bonds of the atom) i and j respectively. Reciprocal squared distance matrix The reciprocal squared distance matrix (H2) is a square symmetric matrix containing the reciprocal squared topological distances, taken pairwise, between the atoms of a molecule. Detour matrix The detour matrix (Dt) of a graph is a symmetric matrix whose entries are the length of the longest path from a pair of vertices (atoms). Distance/Detour matrix The Distance/Detour matrix (D/Dt) is a square symmetric matrix containing the ratios between the shortest and the longest path of all pair of vertices (atoms)8. In an acyclic graph the shortest and longest paths are the same for any pair of vertices (atoms), so this matrix would contain 0 in the diagonal and 1 elsewhere. The calculation of the detour matrix can be time demanding for molecules including a big number of fused rings. In order to avoid an excessive increase of the calculation time a constraint can be defined. See Complex molecules Barysz matrix The Barysz matrix (Dz(w)) is a square symmetric matrix containing the weigthed distances, taken pairwise, between the atoms of a molecule 9: where: w is one of the following properties: Z: atomic number m: atomic mass v: atomic van der Waals volume e: Sanderson electronegativity p: atomic polarizability i: atomic ionization potential u03c0b is the bond order expressed as: 1: single bond 2: double bond 3: triple bond 1.5: aromatic bond wC is the property calculated on a carbon atom b is the bond under consideration dij is the topological distance between i and j wb(1) and wb(2) are the properties for the two atoms connected by the bond under consideration The bonds considered in the summation are the ones involved in the shortest path to go from i to j. If there is more than one path, the one with the minimum sum is taken. Burden matrix The Burden matrix (B(w)) is a square symmetric matrix defined as follow: where w is the atom property and u03c0b is the bond order (see Barysz matrix) List of the 2D matrix-based descriptors The list of 2D matrix-based descriptors is in the Matrix-based descriptors chapter. Consonni, V., &amp; Todeschini, R. (2012). Multivariate Analysis of Molecular Descriptors. In Statistical Modelling of Molecular Descriptors in QSAR/QSPR (Vol. 2, pp. 111 u2013147). Wiley-VCH Verlag GmbH &amp; Co. KGaA. https://doi.org/10.1002/9783527645121.ch4&#160;&#x21a9;&#xfe0e; Grisoni, F., Reker, D., Schneider, P., Friedrich, L., Consonni, V., Todeschini, R., Koeberle, A., Werz, O., &amp; Schneider, G. (2017). Matrix-based Molecular Descriptors for Prospective Virtual Compound Screening. Molecular Informatics, 36(1 u20132), 1600091. https://doi.org/10.1002/minf.201600091&#160;&#x21a9;&#xfe0e; Fiedler, M. (1973). Algebraic connectivity of graphs. Czechoslovak Mathematical Journal, 23(2), 298 u2013305. https://doi.org/10.21136/cmj.1973.101168&#160;&#x21a9;&#xfe0e; Mohar, B., Babi u0107, D., &amp; Trinajsti u0107, N. (1993). A Novel Definition of the Wiener Index for Trees. Journal of Chemical Information and Computer Sciences, 33(1), 153 u2013154. https://doi.org/10.1021/ci00011a023&#160;&#x21a9;&#xfe0e; Gutman, I., Yeh, Y.-N., Lee, S.-L., &amp; Luo, Y.-L. (1993). Some recent results in the theory of the Wiener number. Indian Journal of Chemistry -Section A (IJC-A), 32(08), 651 u2013661.&#160;&#x21a9;&#xfe0e; Mohar, B.. u201cLaplacian matrices of graphs. u201d Studies in Physical and Theoretical Chemistry 63 (1989): 1-8&#160;&#x21a9;&#xfe0e; Trinajsti u0107, N., Babi u0107, D., Nikoli u0107, S., Plav u0161i u0107, D., Ami u0107, D., &amp; Mihali u0107, Z. (1994). The Laplacian Matrix in Chemistry. Journal of Chemical Information and Computer Sciences, 34(2), 368 u2013376. https://doi.org/10.1021/ci00018a023&#160;&#x21a9;&#xfe0e; Randi u0107, M. (1997). On Characterization of Cyclic Structures. Journal of Chemical Information and Computer Sciences, 37(6), 1063 u20131071. https://doi.org/10.1021/ci960174t&#160;&#x21a9;&#xfe0e; Ivanciuc, O. (2000). QSAR Comparative Study of Wiener Descriptors for Weighted Molecular Graphs. Journal of Chemical Information and Computer Sciences, 40(6), 1412 u20131422. https://doi.org/10.1021/ci000068y&#160;&#x21a9;&#xfe0e; " }, { "uri": "./gui/script_wizard.html", "title": "Script wizard", "tags": [], "description": "", "content": "The Script wizard is a graphical user interface that allows the generation a script file through a series of steps. The script file generated can then be used with the alvaDescCLI software to automate the calculation of descriptors and fingerprints. To access it click File on the menu bar and choose Script wizard. " }, { "uri": "./descriptors/2d_autocorrelation.html", "title": "2D autocorrelations", "tags": [], "description": "", "content": "Autocorrelation descriptors are a heterogeneous class of molecular descriptors whose general formula is the following: where L is a generic local vertex invariant; u03b1 and u03bb are a scaling and a power parameter, respectively; and u03b4(dij;k) is a Kronecker delta function equal to one for pairs of substructure centres at topological distance dij = k and zero otherwise; A is the number of substructure centres that typically are the molecule atoms. Autocorrelation descriptors are related to the spatial distribution of a generic molecular property, the local vertex invariant L, on the molecular structure space and measure the strength of a relationship between atoms at a predefined distance equal to k. alvaDesc calculates autocorrelation descriptors on a H-filled molecular graph considering the molecule atoms as the substructure centres and using the following atomic properties: atomic mass (m) van der Waals volume (v) atom electronegativity (e) atom polarizability (p) ionization potential (i) intrinsic state (s) Except for intrinsic state (s), 2D autocorrelation descriptors are calculated using the carbon-scaled form of the atomic property. i.e., the ratio between the atomic property and the carbon atom property alvaDesc includes the calculation of the most known autocorrelation descriptors: Moreau-Broto autocorrelation descriptors Moran autocorrelation descriptors Geary autocorrelation descriptors Additionally alvaDesc calculates the Topological charge autocorrelation descriptors. alvaDesc calculates Moreau-Broto, Moran and Geary 2D autocorrelation descriptors from lag 1 to lag 8 Moreau-Broto autocorrelation descriptors (ATS) Moreau and Broto applied for the first time the spatial autocorrelation function to a molecular graph to measure the distribution of atomic properties on the molecular graph1,2. They named the proposed descriptors as the autocorrelation of a topological structure (ATS); these can be derived from the general equation (ACD) setting u03b1 = 1/2 and u03bb = 1 and substituting the generic local vertex invariant Li with any atomic property wi as follows: where nAT is the number of vertices of the H-filled molecular graph, wi and wj are the considered atomic property for atom i and j, and u03b4(dij;k) is a Kronecker delta function equal to one if the topological distance dij is equal to k, zero otherwise. alvaDesc applies a logarithmic transformation to Moreau-Broto autocorrelations to avoid large numbers In addition to Moreau-Broto autocorrelation descriptors alvaDesc provides the calculation of Centred Broto-Moreau autocorrelations (ATSCkw). They are calculated in the same way as Broto-Moreau autocorrelations (ATSkw) descriptors, but using atomic property values centred on the molecule3: where nAT is the number of vertices of the H-filled molecular graph, wi and wj are the considered atomic property for atom i and j, respectively, w is the average value of the considered property on the molecule, and u03b4(dij;k) is a Kronecker delta function equal to one if the topological distance dij is equal to k, zero otherwise. Moran autocorrelation descriptors (MATS) Moran autocorrelation functions have been originally applied in ecological studies to evaluate the spatial distribution of environmental properties4. Moran autocorrelations can be derived from the general equation (ACD) just replacing the scaling and power parameters and the local vertex invariants as follows: where nAT is the number of vertices of the H-filled molecular graph, wi and wj are the considered atomic property for atom i and j, respectively, w is the average value of the considered property on the molecule, and u0394k is the sum of the Kronecker deltas, that is, the number of atom pairs at distance equal to k. Moran coefficient usually takes value in the interval [-1,+1]. Positive values of the coefficients reflect positive spatial autocorrelation, whereas negative values of the coefficients reflect negative spatial autocorrelation; values around 0 mean that there is no specific pattern in the atom property distribution in the molecule Geary autocorrelation descriptors (GATS) Geary autocorrelation functions have been originally applied in ecological studies to evaluate the spatial distribution of environmental properties5. Geary autocorrelations can be derived from the general equation (ACD) just replacing the scaling and power parameters and the local vertex invariants as follows: where nAT is the number of vertices of the H-filled molecular graph, wi and wj are the considered atomic property for atom i and j, respectively, w is the average value of the considered property on the molecule, and u0394k is the sum of the Kronecker deltas, that is, the number of atom pairs at distance equal to k. Geary coefficient is a distance-type function whose values vary from zero to infinite. Small values of Geary autocorrelation coefficients reflect strong spatial autocorrelation; values between 0 and 1 reveal positive autocorrelation, while values larger than 1 are associated to negative spatial autocorrelation Topological charge autocorrelation descriptors Topological charge autocorrelation descriptors6 are derived from an unsymmetrical matrix CT, whose single elements are defined as: where the diagonal elements are the vertex degrees of the atoms, that is, the number of connected atoms in the H-depleted molecular graph, and mij are the elements of the matrix obtained by multiplying the adjacency matrix by the reciprocal squared distance matrix. The diagonal entries of the matrix CT represent the topological valence of the atoms; the off-diagonal entries CTij represent a measure of the net charge transferred from the atom j to the atom i. For each path of length k, a topological charge index (GGIk) is defined as the half-sum of absolute values of all charge terms CTij corresponding to pair of vertices with topological distance equal to k. In addition to the GGIk descriptors alvaDesc calculates the mean topological charge descriptors (JGIk), obtained as the fraction of the corresponding topological charge index GGIk and the total number of summation terms in GGIk. alvaDesc calculates topological charge indices from order 1 to order 10 Topological charge autocorrelation descriptors have been originally proposed to evaluate the charge transfer between pairs of atoms, and therefore the global charge transfer in the molecule alvaDesc calculates the global topological charge index (JGT) defined as the sum of the mean topological charge indices. Moreau, J. L., &amp; Broto, P. (1980). Autocorrelation of molecular structures: Application to SAR studies. Nouveau Journal de Chimie, 4, 757 u2013764&#160;&#x21a9;&#xfe0e; Broto, P., Moreau, G., &amp; Vandycke, C. (1984). Molecular structures: Perception, autocorrelation descriptor and sar studies. European Journal of Medicinal Chemistry, 19, 66 u201370&#160;&#x21a9;&#xfe0e; Hollas, B. (2003). An analysis of the autocorrelation descriptor for molecules. Journal of Mathematical Chemistry, 33(2), 91 u2013101. https://doi.org/10.1023/A:1023247831238&#160;&#x21a9;&#xfe0e; Moran, P. A. P. (1950). Notes on Continuous Stochastic Phenomena. Biometrika, 37(1/2), 17. https://doi.org/10.2307/2332142&#160;&#x21a9;&#xfe0e; Geary, R. C. (1954). The Contiguity Ratio and Statistical Mapping. The Incorporated Statistician, 5(3), 115. https://doi.org/10.2307/2986645&#160;&#x21a9;&#xfe0e; Galvez, J., Garcia, R., Salabert, M. T., &amp; Soler, R. (1994). Charge Indexes. New Topological Descriptors. Journal of Chemical Information and Computer Sciences, 34(3), 520 u2013525. https://doi.org/10.1021/ci00019a008&#160;&#x21a9;&#xfe0e; " }, { "uri": "./gui/log_file.html", "title": "Log file", "tags": [], "description": "", "content": "In case of problems during the use of alvaDesc, the Alvascience team may ask you to send them a log file. Here is a simple procedure for saving a log file: click Help (or the Apple symbol on macOS) on the menu bar and choose Log file change the Log level to the number indicated by the Alvascience team click Ok reproduce the sequence of actions that generate your problem open the Log file interface again and click Save As to save a copy of the log file The log should be activated only for debugging purposes and if indicated by the Alvascience team. To deactivate the logging: click Help (or the Apple symbol on macOS) on the menu bar and choose Log file change the Log level to zero click Ok " }, { "uri": "./descriptors/burden_eigenvalues.html", "title": "Burden eigenvalues", "tags": [], "description": "", "content": "The Burden eigenvalues are calculated on the Burden matrices1 of the H-filled molecular graph. alvaDesc calculates two groups of eigenvalues: Largest eigenvalues which contains the 8 largest positive eigenvalues (SpMax1_Bh(w) &hellip; SpMax8_Bh(w)) Smallest eigenvalues which contains the absolute value of the 8 smallest negative eigenvalues (SpMin1_Bh(w) &hellip; SpMin8_Bh(w)) where Bh is the Burden matrix calculated on the H-filled molecular graph and the w is one of the following weighting schemes: atomic mass (m) van der Waals volume (v) atom electronegativity (e) atom polarizability (p) ionization potential (i) intrinsic state (s) Except for intrinsic state (s), Burden eigenvalues are calculated using the carbon-scaled form of the atomic property. i.e., the ratio between the atomic property and the carbon atom property. These eigenvalues contain information related to the molecular structure and they can be used to identify molecular similarity2. Burden, F. R. (1989). Molecular identification number for substructure searches. Journal of Chemical Information and Modeling, 29(3), 225 u2013227. https://doi.org/10.1021/ci00063a011&#160;&#x21a9;&#xfe0e; Pearlman, R. S., &amp; Smith, K. M. (2002). Novel Software Tools for Chemical Diversity. In 3D QSAR in Drug Design (Vol. 9, pp. 339 u2013353). Springer Netherlands. https://doi.org/10.1007/0-306-46857-3_18&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/p_vsa.html", "title": "P_VSA-like descriptors", "tags": [], "description": "", "content": "alvaDesc calculates 69 P_VSA-like descriptors. P_VSA-like descriptors are based on the sum of atomic contributions to van der Waals surface area, for the atoms having a property in a defined range of values1. The van der Waals surface area is defined for each atom as: where Ri is the atomic van der Waals radius of the atom i as defined in Table 1, nAT is the number of atoms, aij are the elements of the adjacency matrix and dij is calculated as: where bij is the ideal bond length between atom i and j and it is defined as: where rij is the reference bond length as defined in Table 2 and cij is a correction term related to the bond multiplicity: 0 for single bond, 0.1 for aromatic, 0.2 for double and 0.3 for triple bonds. Atom-type R Atom-type R H ( u2013O) 0.8 O (other) 1.779 H ( u2013N, u2013P) 0.7 F 1.496 H (other) 1.485 P 2.287 C 1.950 S 2.185 N 1.950 Cl 2.044 O (oxide) 1.810 Br 2.166 O (acid) 2.152 I 2.358 Table 1: Van der Waals radii used for VSA calculations (Radii are given in Angstroms) Bond-type r Bond-type r Br - Br 2.54 F - F 1.28 Br - C 1.97 F - H 0.87 Br - Cl 2.36 F - I 2.04 Br - F 1.85 F - N 1.41 Br - H 1.44 F - O 1.32 Br - I 2.65 F - P 1.5 Br - N 1.84 F - S 1.64 Br - O 1.58 H - I 1.63 Br - P 2.37 H - N 1.01 Br - S 2.21 H - O 0.97 C - C 1.54 H - P 1.41 C - Cl 1.8 H - S 1.31 C - F 1.35 I - I 2.92 C - H 1.06 I - N 2.26 C - I 2.12 I - O 2.14 C - N 1.47 I - P 2.49 C - O 1.43 I - S 2.69 C - P 1.85 N - N 1.45 C - S 1.81 N - O 1.46 Cl - Cl 2.31 N - P 1.6 Cl - F 1.63 N - S 1.76 Cl - H 1.22 O - O 1.47 Cl - I 2.56 O - P 1.57 Cl - N 1.74 O - S 1.57 Cl - O 1.41 P - P 2.26 Cl - P 2.01 P - S 2.07 Cl - S 2.07 S - S 2.05 Table 2: Reference bond lengths used for VSA calculation (Lengths are given in Angstroms) P_VSA-like descriptors are defined as: where: nAT is the number of atoms VSAi is the van der Waals surface area of the i-th atom w is one of the following weighting scheme: LogP: log P (octanol/water) MR: molar refractivity m: atomic mass v: atomic van der Waals volume e: Sanderson electronegativity p: atomic polarizability i: atomic ionization potential s: intrinsic state ppp: potential pharmacophore points charge: partial charge u03b4: is a Kronecker delta function equal to one for atoms with property value in the specified range, and zero otherwise (see following table) k: is the bin number (see following table) Property Bins Bin 1 Bin 2 Bin 3 Bin 4 Bin 5 Bin 6 Bin 7 Bin 8 Bin 9 Bin 10 Bin 11 Bin 12 Bin 13 Bin 14 logP 8 (- u221e,-1.5) [-1.5,-0.5) [-0.5,-0.25) [-0.25,0) [0,0.25) [0.25,0.52) [0.52,0.75) [0.75,+ u221e) molar refractivity 8 (- u221e,0.9) [0.9,1.5) [1.5,2) [2,2.5) [2.5,3) [3,4) [4,6) [6,+ u221e) mass 5 (- u221e,1) [1,1.2) [1.2,1.6) [1.6,3) [3,+ u221e) Van der Waals volume 4 (- u221e,0.5) [0.5,1) [1,1.3) [1.3,+ u221e) Sanderson electronegativity 6 (- u221e,1) [1,1.1) [1.1,1.2) [1.2,1.3) [1.3,1.4) [1.4,+ u221e) polarizability 4 (- u221e,0.4) [0.4,1) [1,2) [2,+ u221e) ionization potential 4 (- u221e,1) [1,1.15) [1.15,1.25) [1.25,+ u221e) intrinsic state 6 (- u221e,1) [1,1.5) [1.5,2) [2,2.5) [2.5,3) [3.0,+ u221e) partial charge 14 (- u221e,-0.3) [-0.3,-0.25) [-0.25,-0.2) [-0.2,-0.15) [-0.15,-0.1) [-0.1,-0.05) [-0.05,0) [0,0.05) [0.05,0.1) [0.1,0.15) [0.15,0.2) [0.2,0.25) [0.25,0.3) [0.30,+ u221e) Table 3: Property interval ranges for P_VSA-like descriptor calculation In addition to P_VSA-like descriptors based on range of physicochemical properties, alvaDesc calculates 10 pharmacophoric P_VSA-like descriptors. These descriptors are based on atom assignment to the most common Potential Pharmacophore Points (PPP), where w is the symbol defining the potential pharmacophore point. Given a defined pharmacophore type, the P_VSA descriptor is calculated as the sum of the VSA contributions of all the atoms assigned to that PPP. PPP atom-types assignment are used as defined in Potential Pharmacophore Points, in addition the following atom-types have been used: aromatic (ar): atom in an aromatic substructure conjugated (con): atom in a conjugated substructure halogen (hal): any halogen atom cyclic (cyc): atom belonging to cycles terminal (ter): atom with one non-H bond P_VSA-like descriptors can be calculated for compounds comprising only atoms H, C, N, O, S, P and halogens and for which reference bond lengths are defined Labute, P. (2000). A widely applicable set of descriptors. Journal of Molecular Graphics and Modelling, 18(4 u20135), 464 u2013477. https://doi.org/10.1016/S1093-3263(00)00068-1&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/eta_indices.html", "title": "ETA indices", "tags": [], "description": "", "content": "ETA indices comprise a set of two generation descriptors1,2,3 defining chemical features at the atomic as well as molecular level. The ETA indices have been defined using a few graph theoretic parameters as well as some electronic indices to describe the bonding pattern, unsaturation, electronegativity and hydrogen-bonding related information. The definition of the ETA indices is based on five atom parameters: the core count ( u03b1), the electronegativity count ( u03b5), the hydrogen-bonding propensity ( u03c8), the Valence Electron Mobile (VEM) count ( u03b2) and the VEM vertex count ( u03b3). The ETA core count is formally defined as: where PN stands for period number, Z is the atomic number and Zv is the valence electron number. The ETA electronegativity count is defined as follows: where u03b5 is characterised by a significant correlation with Pauling u2019s electronegativity scale1. The ETA hydrogen bonding propensity is defined as follows: The VEM count u03b2 for a non-hydrogen vertex can be defined as: u03b2s is defined as: and u03b2ns is defined as: where the parameters u03c3 and u03c0 correspond respectively to the number of sigma and pi bonds in the connected hydrogen suppressed molecular graph and u03b4 represents a correction factor having a value of 0.5 per atom with loan pair of electrons capable of making resonance with an aromatic ring (e.g. nitrogen of aniline, oxygen of phenol etc.). The parameter x has value equal to 0.5 for sigma bonds between two atoms of similar electronegativity ( u0394 u03b5 u2264 0 3), and for a sigma bond between two atoms of different electronegativity ( u0394 u03b5 &gt;0.3), is considered to be 0.75. The parameter y is assigned depending on the type of double bonds. For a u03c0 bond between two atoms of similar electronegativity ( u0394 u03b5 u2264 0.3), the value of y is 1. For a u03c0 bond between two atoms of different electronegativity ( u0394 u03b5 &gt; 0.3) or for conjugated (non-aromatic) u03c0 system, y is considered to be 1.5. For an aromatic u03c0 system, the value of the y parameter is 2. The VEM vertex count is defined as: where u03b1 is the ETA core count and u03b2 is the VEM count of the considered atom. Here u2019s the list of the ETA indices included in alvaDesc: ID Descriptor name Formula 1 Eta_alpha - eta core count 2 Eta_alpha_A - eta average core count 3 Eta_epsi - eta electronegativity measure 4 Eta_epsi_A - eta average electronegativity measure 5 Eta_betaS - eta sigma VEM count 6 Eta_betaS_A - eta sigma average VEM count 7 Eta_betaP - eta pi and lone pair VEM count 8 Eta_betaP_A - eta pi and lone pair average VEM count 9 Eta_beta - eta VEM count 10 Eta_beta_A - eta average VEM count 11 Eta_C - eta composite index 12 Eta_C_A - eta average composite index 13 Eta_L - eta local composite index 14 Eta_L_A - eta average local composite index 15 Eta_R - eta composite index for reference alkane 16 Eta_F - eta functionality index 17 Eta_F_A - eta average functionality index 18 Eta_LR - eta local composite index for reference alkane 19 Eta_FL - eta local functionality index 20 Eta_FL_A - eta average local functionality index 21 Eta_B - eta branching index 22 Eta_B_A - eta average branching index 23 Eta_sh_p - eta p shape index 24 Eta_sh_y - eta y shape index 25 Eta_sh_x - eta x shape index 26 Eta_D_AlphaA - eta delta alpha a index 27 Eta_D_AlphaB - eta delta alpha b index 28 Eta_epsi_2 - eta electronegativity measure 2 29 Eta_epsi_3 - eta electronegativity measure 3 30 Eta_epsi_4 - eta electronegativity measure 4 31 Eta_epsi_5 - eta electronegativity measure 5 32 Eta_D_epsiA - eta measure of unsaturation and electronegative atom count 33 Eta_D_epsiB - eta measure of unsaturation 34 Eta_D_epsiC - eta measure of electronegativity 35 Eta_D_epsiD - eta measure of hydrogen bond donor atoms 36 Eta_psi1 - eta measure of hydrogen bonding propensity and/or polar surface area 37 Eta_D_psiA - eta measure of hydrogen bonding propensity 38 Eta_D_psiB - eta measure of hydrogen bonding propensity 39 Eta_D_beta - eta measure of electronic features 40 Eta_D_beta_A - eta average measure of electronic features Some notes on the notation used: the subscript R indicates that the considered structure is the structure of the reference alkane, e.g. as in the calculation of the eta composite index for reference alkane (Eta_R). The reference alkane refers to the molecular graph of the original structure where all the heteroatoms are replaced with carbon and multiple bonds are replaced with single bonds NR indicates the total number of atoms including hydrogen in the connected molecular graph of the reference alkane the subscript SS indicates that the considered structure is the saturated carbon skeleton, e.g. as in the calculation of the eta electronegativity measure 4 (Eta_epsi_4). The saturated carbon skeleton refers to the molecular graph of the original structure where all carbon-carbon multiple bonds are considered as single bond NSS indicates the total number of atoms including hydrogen in the connected molecular graph of the saturated carbon skeleton aij indicates the element at position i,j of the adjacency matrix; aij is equal to 1 if the atoms i and j are connected, zero otherwise Eta_B is zero for molecules with nSK less than 3. Roy, K., &amp; Ghosh, G. (2003). Introduction of extended topochemical atom (ETA) indices in the valence electron mobile (VEM) environment as tools for QSAR/QSPR studies. Internet Electronic Journal of Molecular Design, 2, 599 u2013620.&#160;&#x21a9;&#xfe0e; Roy, K., &amp; Das, R. N. (2011). On some novel extended topochemical atom (ETA) parameters for effective encoding of chemical information and modelling of fundamental physicochemical properties. SAR and QSAR in Environmental Research, 22(5 u20136), 451 u2013472.&#160;&#x21a9;&#xfe0e; Roy, K., &amp; Das, R. N. (2015). The u201cETA u201d Indices in QSAR/QSPR/QSTR Research. In K. Roy (Ed.), Quantitative Structure-Activity Relationships in Drug Design, Predictive Toxicology, and Risk Assessment (Vol. i, pp. 46 u201383). Medical Information Science Reference.&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/edge_adjacency_indices.html", "title": "Edge adjacency indices", "tags": [], "description": "", "content": "Edge adjacency indices are topological indices based on edge adjacency matrix (EA), augmented edge adjacency matrices (AEA) and unsymmetrical weighted edge adjacency matrices (EA(w)) of a H-depleted molecular graph. The edge adjacency matrix (EA) is a square symmetric matrix of dimension nBO, where nBO is the number of bonds in an H-depleted molecular graph. The entries of the matrix equal one if the considered bonds are adjacent and zero otherwise. The augmented edge adjacency matrix (AEA) is obtained from the edge adjacency matrix by replacing the zero diagonal elements of the edge adjacency matrix with specific bond properties. The weighted edge adjacency matrix (EA(w)) is obtained from the edge adjacency matrix by replacing the elements corresponding to adjacent bonds with specific bond properties: if two edges i and j are adjacent, the entry i,j is defined by the weight wj of the j-th edge and the symmetric entry j,i is defined by the weight wi of the i-th edge. All other elements are set to 0. Augmented and weighted edge adjacency matrices are defined using the following weights: edge degree (ed) bond order (bo) dipole moments (dm) resonance integrals (ri) Edge adjacency indices block provides four logical groups of descriptors: Spectral indices Connectivity-like indices Spectral moments Eigenvalues Spectral indices Spectral indices are molecular descriptors calculated from the eigenvalues of a graph-theoretical matrix. alvaDesc calculates five spectral indices for each edge adjacency matrix. Leading eigenvalue (SpMax_M(w)) is defined as: Normalized leading eigenvalue (SpMaxA_M(w)) is defined as: Spectral diameter (SpDiam_M(w)) is defined as: Spectral absolute deviation (SpAD_M(w)) is defined as: Spectral mean absolute deviation (SpMAD_M(w)) is defined as: where M is the selected edge adjacency matrix, w the selected weight and u03bb is the eigenvalue. Connectivity-like indices Connectivity-like indices1 are molecular descriptors based on Kier-Hall Connectivity indices. These indices are calculated on the edge adjacency matrices by replacing the vertex degree with the edge degree. The edge degree is the number of edges adjacent to a given edge in the H-depleted molecular graph. alvaDesc provides connectivity-like indices of zero and first order for each type of edge adjacency matrix. Being M the selected edge adjacency matrix and w the selected weight, the edge connectivity indices of order 0 (Chi0_M(w)) is defined as: where u03b4(ei) is the degree of the edge ei and the sum is over all edges in the graph. The edge connectivity indices of order 1 (Chi1_M(w)) is defined as: where u03b4(ei) is the degree of the edge ei, u03b4(ej) is the degree of the edge ej and the sum is carried out over all pairs of adjacent edges in the graph. Spectral moments alvaDesc calculates the spectral moments2,3,4 on the edge adjacency matrix and on weighted and augmented matrices. alvaDesc provides spectral moments up to order 15 and apply a logarithmic transformation defined as: where M is the selected edge adjacency matrix and w the selected weight. Eigenvalues alvaDesc calculates the eigenvalues of the edge adjacency matrix (Eigk_EA), augmented edge adjacency matrices (Eigk_AEA(w)), and weighted edge adjacency matrices (Eigk_EA(w)). For each edge adjacency matrix the first 15 eigenvalues are retained5. Estrada, E., Guevara, N., &amp; Gutman, I. (1998). Extension of Edge Connectivity Index. Relationships to Line Graph Indices and QSPR Applications. Journal of Chemical Information and Computer Sciences, 38(3), 428 u2013431. https://doi.org/10.1021/ci970091s&#160;&#x21a9;&#xfe0e; Estrada, E. (1996). Spectral moments of the edge adjacency matrix in molecular graphs. 1. Definition and applications to the prediction of physical properties of alkanes. Journal of Chemical Information and Computer Sciences, 36(4), 844 u2013849. https://doi.org/10.1021/ci950187r&#160;&#x21a9;&#xfe0e; Estrada, E. (1997). Spectral moments of the edge-adjacency matrix of molecular graphs. 2. Molecules containing heteroatoms and QSAR applications. Journal of Chemical Information and Computer Sciences, 37(2), 320 u2013328. https://doi.org/10.1021/ci960113v&#160;&#x21a9;&#xfe0e; Estrada, E. (1998). Spectral moments of the edge adjacency matrix in molecular graphs. 3. Molecules containing cycles. Journal of Chemical Information and Computer Sciences, 38(1), 23 u201327. https://doi.org/10.1021/ci970030u&#160;&#x21a9;&#xfe0e; Mauri, A., Consonni, V., Pavan, M., &amp; Todeschini, R. (2006). Dragon software: An easy approach to molecular descriptor calculations. Match Communications In Mathematical And In Computer Chemistry, 56(2), 237 u2013248&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/geometrical.html", "title": "Geometrical descriptors", "tags": [], "description": "", "content": "The Geometrical descriptors are a collection of descriptors calculated taking into account the three-dimensional structure of the molecule, i.e., the position of the atoms in the three-dimensional space. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. The Geometrical descriptors are divided in four groups: Size indices Shape indices Delocalization-degree indices COMMA descriptors Geometrical descriptors potentially have higher information content than two-dimensional descriptors, but they have to be treated carefully since their values vary modifying the molecule conformation and may be very sensitive to slight changes in the three-dimensional arrangements. Size indices The Size indices represent information about the size and the atom distribution within the molecule. It comprises the Gravitational indexes1 G1 and G2 which account simultaneously for both the atomic masses (volumes) and for their distribution within the molecular space. The first one considers all the atom pairs and is calculated as follow: Instead, the Gravitational index G2 considers only the pairs of atoms i and j connected by a bond b: In both cases, i and j are the pair of atoms under consideration, mi and mj are the atomic masses normalized by the mass of the carbon atom, rij is the Euclidean distance between the atom pair, nAT is the number of atoms and nBT is the number of bonds in the molecule. The radius of gyration (RGyr) conveys information about the distribution of mass within the molecule. It&rsquo;s a size measure of the molecule that can be calculated without considering the molecular connectivity2: where mi is the atomic mass, ri is the Euclidean distance between the atom and the centre of mass, nAT is the number of atoms and MW is the molecular weight. Note that RGyr will have small values if most atoms are close to the centre of mass. The span R (SPAN) is the radius of the smallest sphere, centred at the center of mass, which completely encloses the distribution of the molecule atoms2. It is calculated as follow: where ri is the Euclidean distance between the atom and the centre of mass. Shape indices The Shape indices groups descriptors that account for the shape of the molecule. It contains the average span R (SPAM) which is calculated as follow: The spherosity (SPH) is based on the covariance matrix calculated from the Geometrical distance matrix (G)3: where u03bb1, u03bb2 and u03bb3 are the eigenvalues of the aforementioned covariance matrix. The spherosity, as defined by alvaDesc, ranges from 0 for a totally spherical molecule to 1 for a flat molecule The 3D Petitjean shape index (PJI3) is a geometrical shape index4,5 calculated as follow: where D is the geometrical diameter which is defined as the longest distance between any pair of atoms (also known as geometrical eccentricity) and R is the geometrical radius which is the shortest distance between any pair of atoms of the molecule. The molecular eccentricity (MEcc) is calculated as2: where u03bb1 and u03bb3 are maximum and minimum eigenvalues of the WHIM covariance matrix weighted by mass. The asphericity (ASP) measures the deviation from the spherical shape which is an aspect of the shape known as anisometry2: where u03bb1, u03bb2 and u03bb3 are the eigenvalues of the WHIM covariance matrix weighted by mass. The length-to-breadth ratio by WHIM (L/Bw)6 is calculated as follow: where u03bb1 and u03bb2 are the eigenvalues of the WHIM covariance matrix weighted by mass. Delocalization-degree indices In this group are collected various indices related to the aromatic character of the molecule (aromaticity) which is associated to the u03c0-electron delocalization7,8. The first example is the Harmonic Oscillator Model of Aromaticity index (HOMA) which is calculated as follow7,8: where B u03c0 is the number of aromatic bonds with both conjugated atoms, Rb is the bond length, Roptb is the optimal bond length and u03b1b is a constant defined by the authors: Bond u03b1 ( u212b) Ropt ( u212b) C-C 257.7 1.388 C-N 93.52 1.334 C-O 157.38 1.265 C-P 118.91 1.698 C-S 94.09 1.677 N-N 130.33 1.309 N-O 57.21 1.248 Bonds among other atom types are not ignored. The resulting value is set to 0 in case no molecule bond matches the requirements. The conjugated maximum bond length (CMBL) is the longest aromatic bond in the molecule. The aromaticity index (AROM) is calculated as follow9,10: where B u03c0 is the number of aromatic bonds with both conjugated atoms, Rb is the bond length and R is the average length of all the considered bonds. The HOMA total (HOMT) is calculated as follow: where B u03c0 has the same meaning as in HOMA. COMMA descriptors The Comparative Molecular Moment Analysis (CoMMA2)11 descriptors are based on the weighted covariance matrix of the WHIM descriptors: DISPw: displacement value QXXw: quadrupole x-component value QYYw: quadrupole y-component value QZZw: quadrupole z-component value where w is the weighting scheme. Katritzky, A. R., Mu, L., Lobanov, V. S., &amp; Karelson, M. (1996). Correlation of Boiling Points with Molecular Structure. 1. A Training Set of 298 Diverse Organics and a Test Set of 9 Simple Inorganics. The Journal of Physical Chemistry, 100(24), 10400 u201310407. https://doi.org/10.1021/jp953224q&#160;&#x21a9;&#xfe0e; Arteca, G. A. (1996). Molecular Shape Descriptors. In Reviews in Computational Chemistry (Vol. 9, pp. 191 u2013253). https://doi.org/10.1002/9780470125861.ch5&#160;&#x21a9;&#xfe0e; Robinson, D. D., Barlow, T. W., &amp; Richards, W. G. (1997). Reduced Dimensional Representations of Molecular Structure. Journal of Chemical Information and Computer Sciences, 37(5), 939 u2013942. https://doi.org/10.1021/ci970424l&#160;&#x21a9;&#xfe0e; Petitjean, M. (1992). Applications of the Radius-Diameter Diagram to the Classification of Topological and Geometrical Shapes of Chemical Compounds. Journal of Chemical Information and Computer Sciences, 32(4), 331 u2013337. https://doi.org/10.1021/ci00008a012&#160;&#x21a9;&#xfe0e; Bath, P. A., Poirrette, A. R., Willett, P., &amp; Allen, F. H. (1995). The Extent of the Relationship between the Graph-Theoretical and the Geometrical Shape Coefficients of Chemical Compounds. Journal of Chemical Information and Computer Sciences, 35(4), 714 u2013716. https://doi.org/10.1021/ci00026a007&#160;&#x21a9;&#xfe0e; Todeschini, R., &amp; Consonni, V. (2000). Handbook of Molecular Descriptors. Wiley. https://doi.org/10.1002/9783527613106&#160;&#x21a9;&#xfe0e; Krygowski, T. M., Cyra u0144ski, M., Ciesielski, A., u015awirska, B., &amp; Leszczy u0144ski, P. (1996). Separation of the energetic and geometric contributions to aromaticity. 2. Analysis of the aromatic character of benzene rings in their various topological environments in the benzenoid hydrocarbons. Crystal and molecular structure of coronene. Journal of Chemical Information and Computer Sciences, 36(6), 1135 u20131141. https://doi.org/10.1021/ci960367g&#160;&#x21a9;&#xfe0e; Krygowski, T. M., Ciesielski, A., Bird, C. W., &amp; Kotschy, A. (1995). Aromatic Character of the Benzene Ring Present in Various Topological Environments in Benzenoid Hydrocarbons. Nonequivalence of Indices of Aromaticity. Journal of Chemical Information and Computer Sciences, 35(2), 203 u2013210. https://doi.org/10.1021/ci00024a006&#160;&#x21a9;&#xfe0e; Bird, C. W. (1985). A new aromaticity index and its application to five-membered ring heterocycles. Tetrahedron, 41(7), 1409 u20131414. https://doi.org/10.1016/S0040-4020(01)96543-3&#160;&#x21a9;&#xfe0e; Bird, C. W. (1985). A new aromaticity index and its application to five-membered ring heterocycles. Tetrahedron, 41(7), 1409 u20131414. https://doi.org/10.1016/S0040-4020(01)96543-3&#160;&#x21a9;&#xfe0e; Silverman, B. D. (2000). Three-Dimensional Moments of Molecular Property Fields. Journal of Chemical Information and Computer Sciences, 40(6), 1470 u20131476. https://doi.org/10.1021/ci000457s&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/3d_matrix-based.html", "title": "3D matrix-based descriptors", "tags": [], "description": "", "content": "The 3D matrix-based descriptors are those calculated using one of the following matrices: ID Matrix Symbol 1 Geometrical distance matrix G 2 Reciprocal squared geometrical distance matrix RG 3 Distance/distance matrix G/D 4 Coulomb matrix Coulomb These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. Geometrical distance matrix The Geometrical distance matrix (G) is a square symmetric matrix containing the Euclidean distances, taken pairwise, between the atoms of a molecule. The geometric distance degree is defined as the sum of the distances between a given atom and any other atom of the molecule. This information can be obtained by summing either a row or a column of this matrix. Reciprocal squared geometrical distance matrix The Reciprocal squared geometrical distance matrix (RG) is a square symmetric matrix containing the inverse of the squared Euclidean distances, taken pairwise, between the atoms of a molecule. Distance/distance matrix The Distance/distance matrix (G/D) is a square symmetric matrix containing the ratios between the Euclidean distance and the Topological distance, taken pairwise, between the atoms of a molecule1. Coulomb matrix The Coulomb matrix (Coulomb) is a square symmetric matrix defined as follows: where Zi is the atomic number of the i-th atom and rij is the interatomic distance between atom i and atom j. In the Coulomb matrix the off-diagonal elements correspond to the Coulomb repulsion between atoms i and j, while diagonal elements encode a polynomial fit of atomic energies to nuclear charge2,3. List of the 3D matrix-based descriptors The list of 3D matrix-based descriptors is in the Matrix-based descriptors chapter. M. Randic, A.F. Kleiner, L.M. DeAlba (1994), Distance/Distance Matrices. J.Chem.Inf.Comput.Sci. 34, 277-286&#160;&#x21a9;&#xfe0e; Rupp, M., Tkatchenko, A., M u00fcller, K. R., &amp; Von Lilienfeld, O. A. (2012). Fast and accurate modeling of molecular atomization energies with machine learning. Physical Review Letters, 108(5), 1 u20135&#160;&#x21a9;&#xfe0e; Schrier, J. (2020). Can One Hear the Shape of a Molecule (from its Coulomb Matrix Eigenvalues)? Journal of Chemical Information and Modeling, 60, 3804 u20133811&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/3d_autocorrelations.html", "title": "3D autocorrelations", "tags": [], "description": "", "content": "3D autocorrelation descriptors are based on a modified autocorrelation approach for which 3D structural information is included. 3D autocorrelation descriptors theory is based on the assumption that the relationship between topological and Euclidean distance is able to provide significant 3D-structural information, i.e. different conformations of the same molecular structure affects the Euclidean but not the topological distance between two atoms 1. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. alvaDesc calculates 3D autocorrelations as: where k and w are the considered lag and atomic property, u0394k is the number of atom pairs located at a topological distance dij equal to k, rij is the geometric distance between the i-th and the j-th atoms and u03b4 is the Kronecker delta, which is equal to one when dij is equal to k, and zero otherwise. nAT is the total number of atoms (H included). alvaDesc calculates 3D autocorrelations up to lag 10 using the following atomic properties: unitary weight (u) atomic mass (m) van der Waals volume (v) atom electronegativity (e) atom polarizability (p) ionization potential (i) intrinsic state (s) covalent radius (r) Atomic properties are used in the carbon-scaled form except for the intrinsic state property. Klein, C. T., Kaiser, D., &amp; Ecker, G. (2004). Topological Distance Based 3D Descriptors for Use in QSAR and Diversity Analysis. Journal of Chemical Information and Computer Sciences, 44(1), 200 u2013209. https://doi.org/10.1021/ci0256236&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/rdf.html", "title": "RDF descriptors", "tags": [], "description": "", "content": "The RDF descriptors take their name from the Radial Distribution Function. The radial distribution function of an ensemble of n atoms can be interpreted as the probability distribution to find an atom in a given spherical volume1. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. They are calculated using the following formula: where: nAT is the number of atoms w is one of the following weighting scheme: u: unweighted, in this case wi and wj are equal to 1 m: atomic mass v: atomic van der Waals volume e: Sanderson electronegativity p: atomic polarizability i: atomic ionization potential s: intrinsic state R: is the radius of the spherical volume. alvaDesc calculates RDF starting from 1 u212b up to 15.5 u212b with a step of 0.5 u212b rij is the distance between the atoms i and j B is the smoothing parameter which defines the probability distribution of the individual distances. alvaDesc uses a B equal to 100 u212b-2 Hemmer, M. C., Steinhauer, V., &amp; Gasteiger, J. (1999). Deriving the 3D structure of organic molecules from their infrared spectra. Vibrational Spectroscopy, 19(1), 151 u2013164. https://doi.org/10.1016/s0924-2031(99)00014-4&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/3d_morse.html", "title": "3D-MoRSE descriptors", "tags": [], "description": "", "content": "3D-MoRSE stands for 3D Molecule Representation of Structures based on Electron diffraction1. The 3D-MoRSE descriptors transform the 3D coordinates into a molecular code with a modified equation used in electron diffraction studies for preparing theoretical scattering curves2. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. They are calculated using the following formula: where: nAT is the number of atoms w is one of the following weighting scheme: u: unweighted, in this case wi and wj are equal to 1 m: atomic mass v: atomic van der Waals volume e: Sanderson electronegativity p: atomic polarizability i: atomic ionization potential s: intrinsic state s: is the scattering parameter rij is the geometrical distance between the atoms i and j The formula is calculated for 32 evenly distributed values of s in the range of 0 - 31.0 u212b-1 from the 3D coordinates of the molecule. 3D-MoRSE descriptors have been used in a number of QSAR/QSPR studies and they have been studied to relate their values and the molecular structure3. Schuur, J. H., Selzer, P., &amp; Gasteiger, J. (1996). The coding of the three-dimensional structure of molecules by molecular transforms and its application to structure-spectra correlations and studies of biological activity. Journal of Chemical Information and Computer Sciences, 36(2), 334 u2013344. https://doi.org/10.1021/ci950164c&#160;&#x21a9;&#xfe0e; Soltzberg, L. J., &amp; Wilkins, C. L. (1977). Molecular Transforms: A Potential Tool for Structure-Activity Studies. Journal of the American Chemical Society, 99(2), 439 u2013443. https://doi.org/10.1021/ja00444a021&#160;&#x21a9;&#xfe0e; Devinyak, O., Havrylyuk, D., &amp; Lesyk, R. (2014). 3D-MoRSE descriptors explained. Journal of Molecular Graphics and Modelling, 54, 194 u2013203. https://doi.org/10.1016/j.jmgm.2014.10.006&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/whim.html", "title": "WHIM descriptors", "tags": [], "description": "", "content": "The Weighted Holistic Invariant Molecular (WHIM)1,2 descriptors comprise information about size, shape, symmetry and atom distribution of the molecule. Due to the way they are calculated, they are invariant to translation and rotation and, therefore, they avoid any molecule alignment problem. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. They are based on an algorithm that involves the calculation of the weighted covariance matrix (S)1,3,4 which is a three by three matrix defined as follow: where i and j represent the three possible coordinates (x, y, z), nAT is the number of atoms, wa is the weighting scheme, qai and qaj are the centred coordinate i and j of the atom a, and q u203ei and q u203ej are the average value of the coordinate i and j respectively. Afterwards, a Principal Component Analysis (PCA) is performed on S obtaining the related eigenvalues and eigenvectors of the three principal component axes. The WHIM descriptors are divided in two groups: Directional descriptors: calculated using the projections of the atoms along each individual principal component axis Global descriptors: calculated as a combination of the principal component axes, giving a global information about the molecule Here&rsquo;s the list of the WHIM descriptors: Descriptor name Group Formula Lkw - size directional WHIM index for component k with weighted scheme w Directional descriptors Pkw - shape directional WHIM index for component k with weighted scheme w Directional descriptors Gkw - symmetry directional WHIM index for component k with weighted scheme w Directional descriptors Ekw - accessibility directional WHIM index for component k with weighted scheme w Directional descriptors Tw - T total size index with weighted scheme w Global descriptors Aw - A total size index with weighted scheme w Global descriptors Gw - total symmetry index with weighted scheme w Global descriptors Kw - K global shape index with weighted scheme w Global descriptors Dw - D total accessibility index with weighted scheme w Global descriptors Vw - V total size index with weighted scheme w Global descriptors Some notes on the notation used: w is the weighting scheme u03bbk is the k eigenvalue of the weighted covariance matrix S with weighted scheme w nAT is the number of atoms nsk is the number of symmetric atoms along the k principal component axis nnk is the number of non-symmetric atoms along the k principal component axis tik is the value of the k coordinate (x, y, z) of the atom i projected on the k principal component axis More notes on the formulas: Lkw descriptors represent the variances of the atoms centred coordinates on the k principal component axis and they are related to the molecular size: a straight molecule has L2w and L3w equal to zero a planar molecule has L3w equal to zero an ideal spherical molecule has L1w, L2w and L3w equal to 1/3 Ekw descriptors are related to the atoms distribution and density around the origin and along the principal component axis k. The calculation is based on the statistical concept of kurtosis. In particular, in order to avoid issues related to very large kurtosis values, the inverse of the kurtosis is used. These descriptors are also called emptiness3 because they are related to the amount of unfilled space per projected atom, and the greater Ekw the greater the projected unfilled space. Kw descriptors account for the molecular shape: a straight molecule has value 1 a planar molecule has value between 0.5 and 1 an ideal spherical molecule has value of 0 Todeschini, R., Lasagni, M., &amp; Marengo, E. (1994). New molecular descriptors for 2D and 3D structures. Theory. Journal of Chemometrics, 8(4), 263 u2013272. https://doi.org/10.1002/cem.1180080405&#160;&#x21a9;&#xfe0e; Todeschini, R., &amp; Gramatica, P. (1997). 3D-modelling and Prediction by WHIM Descriptors. Part 6. Application of WHIM Descriptors in QSAR Studies. Quantitative Structure-Activity Relationships, 16(2), 120 u2013125. https://doi.org/10.1002/qsar.19970160204&#160;&#x21a9;&#xfe0e; Todeschini, R., &amp; Gramatica, P. (1998). New 3D Molecular Descriptors: The WHIM theory and QSAR Applications. Perspectives in Drug Discovery and Design, 9/10/11, 355 u2013380. https://doi.org/https://doi.org/10.1023/A:1027284627085&#160;&#x21a9;&#xfe0e; Todeschini, R., &amp; Gramatica, P. (1997). The Whim Theory: New 3D Molecular Descriptors for Qsar in Environmental Modelling. SAR and QSAR in Environmental Research, 7(1 u20134), 89 u2013115. https://doi.org/10.1080/10629369708039126&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/getaway.html", "title": "GETAWAY descriptors", "tags": [], "description": "", "content": "The GEometry, Topology, and Atom-Weights AssemblY (GETAWAY)1,2 descriptors encode both geometrical and topological information of the molecule. They are all based on the calculation of the molecular influence matrix (H): where M is the symmetric matrix containing the 3D coordinates of the H-filled molecular graph. The 3D coordinates are centred to the geometrical centre of the molecule. Therefore, H is a symmetric matrix and is rotational invariant with respect to the molecule coordinates. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. The diagonal elements of H are called leverages (hii) and they represent the influence of each atom in determining the shape of the molecule. The leverage values (0 u2264 hii u2264 1) decrease as the atoms are closer to the molecule centre. They are affected by conformational changes and by the bond lengths, which in turn are affected by atom and bond types. The influence/distance matrix (R) is the other key matrix for the calculation of the GETAWAY descriptors. It is a symmetric matrix where the diagonal elements are set to zero and the off-diagonal elements are calculated in the following way: where hii and hjj are the leverages (H) of two different atoms and rij is their interatomic distance. Therefore, pairs of atoms far apart give less contribution to this matrix. The GETAWAY descriptors are divided in three groups: Basic descriptors: calculated using the diagonal elements of molecular influence matrix (H) H indices: which combine spatial autocorrelation formulas using a weighting scheme and the molecular influence matrix (H) R indices: which combine matrix operation, spatial autocorrelation formulas using a weighting scheme and the influence/distance matrix (R) Here&rsquo;s the list of the GETAWAY descriptors: Descriptor name Group Formula ITH - total information content on the leverage equality Basic descriptors ISH - standardized information content on the leverage equality Basic descriptors HIC - mean information content on the leverage magnitude Basic descriptors HGM - geometric mean on the leverage magnitude Basic descriptors Hkw - H autocorrelation of lag k with weighted scheme w H indices HTw - H total index with weighted scheme w H indices HATSkw - leverage-weighted autocorrelation of lag k with weighted scheme w H indices HATSw - leverage-weighted total index with weighted scheme w H indices RCON - Randic-type R matrix connectivity R indices RARS - R matrix average row sum R indices REIG - first eigenvalue of the R matrix R indices Rkw - R autocorrelation of lag k with weighted scheme w R indices RTw - R total index R indices Rkw+ - R maximal autocorrelation of lag k with weighted scheme w R indices RTw+ - R maximal index with weighted scheme w R indices Some notes on the notation used: nAT is the number of atoms nBT is the number of bonds nSK is the number of atoms (H excluded) w is the weighting scheme Ng is the number of atoms with the same leverage values G is the number of equivalence classes D is the rank of M and can assume three values: 1 for linear, 2 for planar and 3 for 3D molecules dij is the topological distance between the atoms i and j d is the topological diameter config.toml content hugocomp.sh hugoserv.sh public _&delta;(k; dij)_ is the Dirac-delta function where _&delta;_&#61;1 if _dij_&#61;_k_, _&delta;_&#61;0 otherwise config.toml content hugocomp.sh hugoserv.sh public _&delta;(k; dij; hij)_ is the Dirac-delta function where _&delta;_&#61;1 if _dij_&#61;_k_ and hij&gt;0, _&delta;_&#61;0 otherwise config.toml content hugocomp.sh hugoserv.sh public _&lambda;1_ is the first eigenvalue of _R_ config.toml content hugocomp.sh hugoserv.sh public _RSi_ and _RSj_ are respectively the row sum of row _i_ and _j_ of _R_ More notes on the formulas: molecules having no atoms with different leverages are molecules with no element of symmetry (ITH = nSK config.toml content hugocomp.sh hugoserv.sh public log 2nSK and ISH = 1). Instead, molecules where all the atoms have the same leverage values are completely symmetric (ITH = 0 and ISH = 0) the authors1 pointed out that HIC seems to catch more information related to molecular complexity than ITH and ISH. And contrary to these two, it is sensitive to the presence of multiple bonds. HGM increases when the leverages have similar values (e.g., for molecules of almost spherical shape). To prevent the result from being excessively lowered by one or a few atoms, atoms very close to the centre are excluded from the calculation. Consonni, V., Todeschini, R., Pavan, M., &amp; Gramatica, P. (2002). Structure/Response Correlations and Similarity/Diversity Analysis by GETAWAY Descriptors . 1. Theory of the Novel 3D Molecular Descriptors. J Chem Inf Comput Sci, 42(3), 682 u2013692. https://doi.org/10.1021/ci0155053&#160;&#x21a9;&#xfe0e; Consonni, V., Todeschini, R., Pavan, M., &amp; Gramatica, P. (2002). Structure/response correlations and similarity/diversity analysis by GETAWAY descriptors. 2. Application of the novel 3D molecular descriptors to QSAR/QSPR studies. Journal of Chemical Information and Computer Sciences, 42(3), 693 u2013705. https://doi.org/10.1021/ci0155053&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/randic_molecular_profiles.html", "title": "Randic molecular profiles", "tags": [], "description": "", "content": "The Randic molecular profiles outline a numerical characterization of the molecular shape1. The molecular shape is represented by a sequence of values, e.g.: These sequences can be used to compare molecules. These descriptors can only be calculated for molecules having 3D coordinates. They are calculated using the following formula: where: nAT is the number of atoms rij is the geometrical distance between the atoms i and j The formula is calculated for 20 values of k in the range of 1 - 20 and is applied to: the atoms of the H-depleted molecular graph (Molecular profiles, DPk) the peripheral atoms (number of non-H bonds less than 3) of the H-depleted molecular graph (Shape profiles, SPk) alvaDesc also provides the average shape profile index of order 2 (SHP2) which is calculated as SP02 divided by the number of peripheral atoms. Randi u0107, M. (1995). Molecular Shape Profiles. Journal of Chemical Information and Computer Sciences, 35(3), 373 u2013382. https://doi.org/10.1021/ci00025a005&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/functional_groups.html", "title": "Functional group count", "tags": [], "description": "", "content": "alvaDesc calculates 154 functional group counts. ID Symbol Description 1 nCp number of terminal primary C(sp3) 2 nCs number of total secondary C(sp3) 3 nCt number of total tertiary C(sp3) 4 nCq number of total quaternary C(sp3) 5 nCrs number of ring secondary C(sp3) 6 nCrt number of ring tertiary C(sp3) 7 nCrq number of ring quaternary C(sp3) 8 nCar number of aromatic C(sp2) 9 nCbH number of unsubstituted benzene C(sp2) 10 nCb- number of substituted benzene C(sp2) 11 nCconj number of non-aromatic conjugated C(sp2) 12 nR=Cp number of terminal primary C(sp2) 13 nR=Cs number of aliphatic secondary C(sp2) 14 nR=Ct number of aliphatic tertiary C(sp2) 15 n=C= number of allenes groups 16 nR#CH/X number of terminal C(sp) 17 nR#C- number of non-terminal C(sp) 18 nROCN number of cyanates (aliphatic) 19 nArOCN number of cyanates (aromatic) 20 nRNCO number of isocyanates (aliphatic) 21 nArNCO number of isocyanates (aromatic) 22 nRSCN number of thiocyanates (aliphatic) 23 nArSCN number of thiocyanates (aromatic) 24 nRNCS number of isothiocyanates (aliphatic) 25 nArNCS number of isothiocyanates (aromatic) 26 nRCOOH number of carboxylic acids (aliphatic) 27 nArCOOH number of carboxylic acids (aromatic) 28 nRCOOR number of esters (aliphatic) 29 nArCOOR number of esters (aromatic) 30 nRCONH2 number of primary amides (aliphatic) 31 nArCONH2 number of primary amides (aromatic) 32 nRCONHR number of secondary amides (aliphatic) 33 nArCONHR number of secondary amides (aromatic) 34 nRCONR2 number of tertiary amides (aliphatic) 35 nArCONR2 number of tertiary amides (aromatic) 36 nROCON number of (thio-) carbamates (aliphatic) 37 nArOCON number of (thio-) carbamates (aromatic) 38 nRCOX number of acyl halogenides (aliphatic) 39 nArCOX number of acyl halogenides (aromatic) 40 nRCSOH number of thioacids (aliphatic) 41 nArCSOH number of thioacids (aromatic) 42 nRCSSH number of dithioacids (aliphatic) 43 nArCSSH number of dithioacids (aromatic) 44 nRCOSR number of thioesters (aliphatic) 45 nArCOSR number of thioesters (aromatic) 46 nRCSSR number of dithioesters (aliphatic) 47 nArCSSR number of dithioesters (aromatic) 48 nRCHO number of aldehydes (aliphatic) 49 nArCHO number of aldehydes (aromatic) 50 nRCO number of ketones (aliphatic) 51 nArCO number of ketones (aromatic) 52 nCONN number of urea (-thio) derivatives 53 nC=O(O)2 number of carbonate (-thio) derivatives 54 nN=C-N&lt; number of amidine derivatives 55 nC(=N)N2 number of guanidine derivatives 56 nRC=N number of imines (aliphatic) 57 nArC=N number of imines (aromatic) 58 nRCNO number of oximes (aliphatic) 59 nArCNO number of oximes (aromatic) 60 nRNH2 number of primary amines (aliphatic) 61 nArNH2 number of primary amines (aromatic) 62 nRNHR number of secondary amines (aliphatic) 63 nArNHR number of secondary amines (aromatic) 64 nRNR2 number of tertiary amines (aliphatic) 65 nArNR2 number of tertiary amines (aromatic) 66 nN-N number of N hydrazines 67 nN=N number of N azo-derivatives 68 nRCN number of nitriles (aliphatic) 69 nArCN number of nitriles (aromatic) 70 nN+ number of positively charged N 71 nNq number of quaternary N 72 nRNHO number of hydroxylamines (aliphatic) 73 nArNHO number of hydroxylamines (aromatic) 74 nRNNOx number of N-nitroso groups (aliphatic) 75 nArNNOx number of N-nitroso groups (aromatic) 76 nRNO number of nitroso groups (aliphatic) 77 nArNO number of nitroso groups (aromatic) 78 nRNO2 number of nitro groups (aliphatic) 79 nArNO2 number of nitro groups (aromatic) 80 nN(CO)2 number of imides (-thio) 81 nC=N-N&lt; number of hydrazones 82 nROH number of hydroxyl groups 83 nArOH number of aromatic hydroxyls 84 nOHp number of primary alcohols 85 nOHs number of secondary alcohols 86 nOHt number of tertiary alcohols 87 nROR number of ethers (aliphatic) 88 nArOR number of ethers (aromatic) 89 nROX number of hypohalogenides (aliphatic) 90 nArOX number of hypohalogenides (aromatic) 91 nO(C=O)2 number of anhydrides (-thio) 92 nH2O number of water molecules 93 nSH number of thiols 94 nC=S number of thioketones 95 nRSR number of sulfides 96 nRSSR number of disulfides 97 nSO number of sulfoxides 98 nS(=O)2 number of sulfones 99 nSOH number of sulfenic (thio-) acids 100 nSOOH number of sulfinic (thio-/dithio-) acids 101 nSO2OH number of sulfonic (thio-/dithio-) acids 102 nSO3OH number of sulfuric (thio-/dithio-) acids 103 nSO2 number of sulfites (thio-/dithio-) 104 nSO3 number of sulfonates (thio-/dithio-) 105 nSO4 number of sulfates (thio-/dithio-) 106 nSO2N number of sulfonamides (thio-/dithio-) 107 nPO3 number of phosphites/thiophosphites 108 nPO4 number of phosphates/thiophosphates 109 nPR3 number of phosphanes 110 nP(=O)O2R number of phosphonates (thio-) 111 nP(=O)R3/nPR5 number of phosphoranes (thio-) 112 nCH2RX number of CH2RX 113 nCHR2X number of CHR2X 114 nCR3X number of CR3X 115 nR=CHX number of R=CHX 116 nR=CRX number of R=CRX 117 nR#CX number of R#CX 118 nCHRX2 number of CHRX2 119 nCR2X2 number of CR2X2 120 nR=CX2 number of R=CX2 121 nCRX3 number of CRX3 122 nArX number of X on aromatic ring 123 nCXr number of X on ring C(sp3) 124 nCXr= number of X on ring C(sp2) 125 nCconjX number of X on exo-conjugated C 126 nAziridines number of Aziridines 127 nOxiranes number of Oxiranes 128 nThiranes number of Thiranes 129 nAzetidines number of Azetidines 130 nOxetanes number of Oxetanes 131 nThioethanes number of Thioethanes 132 nBeta-Lactams number of Beta-Lactams 133 nPyrrolidines number of Pyrrolidines 134 nOxolanes number of Oxolanes 135 ntH-Thiophenes number of tetrahydro-thiophenes 136 nPyrroles number of Pyrroles 137 nPyrazoles number of Pyrazoles 138 nImidazoles number of Imidazoles 139 nFuranes number of Furanes 140 nThiophenes number of Thiophenes 141 nOxazoles number of Oxazoles 142 nIsoxazoles number of Isoxazoles 143 nThiazoles number of Thiazoles 144 nIsothiazoles number of Isothiazoles 145 nTriazoles number of Triazoles 146 nPyridines number of Pyridines 147 nPyridazines number of Pyridazines 148 nPyrimidines number of Pyrimidines 149 nPyrazines number of Pyrazines 150 n135-Triazines number of 1-3-5-Triazines 151 n124-Triazines number of 1-2-4-Triazines 152 nHDon number of donor atoms for H-bonds (N and O) 153 nHAcc number of acceptor atoms for H-bonds (N,O,F) 154 nHBonds number of intramolecular H-bonds (with N,O,F) Table 1: Symbol and descriptions of the 154 functional groups calculated by alvaDesc The number of donor atoms for H-bonds (nHDon) is calculated by summing the hydrogen atoms bonded to any nitrogen or oxygen. The number of acceptor atoms for H-bonds (nHAcc) is calculated by summing any nitrogen, oxygen and fluorine, excluding quaternary nitrogen and pyrrolyl form of nitrogen. The number of intramolecular H-bonds (nHBonds) is a three-dimensional descriptor counting the atom pairs (i,j) where: atom i and j are at a topological distance equal to 3 or 4 atom i and j are at a geometrical distance between 1 and 2.7 u00c5 atom i is a boron, nitrogen, oxygen, aluminium, phosphor or sulfur with at least one bonded hydrogen atom j is a nitrogen, oxygen or fluorine atom " }, { "uri": "./descriptors/atom_centred_fragments.html", "title": "Atom-centred fragments", "tags": [], "description": "", "content": "Atom-centred fragments block includes 115 structural features originally proposed for the prediction of octanol-water partition coefficient (log P) and molar refractivity (MR)1,2. Structural feature definitions and their coefficient for the calculation of ALOGP and AMR descriptors is provided in the table below. ID Symbol Description Hydrophobicity MR 1 C-001 CH3R, CH4 -1.5603 2.9680 2 C-002 CH2R2 -1.0120 2.9116 3 C-003 CHR3 -0.6681 2.8028 4 C-004 CR4 -0.3698 2.6205 5 C-005 CH3X -1.7880 3.0150 6 C-006 CH2RX -1.2486 2.9244 7 C-007 CH2X2 -1.0305 2.6329 8 C-008 CHR2X -0.6805 2.5040 9 C-009 CHRX2 -0.3858 2.3770 10 C-010 CHX3 0.7555 2.5590 11 C-011 CR3X -0.2849 2.3030 12 C-012 CR2X2 0.0200 2.3006 13 C-013 CRX3 0.7894 2.9627 14 C-014 CX4 1.6422 2.3038 15 C-015 =CH2 -0.7866 3.2001 16 C-016 =CHR -0.3962 4.2654 17 C-017 =CR2 0.0383 3.9392 18 C-018 =CHX -0.8051 3.6005 19 C-019 =CRX -0.2129 4.4870 20 C-020 =CX2 0.2432 3.2001 21 C-021 u2261CH 0.4697 3.4825 22 C-022 u2261CR, R=C=R 0.2952 4.2817 23 C-023 u2261CX - 3.9556 24 C-024 R:CH:R -0.3251 3.4491 25 C-025 R:CR:R 0.1492 3.8821 26 C-026 R:CX:R 0.1539 3.7593 27 C-027 R:CH:X 0.0005 2.5009 28 C-028 R:CR:X 0.2361 2.5000 29 C-029 R:CX:X 0.3514 3.0627 30 C-030 X:CH:X 0.1814 2.5009 31 C-031 X:CR:X 0.0901 - 32 C-032 X:CX:X 0.5142 2.6632 33 C-033 R:CH u2022 u2022 u2022X -0.3723 3.4671 34 C-034 R:CR u2022 u2022 u2022X 0.2813 3.6842 35 C-035 R:CX u2022 u2022 u2022X 0.1191 2.9372 36 C-036 Al-CH=X -0.1320 4.0190 37 C-037 Ar-CH=X -0.0244 4.7770 38 C-038 Al-C(=X)-Al -0.2405 3.9031 39 C-039 Ar-C(=X)-R -0.0909 3.9964 40 C-040 R-C(=X)-X, R-C u2261X, X=C=X -0.1002 3.4986 41 C-041 X-C(=X)-X 0.4182 3.4997 42 C-042 X:CH u2022 u2022 u2022X -0.2147 2.7784 42 C-043 X:CR u2022 u2022 u2022X -0.0009 2.6267 44 C-044 X:CX u2022 u2022 u2022X 0.1388 2.5000 45 H-046 H attached to C0sp3 no X attached to next C 0.7341 0.8447 46 H-047 H attached to C1sp3, C0sp2 0.6301 0.8939 47 H-048 H attached to C2sp3, C1sp2, C0sp 0.5180 0.8005 48 H-049 H attached to C3sp3, C2-3sp2, C3sp -0.0371 0.8320 49 H-050 H attached to heteroatom -0.1036 0.8000 50 H-051 H attached to u03b1-C 0.5234 0.8188 51 H-052 H attached to C0sp3 with 1X attached to next C 0.6666 0.9215 52 H-053 H attached to C0sp3 with 2X attached to next C 0.5372 0.9769 53 H-054 H attached to C0sp3 with 3X attached to next C 0.6338 0.7701 54 H-055 H attached to C0sp3 with 4X attached to next C 0.3620 - 55 O-056 alcohol -0.3567 1.7646 56 O-057 phenol, enol, carboxyl OH -0.0127 1.4778 57 O-058 =O -0.0233 1.4429 58 O-059 Al-O-Al -0.1541 1.6191 59 O-060 Al-O-Ar, Ar-O-Ar, R u2022 u2022 u2022O u2022 u2022 u2022R, R-O-C=X 0.0324 1.3502 60 O-061 -O (as in nitro, N-oxide) 1.0520 1.9450 61 O-062 O- (negatively charged) -0.7941 - 62 O-063 R-O-O-R 0.4165 - 63 Se-064 Any-Se-Any 0.6601 - 64 Se-065 =Se - - 65 N-066 Al-NH2 -0.5427 2.6221 66 N-067 Al2-NH -0.3168 2.5000 67 N-068 Al3-N 0.0132 2.8980 68 N-069 Ar-NH2, X-NH2 -0.3883 3.6841 69 N-070 Ar-NH-Al -0.0389 4.2808 70 N-071 Ar-NAl2 0.1087 3.6189 71 N-072 RCO-N&lt;, &gt;N-X=X -0.5113 2.5000 72 N-073 Ar2NH, Ar3N, Ar2N-Al, R u2022 u2022 u2022N u2022 u2022 u2022R (pyrrole-type structure) 0.1259 2.7956 73 N-074 R u2261N, R=N- 0.1349 2.7000 74 N-075 R:N:R, R:N:X (pyridine-type structure) -0.1624 4.2063 75 N-076 Ar-NO2, R:N(:R):O (pyridine N-oxide type structure), RO-NO -2.0585 4.0184 76 N-077 Al-NO2 -1.9150 3.0009 77 N-078 Ar-N=X, X-N=X 0.4208 4.7142 78 N-079 N+ (positively charged) -1.4439 - 79 F-081 F attached to C1sp3 0.4797 0.8725 80 F-082 F attached to C22sp3 0.2358 1.1837 81 F-083 F attached to C3sp3 0.1029 1.1573 82 F-084 F attached to C1sp2 0.3566 0.8001 83 F-085 F attached to C2-4sp2, C1sp, C4sp3, X 0.1988 1.5013 84 Cl-086 Cl attached to C1sp3 0.7443 5.6156 85 Cl-087 Cl attached to C2sp3 0.5337 6.1022 86 Cl-088 Cl attached to C3sp3 0.2996 5.9921 87 Cl-089 Cl attached to C1sp2 0.8155 5.3885 88 Cl-090 Cl attached to C2-4sp2, C1sp, C4sp3, X 0.4856 6.1363 89 Br-091 Br attached to C1sp3 0.8888 8.5991 90 Br-092 Br attached to C2sp3 0.7452 8.9188 91 Br-093 Br attached to C3sp3 0.5034 8.8006 92 Br-094 Br attached to C1sp2 0.8995 8.2065 93 Br-095 Br attached to C2-4sp2, C1sp, C4sp3, X 0.5946 8.7352 94 I-096 I attached to C1sp3 1.4201 13.9462 95 I-097 I attached to C2sp3 1.1472 14.0792 96 I-098 I attached to C3sp3 - 14.0730 97 I-099 I attached to C1sp2 0.7293 12.9918 98 I-100 I attached to C2-4sp2, C1sp, C4sp3, X 0.7173 13.3408 99 F-101 fluoride ion - - 100 Cl-102 chloride ion -2.6737 - 101 Br-103 bromide ion -2.4178 - 102 I-104 iodide ion -3.1121 - 103 S-106 R-SH 0.6146 7.8916 104 S-107 R2S, RS-SR 0.5906 7.7935 105 S-108 R=S 0.8758 9.4338 106 S-109 R-SO-R -0.4979 7.7223 107 S-110 R-SO2-R -0.3786 5.7558 108 Si-111 &gt;Si&lt; 1.5188 - 109 B-112 &gt;B- as in boranes 1.0255 - 110 P-115 P ylids - - 111 P-116 R3-P=X -0.9359 5.5306 112 P-117 X3-P=X (phosphate) -0.1726 5.5152 113 P-118 PX3 (phosphite) -0.7966 6.8360 114 P-119 PR3 (phosphine) 0.6705 10.0101 115 P-120 C-P(X)2=X (phosphonate) -0.4801 5.2806 Table 1: Structural feature definitions and their coefficients for the calculation of ALOGP and AMR descriptors. Atomic hydrophobicity is expressed in the unit of log P(octanol-water) In Table 1 R represents any group linked through carbon; X represents any heteroatom (O, N, S, P, Se, and halogens); Al and Ar represent aliphatic and aromatic groups, respectively; = represents a double bond; u2261 represents a triple bond; : represents an aromatic bond as in benzene or delocalised bonds such as the N-O bond in a nitro group; u2022 u2022 u2022 u201a represents aromatic single bonds as the C-N bond in pyrrole. On carbon atoms the subscript represents hybridisation and the superscript its formal oxidation number. The formal oxidation number of a carbon atom equals the sum of the formal bond orders with electronegative atoms; the C:N bond order in pyridine may be considered as 2 while we have one such bond and 1.5 when we have two such bonds; the C u2022 u2022 u2022X bond order in pyrrole or furan may be considered as 1. An u03b1-C may be defined as a C attached through a single bond with -C=X, -C u2261X or -C:X. V.N. Viswanadhan, A.K. Ghose, G.R. Revankar, R.K. Robins (1989). Atomic Physicochemical Parameters for Three Dimensional Structure Directed Quantitative Structure-Activity Relationships. 4. Additional Parameters for Hydrophobic and Dispersive Interactions and Their Application for an Automated Superposition of Certain Naturally Occurring Nucleoside Antibiotics. J.Chem.Inf.Comput.Sci. 29, 163-172&#160;&#x21a9;&#xfe0e; Ghose et al (1998). Prediction of Hydrophobic (Lipophilic) Properties of Small Organic Molecules Using Fragmental Methods: An Analysis of ALOGP and CLOGP Methods. J.Phys.Chem. Vol. 102, No. 21&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/atom-type_e-state_indices.html", "title": "Atom-type E-state indices", "tags": [], "description": "", "content": "The Atom-type E-state indices provide information combining both the electronic character and the topological environment of the atoms in a molecule1. These indices are associated to specific Atom-type as proposed by Hall and Kier2. They are divided in four groups: E-state sums Atom-type counts E-state minimum E-state maximum The E-state sums (S followed by the Kier-Hall symbol) are the summation of the electrotopological states of the specific atom-types in the molecule. The E-state minimum and maximum symbols are defined using the prefix min and max, followed by the Kier-Hall symbol. The electrotopological state (E-state) of an atom i is formulated as: where Ii is the intrinsic state of the i-th atom and u0394Ii is the perturbation effect on the i-th atom; dij is the topological distance between atom i and atom j and nSK is the number of non-hydrogen atoms. The Atom-type counts (N followed by the Kier-Hall symbol) represent the number of the specific atom-types in the molecule. alvaDesc calculates also the minimum and the maximum atom E-state value in a moleucle, respectively the gmin and gmax descriptors. Atom-types proposed by Hall and Kier Each atom in the molecule is classified by four characteristics: atomic number (Z) valence state including aromaticity indication number of bonded hydrogen atoms in a few cases, the identity of other bonded atoms Here&rsquo;s a list of all the possible atom types: No. Atom-type formula Kier-Hall symbol 1 -CH3 sCH3 2 =CH2 dCH2 3 -CH2- ssCH2 4 #CH tCH 5 =CH- dsCH 6 a(a)CH aaCH 7 &gt;CH- sssCH 8 =C= ddC 9 #C- tsC 10 =C&lt; dssC 11 C(a)(a)- aasC 12 C(a)(a)a aaaC 13 &gt;C&lt; ssssC 14 -NH2 sNH2 15 -NH- ssNH 16 =NH dNH 17 &gt;N- sssN 18 =N- dsN 19 aNa aaN 20 #N tN 21 -NH3[+1] sNH3+ 22 -NH2-[+1] ssNH2+ 23 =NH2[+1] dNH2+ 24 &gt;NH-[+1] sssNH+ 25 &gt;N&lt;[+1] ssssN+ 26 -N(=)= (nitro) ddsN 27 a(a)N= aadN 28 a(a)N- (N-oxide) aasN 29 a(a)NH aaNH 30 -OH sOH 31 =O dO 32 -O- ssO 33 aOa aaO 34 -PH2 sPH2 35 -PH- ssPH 36 &gt;P- sssP 37 -(&gt;)P= dsssP 38 -P(=)= ddsP 39 -(&gt;)P&lt; sssssP 40 -SH sSH 41 =S dS 42 -S- ssS 43 S(a)a aaS 44 &gt;S= dssS (sulfone) 45 =(&gt;)S= ddssS (sulfate) 46 -(&gt;)S(&lt;)- ssssssS 47 -F sF 48 -Cl sCl 49 -Br sBr 50 -I sI 51 -Li sLi 52 -Be- ssBe 53 &gt;S&lt;[-1] ssssBe- 54 -BH2 sBH2 55 -BH- ssBH 56 &gt;B- sssB 57 &gt;B&lt;[-1] ssssB- 58 -GeH3 sGeH3 59 -GeH2- ssGeH2 60 &gt;GeH- sssGeH 61 &gt;Ge&lt; ssssGe 62 -AsH2 sAsH2 63 &gt;AsH ssAsH 64 &gt;As- sssAs 65 &gt;As(&lt;)- sssssAs 66 &gt;As(=)- dsssAs 67 =As(=)- ddsAs 68 -SeH sSeH 69 =Se dSe 70 -Se- ssSe 71 Se(a)a aaSe 72 &gt;Se= dssSe 73 &gt;(&gt;)Se&lt; ssssssSe 74 &gt;Se(=)(=) ddssSe 75 -SnH3 sSnH3 76 &gt;SnH2 ssSnH2 77 &gt;SnH- sssSnH 78 &gt;Sn&lt; ssssSn 79 -PbH3 sPbH3 80 &gt;PbH2 ssPbH2 81 &gt;PbH- sssPbH 82 &gt;Pb&lt; ssssPb 83 -SiH3 sSiH3 84 &gt;SiH2 ssSiH2 85 &gt;SiH- sssSiH 86 &gt;Si&lt; ssssSi Atom-type formula symbols: &lsquo;-&rsquo; is a single bond &lsquo;=&rsquo; is a double bond &lsquo;#&rsquo; is a triple bond &lsquo;&lt;&rsquo; and &lsquo;&gt;&rsquo; are two single bonds &lsquo;a&rsquo; represent an aromatic bond Hall, L. H., Mohney, B., &amp; Kier, L. B. (1991). The Electrotopological State: An Atom Index for QSAR. Quantitative Structure u2010Activity Relationships, 10(1), 43 u201351.&#160;&#x21a9;&#xfe0e; Hall, L. H., &amp; Kier, L. B. (1995). Electrotopological State Indices for Atom Types: A Novel Combination of Electronic, Topological, and Valence State Information. Journal of Chemical Information and Computer Sciences, 35(6), 1039 u20131045.&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/pharmacophore.html", "title": "Pharmacophore descriptors", "tags": [], "description": "", "content": "Pharmacophore descriptors block includes two different types of descriptors: Chemically Advanced Template Search descriptors (CATS 2D) Shannon Entropy Descriptors (SHED) Chemically Advanced Template Search descriptors (CATS 2D) CATS 2D descriptors, as well as 2D atom pairs, are a particular case of autocorrelation descriptors, the atom-type autocorrelation descriptors (ATAC)1 and they are defined as: where u and v represent two different atom types and u03b4(i;u), u03b4(j;v), and u03b4(dij;k) are three Kronecker delta functions equal to one if atom i is of type u, atom j is of type v, and the topological distance dij is equal to k, respectively, zero otherwise. Conversely to 2D atom pairs that identify the atom types simply considering their atomic number, in CATS 2D descriptors the atom-type definition is related to the concept of potential pharmacophore points (PPP), where a PPP is a generalised atom type defined considering some physicochemical aspects2,3. CATS 2D descriptors are calculated on a topological distance varying from 0 to 9 and any atom of the molecule can be assigned to none, one, or two atom types ((DD, DA, DP, DN, DL, AA, AP, AN, AL, PP, PN, PL, NN, NL, LL)) resulting into a vector of 150 frequencies. SHannon Entropy Descriptors (SHED) SHannon Entropy Descriptors (SHED) are derived from the distributions of potential pharmacophore points (PPP) extracted directly from the topology of molecules on which is applied the Shannon entropy to quantify the variability in a feature-pair distribution4. Considering SHED descriptors as a whole they reflects the overall distribution of pharmacophore features in a molecule, the whole set of SHED descriptors constitutes the molecular SHED profile. alvaDesc implements SHannon Entropy Descriptors (SHED) using the previously defined potential pharmacophore points resulting in a vector of 15 descriptors. SHED profiles can be used to evaluate similarity between pairs of molecules, typically calculating the Euclidean distance between molecules since molecules having similar pharmacological profiles should contain similar features distributed in a similar manner. Carhart, R. E., Smith, D. H., &amp; Venkataraghavan, R. (1985). Atom Pairs as Molecular Features in Structure-Activity Studies: Definition and Applications. Journal of Chemical Information and Computer Sciences, 25(2), 64 u201373. https://doi.org/10.1021/ci00046a002&#160;&#x21a9;&#xfe0e; Schneider, G., Neidhart, W., Giller, T., &amp; Schmid, G. (1999). u201cScaffold-Hopping u201d by topological pharmacophore search: A contribution to virtual screening. Angewandte Chemie - International Edition, 38(19), 2894 u20132896. https://doi.org/10.1002/(SICI)1521-3773(19991004)38:19&lt;2894::AID-ANIE2894&gt;3.0.CO;2-F&#160;&#x21a9;&#xfe0e; Renner, S., Fechner, U., &amp; Schneider, G. (2006). Alignment-free pharmacophore patterns u2013 A correlation vector approach. In T. Langer &amp; R. D. Hoffmann (Eds.), Pharmacophores and pharmacophore searches (pp. 49 u201379). Weinheim: Wiley-VCH&#160;&#x21a9;&#xfe0e; Gregori-Puigjan u00e9, E., &amp; Mestres, J. (2006). SHED: Shannon entropy descriptors from topological feature distributions. Journal of Chemical Information and Modeling, 46(4), 1615 u20131622. https://doi.org/10.1021/ci0600509&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/2d_atom_pairs.html", "title": "2D Atom Pairs", "tags": [], "description": "", "content": "2D atom pairs block include two different type of descriptors: Weighted topological atom pairs Binary / Frequency atom pairs Weighted topological atom pairs Weighted topological atom pairs are defined as the sums of topological distances between all pairs of atom type included in the following set: N, O, S, P, F, Cl, Br, I. They are defined as follows: where u and v represent two different atom types, dij is the topological distance between the atom i and the atom j and u03b4(i;u), u03b4(j;v) are two Kronecker delta functions equal to one if atom i is of type u, atom j is of type v, respectively, zero otherwise. Binary / Frequency atom pairs The Binary / Frequency atom pairs are a particular case of autocorrelation descriptors, the atom-type autocorrelation descriptors (ATAC). Atom-type autocorrelation descriptors are discrete descriptors, that account for the presence/absence of atom pairs (&ldquo;Binary&rdquo;) or count their occurrences (&ldquo;Frequency&rdquo;), where atom pairs are defined as any pair of atoms having particular features at a predefined topological distance k1. They can be defined as follows: where u and v represent two different atom types and u03b4(i;u), u03b4(j;v), and u03b4(dij;k) are three Kronecker delta functions equal to one if atom i is of type u, atom j is of type v, and the topological distance dij is equal to k, respectively, zero otherwise. Specifically alvaDesc implements the atom pairs of C, N, O, S, P, F, Cl, Br, I, B, Si and X (where X represents any other atom, except hydrogen) calculated on a range of topological distances from 1 (connected atoms) to 10. alvaDesc provides both the binary vector representing if a specific atom pair is present or absent in the molecule (e.g., B01[C-C]) and the occurrence frequency vector that reflects the occurrences of atom pairs (e.g., F01[C-C]). Carhart, R. E., Smith, D. H., &amp; Venkataraghavan, R. (1985). Atom pairs as molecular features in structure-activity studies: definition and applications. Journal of Chemical Information and Computer Sciences, 25(2), 64 u201373. https://doi.org/10.1021/ci00046a002&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/3d_atom_pairs.html", "title": "3D Atom Pairs", "tags": [], "description": "", "content": "3D Atom Pairs are defined as the weighted topological atom pairs, but using Euclidean interatomic distances instead of topological distances. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. They are defined as follows: where u and v represent two different atom types, rij is the Euclidean distance between the atom i and the atom j and u03b4(i;u), u03b4(j;v) are two Kronecker delta functions equal to one if atom i is of type u, atom j is of type v, respectively, zero otherwise. alvaDesc provides weighted geometrical atom pairs G(X..Y), which are sums of the Euclidean interatomic distances between all the possible combinations of types X and Y where X and Y referring to any heteroatom among N, O, S, P, F, Cl, Br, I. " }, { "uri": "./descriptors/charge_descriptors.html", "title": "Charge descriptors", "tags": [], "description": "", "content": "The Charge descriptors describe the molecule based on its atomic charges. The molecule partial charges can be automatically calculated by alvaDesc. Please check Partial charges chapter for more information. Given that the partial charge (qi) of a given atom i can be either positive: or negative: these are the formulas of the charge descriptors: Symbol Descriptor name Formula qpmax Maximum Positive Charge qnmax Maximum Negative Charge Qpos Total Positive Charge Qneg Total Negative Charge Qtot Total Absolute Charge (Electronic Charge Index - ECI)1 Qmean Mean Absolute Charge (Charge Polarization)2 Q2 Total Squared Charge RPCG Relative Positive Charge3 RNCG Relative Negative Charge3 SPP Submolecular Polarity Parameter4,5 TE1 Topographic Electronic Descriptor2,6,5 TE2 Topographic Electronic Descriptor (Bond Restricted) PCWTE1 Partial Charge Weighted Topological Electronic Index7 PCWTE2 Partial Charge Weighted Topological Electronic Index (Bond Restricted) LDI Local Dipole Index8,9,5 Notes on the formulas: Qtot is a measure of the molecule polarity and is also known as Electronic Charge Index (ECI) 1 Qmean is the polarization of the molecule 5 Q2 analogously with Qtot is a measure of the molecular polarity rij is the geometrical distance between the atoms i and j and rb is the geometrical distance between the atoms connected by bond b TE2 is like TE1, but it considers only the pair of bonded atoms (therefore, given a bond b, it works with its two atoms b,1 and b,2) TE1, TE2, PCWTE1, PCWTE2 descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. Collantes, E. R., &amp; Dunn, W. J. (1995). Amino Acid Side Chain Descriptors for Quantitative Structure-Activity Relationship Studies of Peptide Analogues. Journal of Medicinal Chemistry, 38(14), 2705 u20132713. https://doi.org/10.1021/jm00014a022&#160;&#x21a9;&#xfe0e; O u015bmia u0142owski, K., Halkiewicz, J., &amp; Kaliszan, R. (1986). Quantum chemical parameters in correlation analysis of gas-liquid chromatographic retention indices of amines. II. Topological electronic index. Journal of Chromatography A, 361, 63 u201369. https://doi.org/10.1016/S0021-9673(01)86894-1&#160;&#x21a9;&#xfe0e; Stanton, D. T., &amp; Jurs, P. C. (1990). Development and Use of Charged Partial Surface Area Structural Descriptors in Computer-Assisted Quantitative Structure-Property Relationship Studies. Analytical Chemistry, 62(21), 2323 u20132329. https://doi.org/10.1021/ac00220a013&#160;&#x21a9;&#xfe0e; Kaliszan, R., Osmialowski, K., Tomellini, S. A., Hsu, S.-H., Fazio, S. D., &amp; Hartwick, R. A. (1985). Non-empirical descriptors of sub-molecular polarity and dispersive interactions in reversed-phase HPLC. Chromatographia, 20(12), 705 u2013708. https://doi.org/10.1007/BF02282926&#160;&#x21a9;&#xfe0e; Karelson, M., Lobanov, V. S., &amp; Katritzky, A. R. (1996). Quantum-Chemical Descriptors in QSAR/QSPR Studies. Chemical Reviews, 96(3), 1027 u20131044. https://doi.org/10.1021/cr950202r&#160;&#x21a9;&#xfe0e; Katritzky, A. R., &amp; Gordeeva, E. V. (1993). Traditional topological indexes vs electronic, geometrical, and combined molecular descriptors in QSAR/QSPR research. Journal of Chemical Information and Modeling, 33(6), 835 u2013857. https://doi.org/10.1021/ci00016a005&#160;&#x21a9;&#xfe0e; Katritzky, A. R., Mu, L., &amp; Karelson, M. (1996). A QSPR study of the solubility of gases and vapors in water. Journal of Chemical Information and Computer Sciences, 36(6), 1162 u20131168. https://doi.org/10.1021/ci960081b&#160;&#x21a9;&#xfe0e; Kikuchi, O. (1987). Systematic QSAR Procedures with Quantum Chemical Descriptors. Quantitative Structure u2010Activity Relationships, 6(4), 179 u2013184. https://doi.org/10.1002/qsar.19870060406+&#160;&#x21a9;&#xfe0e; Clare, B. W., &amp; Supuran, C. T. (1994). Carbonic Anhydrase Activators. 3: Structure-Activity Correlations for a Series of Isozyme I1 Activators. Journal of Pharmaceutical Sciences, 83(6), 768 u2013773. https://doi.org/10.1002/jps.2600830603&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/molecular_properties.html", "title": "Molecular properties", "tags": [], "description": "", "content": "alvaDesc calculates 27 molecular properties. Uc is an information index considering the unsaturated bonds and it is defined as: where nDB, nTB and nAB are respectively the number of double, triple and aromatic bonds. Ui is the unsaturation index1 and is calculated as follow: where SCBO is the sum of conventional bond orders in the H-depleted molecular graph and nBO is the total number of non-hydrogen bonds. Hy is the hydrophilic factor which is a hydrophilicity descriptor defined as2: where NHy is the number of hydrophilic groups (-OH, -SH, -NH), nC is the number of carbon atoms and nSK is the number of non-hydrogen atoms. alvaDesc calculates two molar refractivity (MR) models and a MR consensus model (MRcons). AMR is the Ghose-Crippen molar refractivity and it is based on the Ghose-Crippen group contribution model3,4,5, and MR99 is based on the Wildman-Crippen group contribution model6. They are defined as: where ni is the number of atom of type i and mri is the corresponding molar refractivity contribution. AMR can be calculated only for molecules whose atoms can be assigned to one of the 115 atom types defined by Ghose-Crippen. The molar refractivities of 538 compounds have been used in order to obtain atomic refractivity contributions. The topological polar surface area (TPSA)7 of a molecule is defined as the sum of the surfaces of all polar atoms. alvaDesc calculates TPSA(tot) considering nitrogen, oxygen and &ldquo;slightly polar&rdquo; fragments containing phosphorus and sulfur, and TPSA(NO) considering only nitrogen and oxygen contributions. No. Atom type TPSA contribution No. Atom type TPSA contribution 1 [N](-)(-)-* 3.24 23 [nH](:): 15.79 2 [N](-)= 12.36 24 [n+](:)(:):* 4.10 3 [N]#* 23.79 25 [n+](-)(:):* 3.88 4 [N](-)(=)=*b 11.68 26 [nH+](:): 14.14 5 [N](=)#c 13.60 27 [O](-)- 9.23 6 [N]1(-)--*-1_d_ 3.01 28 [O]1---1_d_ 12.53 7 [NH](-)- 12.03 29 [O]=* 17.07 8 [NH]1---1_d_ 21.94 30 [OH]-* 20.23 9 [NH]=* 23.85 31 [O-]-* 23.06 10 [NH2]-* 26.02 32 [o](:): 13.14 11 [N+](-)(-)(-)- 0.00 33 [S](-)- 25.30 12 [N+](-)(-)=* 3.01 34 [S]=* 32.09 13 [N+](-)#e 4.36 35 [S](-)(-)=* 19.21 14 [NH+](-)(-)-* 4.44 36 [S](-)(-)(=)= 8.38 15 [NH+](-)= 13.97 37 [SH]-* 38.80 16 [NH2+](-)- 16.61 38 [s](:): 28.24 17 [NH2+]=* 25.59 39 [s](=)(:):* 21.70 18 [NH3+]-* 27.64 40 [P](-)(-)-* 13.59 19 [n](:): 12.89 41 [P](-)= 34.14 20 [n](:)(:):* 4.41 42 [P](-)(-)(-)= 9.81 21 [n](-)(:):* 4.93 43 [PH](-)(-)=* 23.47 22 [n](=)(:):*f 8.39 Table 1: Atomic Contributions ( u00c52) to TPSA: an asterisk config.toml content hugocomp.sh hugoserv.sh public stands for any non-hydrogen atom, - for a single bond, d for a double bond, # for a triple bond, : for an aromatic bond; an atomic symbol in lowercase means that the atom is part of an aromatic system. (b) is the nitrogen atom as in nitro group. (c) is the middle nitrogen as in azide group. (d) is the atom in a three-membered ring. (e) is the nitrogen atom as in isocyano group. (f) is the nitrogen atom as in pyridine N-oxide Polar surface area has been used in medicinal chemistry for the optimisation of drug&rsquo;s capability to permeate cells8,9. Pajouhesh and Lenz published a paper where molecules with a polar surface area greater than 140 u00c52 have been correlated to a low ability at permeating cell membranes while drugs with a PSA of 60 u00c52 or less are completely absorbed. Hitchcock and Pennington suggested a threshold of 90 u00c52 for increasing the potential for blood u2013brain barrier penetration alvaDesc calculates three octanol-water partition coefficients (LogP) models and a LogP consensus model (LOGPcons). The Moriguchi octanol-water partition coefficient (MLOGP)10,11, the Ghose-Crippen-Viswanadhan octanol-water partition coefficient (ALOGP)12,4,5 and the Wildman-Crippen octanol-water partition coefficient (LOGP99)6. ALOGP can be calculated only for molecules whose atoms can be assigned to one of the 115 atom types defined by Ghose-Crippen The octanol-water partition coefficient of 8,364 compounds have been used in order to obtain atomic hydrophobicity contributions for ALOGP. The square value of ALOGP (ALOGP2) is provided. MLOGP is a regression model based on 13 structural parameters6,7. The regression coefficients were evaluated by a training set of 1,230 organic molecules resulting in a R2 equal to 0.906. Training set includes aliphatic, aromatic, and heterocyclic compounds, containing the following atom types: C, H, N, O, S, P, F, Cl, Br, I. The square value of MLOGP (MLOGP2) is provided. MLOGP model is defined as: Parameter Description CX Summation of the number of carbon and halogen atoms weighted by: C: 1.0, F: 0.5, Cl: 1.0, Br: 1.5 and I: 2.0 NO Total number of N and O atoms PRX Proximity effect of N/O; X-Y: 2.0, X-A-Y: 1.0 (X, Y: N/O, A: C, S or P) with a correction (-1) for carboxamide/sulfonamide UB Total number of unsaturated bonds except those in NO2 HB Dummy variable for the presence of intramolecular hydrogen bond as ortho-OH and -CO-R, -OH and -NH2, -NH2 and -COOH, or 8-OH/NH2 in quinolines, 5 or 8-OH/NH2 in quinoxalines, etc. POL Number of aromatic polar substituents (aromatic substituents excluding Ar-CX2- and Ar-CX=C&lt;, X: C or H). Upper limit = 4. AMP Amphoteric property; u03b1-aminoacid: 1.0, aminobenzoic acid: 0.5, pyridinecarboxylic acid: 0.5. ALK Dummy variable for alkane, alkene, cycloalkane or cycloalkene (hydrocarbons with 0 or 1 double bond) or hydrocarbon chain with at least 7 carbon atoms RNG Dummy variable for the presence of ring structures except benzene and its condensed rings (aromatic, heteroaromatic, and hydrocarbon rings) QN Quaternary nitrogen &gt;N+&lt;: 1.0; N oxide: 0.5 NO2 Number of nitro groups NCS Isothiocyanate (-N=C=S): 1.0; thiocyanate (-S-C u2261N): 0.5 BLM Dummy variable for the presence of u00df-lactam alvaDesc implementation of MLOGP model has been evaluated on a set of 3,576 compounds with known experimental logP taken from the NCI Open DataBase, resulting in a r2 equal to 0.935 ALOGP and LOGP99 models are defined as: where ni is the number of atom of type i and hi is the corresponding hydrophobicity contribution. alvaDesc implementation of ALOGP model has been evaluated using a set of 3,576 compounds with known experimental logP taken from the NCI Open DataBase, resulting in a r2 equal to 0.931 ALOGP can be calculated for compounds including fluoride, bromide, chloride and iodide ions. LogS aqueous solubility (ESOL) is based on the paper by Delaney13, and the coefficients have been recalibrated using the AqSolDB14 curated dataset as: where MW is the molecular weight, RBN is the number of rotatable bonds, AP is the aromatic proportion and LOGPcons is alvaDesc LogP consensus model. SAtot is the total surface area, SAacc is the surface area calculated considering only acceptor atoms and SAdon is the surface area calculated considering only donor atoms. SAtot, SAacc and SAdon are calculated considering the parameters defined for the calculation of P_VSA-like descriptors. McGowan volume Vx15,16 is defined as: where wi are Mc Gowan u2019s volume atomic parameters, nAT is the number of atoms and nBT is the number of bonds. VvdwMG is the van der Waals volume derived from McGowan volume and it is defined as: where Vx is the McGowan volume. VvdwZAZ is the van der Waals volume calculated from Zhao-Abraham-Zissimos equation17 as: where wi is the volume atom contribution, nBT is the number of bonds, Rar and Rnar are respectively the number of aromatic and not-aromatic rings. Volumes are expressed as u00c53/molecule PDI is the packing density index and it is defined as the ratio between the McGowan volume (Vx) and the total surface area from P_VSA-like descriptors (SAtot): Three Verhaar baseline toxicities for Fish, Daphnia and Algae defined by Verhaar and based on Moriguchi LogP are provided18,19. Verhaar Fish baseline toxicity (BLTF96) is defined as: Verhaar Daphnia baseline toxicity (BLTD48) is defined as: Verhaar Algae baseline toxicity (BLTA96) is defined as: Verhaar base-line toxicities are not calculated for molecules having a MLOGP value lower than -6. Verhaar baseline toxicities are an estimation of baseline log(LC50) toxicity values. MLOGP and Verhaar baseline toxicities (BLTF96, BLTD48 and BLTA96) cannot be calculated for disconnected structures alvaDesc calculates the synthetic accessibility score of drug-like molecules (SAscore)20 that provides a quantitative estimation of synthetic accessibility of molecules. SAscore is calculated as a combination of two components: The fragmentScore tries to capture the &ldquo;historical synthetic knowledge&rdquo; by analysing common structural features in a large number of already synthesised molecules. The complexityScore is affected by the presence of complex structural features in the molecules. It is calculated as a combination of a ring complexity score, a stereo complexity score, a macrocycle penalty and a size penalty20. SAscore values are in the range between 1 and 10, molecules with high SAscore values should be difficult to synthesise, whereas, molecules with the low SAscore values should be easily synthesizable The Plane of Best Fit (PBF) quantifies the 3D character of a molecule.21 The authors proposed it as a descriptor for quantitative analysis of the molecular 3D shape. They noticed that the PBF has a theoretical range from zero to infinity, but it tends to be less than 2 u212b for drug-like small molecules and less than 10 u212b for proteins. Plane of Best Fit (PBF) can only be calculated for molecules having 3D coordinates. The calculation first identifies the 3D plane where the molecule is best fitted and then takes the average of the distances of each atom from this plane: where nSK is the number atoms in the H-depleted graph and A, B, C, D represent the plane in the form Ax + By + Cz = D. Todeschini, R., &amp; Consonni, V. (2009). Molecular Descriptors for Chemoinformatics, Volumes I &amp; II. Wiley. https://doi.org/10.1002/9783527628766&#160;&#x21a9;&#xfe0e; Todeschini, R., Vighi, M., Finizio, A., &amp; Gramatica, P. (1997). 3D-modelling and Prediction by WHIM Descriptors. Part 8. Toxicity and Physico-chemical Properties of Environmental Priority Chemicals by 2D-TI and 3D-WHIM Descriptors. SAR and QSAR in Environmental Research, 7(1 u20134), 173 u2013193. https://doi.org/10.1080/10629369708039130&#160;&#x21a9;&#xfe0e; Ghose, A. K., &amp; Crippen, G. M. (1987). Atomic Physicochemical Parameters for Three-Dimensional-Structure-Directed Quantitative Structure-Activity Relationships. 2. Modeling Dispersive and Hydrophobic Interactions. Journal of Chemical Information and Computer Sciences, 27(1), 21 u201335. https://doi.org/10.1021/ci00053a005&#160;&#x21a9;&#xfe0e; Viswanadhan, V. N., Ghose, A. K., Revankar, G. R., &amp; Robins, R. K. (1989). Atomic Physicochemical Parameters for Three Dimensional Structure Directed Quantitative Structure-Activity Relationships. 4. Additional Parameters for Hydrophobic and Dispersive Interactions and Their Application for an Automated Superposition of Certain . Journal of Chemical Information and Computer Sciences, 29(3), 163 u2013172. https://doi.org/10.1021/ci00063a006&#160;&#x21a9;&#xfe0e; Ghose, A. K., Viswanadhan, V. N., &amp; Wendoloski, J. J. (1998). Prediction of hydrophobic (lipophilic) properties of small organic molecules using fragmental methods: An analysis of ALOGP and CLOGP methods. Journal of Physical Chemistry A, 102(21), 3762 u20133772. https://doi.org/10.1021/jp980230o&#160;&#x21a9;&#xfe0e; Wildman, S. A., &amp; Crippen, G. M. (1999). Prediction of Physicochemical Parameters by Atomic Contributions. Journal of Chemical Information and Computer Sciences, 39(5), 868 u2013873. https://doi.org/10.1021/ci990307l&#160;&#x21a9;&#xfe0e; Ertl, P., Rohde, B., &amp; Selzer, P. (2000). Fast calculation of molecular polar surface area as a sum of fragment-based contributions and its application to the prediction of drug transport properties. Journal of Medicinal Chemistry, 43(20), 3714 u20133717. https://doi.org/10.1021/jm000942e&#160;&#x21a9;&#xfe0e; Pajouhesh, H., &amp; Lenz, G. R. (2005). Medicinal chemical properties of successful central nervous system drugs. NeuroRx, 2(4), 541 u2013553. https://doi.org/10.1602/neurorx.2.4.541&#160;&#x21a9;&#xfe0e; Hitchcock, S. A., &amp; Pennington, L. D. (2006). Structure-brain exposure relationships. Journal of Medicinal Chemistry, 49(26), 7559 u20137583. https://doi.org/10.1021/jm060642i&#160;&#x21a9;&#xfe0e; Moriguchi, I., Hirono, S., Liu, Q., Nakagome, I., &amp; Matsushita, Y. (1992). Simple Method of Calculating Octanol/Water Partition Coefficient. Chemical and Pharmaceutical Bulletin, 40(1), 127 u2013130. https://doi.org/10.1248/cpb.40.127&#160;&#x21a9;&#xfe0e; Moriguchi, I., Hirono, S., Nakagome, I., &amp; Hirano, H. (1994). Comparison of Reliability of log P Values for Drugs Calculated by Several Methods. Chemical and Pharmaceutical Bulletin, 42(4), 976 u2013978. https://doi.org/10.1248/cpb.42.976&#160;&#x21a9;&#xfe0e; Ghose, A. K., &amp; Crippen, G. M. (1986). Atomic Physicochemical Parameters for Three-Dimensional Structure-Directed Quantitative Structure-Activity Relationships I. Partition Coefficients as a Measure of Hydrophobicity. Journal of Computational Chemistry, 7(4), 565 u2013577. https://doi.org/10.1002/jcc.540070419&#160;&#x21a9;&#xfe0e; Delaney, J. S. (2004). ESOL: Estimating aqueous solubility directly from molecular structure. Journal of Chemical Information and Computer Sciences, 44(3), 1000 u20131005. https://doi.org/10.1021/ci034243x&#160;&#x21a9;&#xfe0e; Sorkun, M. C., Khetan, A., &amp; Er, S. (2019). AqSolDB, a curated reference set of aqueous solubility and 2D descriptors for a diverse set of compounds. Scientific Data, 6(1), 143. https://doi.org/10.1038/s41597-019-0151-1&#160;&#x21a9;&#xfe0e; Abraham, M. H., &amp; McGowan, J. C. (1987). The use of characteristic volumes to measure cavity terms in reversed phase liquid chromatography. Chromatographia, 23(4), 243 u2013246. https://doi.org/10.1007/BF02311772&#160;&#x21a9;&#xfe0e; Zhao, Y. H., Abraham, M. H., &amp; Zissimos, A. M. (2003). Determination of McGowan Volumes for Ions and Correlation with van der Waals Volumes. Journal of Chemical Information and Computer Sciences, 43(6), 1848 u20131854. https://doi.org/10.1021/ci0341114&#160;&#x21a9;&#xfe0e; Zhao, Y. H., Abraham, M. H., &amp; Zissimos, A. M. (2003). Fast Calculation of van der Waals Volume as a Sum of Atomic and Bond Contributions and Its Application to Drug Compounds. The Journal of Organic Chemistry, 68(19), 7368 u20137373. https://doi.org/10.1021/jo034808o&#160;&#x21a9;&#xfe0e; Verhaar, H. J. M., Van Leeuwen, C. J., &amp; Hermens, J. L. M. (1992). CLASSIFYING ENVIRONMENTAL-POLLUTANTS .1. STRUCTURE-ACTIVITY-RELATIONSHIPS FOR PREDICTION OF AQUATIC TOXICITY. Chemosphere, 25(4), 471 u2013491. https://doi.org/10.1016/0045-6535(92)90280-5&#160;&#x21a9;&#xfe0e; Verhaar, H. J. M., Solb u00e9, J., Speksnijder, J., Van Leeuwen, C. J., &amp; Hermens, J. L. M. (2000). Classifying environmental pollutants: Part 3. External validation of the classification system. Chemosphere, 40(8), 875 u2013883. https://doi.org/10.1016/S0045-6535(99)00317-3&#160;&#x21a9;&#xfe0e; Ertl, P., &amp; Schuffenhauer, A. (2009). Estimation of synthetic accessibility score of drug-like molecules based on molecular complexity and fragment contributions. Journal of Cheminformatics, 1(1), 1 u201311. https://doi.org/10.1186/1758-2946-1-8&#160;&#x21a9;&#xfe0e; Firth, N. C., Brown, N., &amp; Blagg, J. (2012). Plane of best fit: A novel method to characterize the three-dimensionality of molecules. Journal of Chemical Information and Modeling, 52(10), 2516 u20132525. https://doi.org/10.1021/ci300293f&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/drug_like_indices.html", "title": "Drug-like indices", "tags": [], "description": "", "content": "alvaDesc calculates 30 drug-like indices. Ro5 is the Lipinski&rsquo;s rule of five, a rule of thumb to evaluate drug-likeness of chemical compounds. Ro5 have been applied to the design of orally active compounds1,2. Ro5 states that poor absorption or permeation are more likely when: there are more than 5 H-bond donors nHDon there are more than 10 H-bond acceptors (N + O) molecular weight MW &gt; 500 Moriguchi&rsquo;s logP MLOGP &gt; 4.15 Since alvaDesc uses MLOGP instead of ClogP, the threshold is set to 4.15 instead of 5 as proposed by Lipinski et al1. In general, an orally active drug has no more than one violation of the Ro5 criteria. Ro5 is a dummy variable taking value 1 when more than one property is outside of the range. In addition to Ro5, alvaDesc provides cRo5, a drug-like index calculated as 1 minus Ro5. All Lipinski&rsquo;s Ro5 rules have numbers that are multiples of five, which is the origin of the rule&rsquo;s name Druglikeness is a qualitative concept used in drug design that evaluates if a substance is potentially bioavailable. alvaDesc provides 8 drug-like scores (DLS) and 2 lead-like Scores (LLS). These indices have been defined as the ratio between the number of satisfied rules and the total number of rules: where tRules is the total number of rules and nRules is the number of satisfied rules. DLS_01 is a modified drug-like score from Lipinski&rsquo;s rules (4 rules) and according to the LS formula, tRules is set to 4 and nRules is the number of unsatisfied rules from Ro5. The index DLS_02 is a drug-like score based on the following 6 rules3,4: H-bond donors nHDon u2264 5 H-bond acceptors (N + O) in the range from 1 to 8 molecular weight MW in the range from 200 to 450 Moriguchi&rsquo;s logP MLOGP in the range from -2.0 to 4.5 rotatable bond number RBN in the range from 1 to 9 number of rings nCIC u2264 5 DLS_03 is a drug-like score based on the following 6 rules5: H-bond donors nHDon u2264 5 H-bond acceptors (N + O) u2264 10 molecular weight MW in the range from 200 to 500 Moriguchi&rsquo;s logP MLOGP in the range from -5.0 to 5.0 rotatable bond number RBN u2264 8 totalcharge in the range from -2 to 2 DLS_04 is a drug-like score based on the following 7 rules6: H-bond donors nHDon u2264 5 H-bond acceptors (N + O) in the range from 2 to 10 molecular weight MW in the range from 78 to 500 Moriguchi&rsquo;s logP MLOGP in the range from -0.5 to 5.0 descriptor C3p in the range 0.15 u2013 0.8 descriptor h-p in the range 0.6 u2013 1.6 descriptor Unsat-p in the range 0.10 u2013 0.45 where C3p is the ratio of the number of Csp3 atoms to the total number of non-halogen atoms, h-p is the ratio of the number of hydrogen atoms to the total number of non-halogen atoms, Unsat-p is the ratio between the molecular unsaturation (Unsat index) and the total number of nonhalogen heavy atoms. The Unsat index6 is defined as: where NRG567 is the number of 5-, 6-, and 7-membered rings, nDB is the number of double bonds, nTB is the number of triple bonds, and nAB is the number of aromatic bonds. DLS_05 is a drug-like score based on the following 2 rules7: descriptor NO_C3 in the range 0.10 u2013 1.80 descriptor Unsat-p u2264 0.43 where Unsat-p is defined as for the index DLS_04, NO_C3 is an index related to the proportion of heteroatoms, defined as the ratio between the total number of oxygen and nitrogen atoms and the number of Csp3 atoms. DLS_06 is a drug-like score based on the following 6 rules8: H-bond donors nHDon u2264 5 H-bond acceptors (N + O) u2264 10 molecular weight MW u2264 500 Moriguchi&rsquo;s logP MLOGP u2264 5 rotatable bond number RBN u2264 10 polar surface area TPSA(tot) u2264 140 DLS_07 is a drug-like score based on the following 2 rules9: number of rotatable bonds RBN u2264 10 polar surface area TPSA(tot) u2264 140 u00c52 or the sum of H-bond acceptors and H-bond donors u2264 12 DLS_cons (Dragon consensus drug-like score) is defined as the average of the drug-like scores previously defined: In drug discovery a lead-like compound is a chemical compound that has pharmacological or biological activity. alvaDesc lead-like scores can be used as filters to determine whose compounds can be considered as lead compounds. Typically lead compounds, compared to drugs, have a lower molecular complexity, a smaller number of rings (nCIC), a smaller number of rotatable bonds (RBN), a lower MW and are more polar10. LLS_01 is a lead-like score based on the following 6 rules11: H-bond donors nHDon u2264 3 H-bond acceptors (N + O) in u2264 3 molecular weight MW u2264 300 Moriguchi&rsquo;s logP MLOGP u2264 3 rotatable bond number RBN u2264 3 polar surface area TPSA(tot) u2264 60 The rules used by LLS_01 are based on the well known rule of three (Ro3), where all numbers are multiples of three, which is the origin of the rule&rsquo;s name LLS_02 is a lead-like score based on the following 6 rules12: H-bond donors u2264 5 H-bond acceptors in u2264 9 molecular weight MW u2264 460 Moriguchi&rsquo;s logP MLOGP in the range from u20134.0 to 4.2 rotatable bond number RBN u2264 10 number of rings nCIC u2264 4 number of halogens nX u2264 7 total number of nN + nO u2265 1 where H-bond donors is the number of hydrogen donors defined as the sum of any heteroatom with at least one connected hydrogen atom and with neutral or positive formal charge. H-bond acceptors is the number of hydrogen acceptors defined as the sum of any atoms of nitrogen, oxygen, phosphor and sulfur atoms, with the exception of aromatic oxygen and sulfur, aromatic nitrogen connected to three other atoms, nitrogen with valence 5 and sulfur with valence 6 or 7. In addition to the previously described indices, alvaDesc implements sixteen drug-like indices as proposed by Ghose-Viswanadhan-Wendoloski13. These indices can be used in the design of drug-like combinatorial libraries and for drug design. The Comprehensive Medicinal Chemistry database (version 97.1) contains the structures and biological properties of 7183 compounds and it has been used to evaluate the distribution of some physicochemical properties (logP, molar refractivity, molecular weight, number of atoms). The whole CMC database and seven different subsets belonging to different classes of drug molecules have been analysed. These indices are binary variables whose value is equal to 1 when logP (ALOGP), molar refractivity (AMR), molecular weight (MW) and number of atoms (nAT) of the considered compound fall within the property range shown in Table 1. Moreover, the considered compound must include at least one of the following functional groups: a benzene ring a heterocyclic ring (both aliphatic and aromatic) an aliphatic amine a carboxamide group an alcoholic hydroxyl group a carboxy ester a keto group For each drug class two different ranges have been proposed for the considered properties: the qualifying range that covers approximately 80% of the drugs studied the preferred range that covers approximately 50% of the drugs studied The result are two drug-like scores for each drug class. CMC has been derived from the analysis of the whole CMC database while the other indices from the analysis of the compounds belong to the considered drug classes. Symbol Drug class No. studied drugs ALOGP AMR MW nAT CMC-80 CMC 6304 -0.4; 5.6 40; 130 160; 480 20; 70 CMC-50 1.3; 4.1 70; 110 230; 390 30; 55 Inflammat-80 Antiinflammatory 290 1.4; 4.5 59; 119 212; 447 24; 59 Inflamm-50 2.6; 4.2 67; 97 260; 380 28; 40 Depressant-80 Antidepressant 208 1.4; 4.9 62; 114 210; 380 32; 56 Depressant-50 2.1; 4.0 75; 95 260; 330 37; 48 Psychotic-80 Antipsychotic 105 2.3; 5.2 85; 131 274; 464 40; 63 Psychotic-50 3.3; 5.0 94; 120 322; 422 49; 61 Hypertens-80 Antihypertensive 269 -0.5; 4.5 54; 128 206; 506 28; 66 Hypertens-50 1; 3.4 68; 116 281; 433 36; 58 Hypnotic-80 Hypnotic 74 0.5; 3.9 43; 97 162; 360 20; 45 Hypnotic-50 1.3; 3.5 43; 73 212; 306 29; 38 Neoplastic-80 Antineoplastic 349 -1.5; 4.7 43; 128 180; 475 21; 63 Neoplastic-50 0.0; 3.7 60; 107 258; 388 30; 55 Infective-80 Antiinfective 39 -0.3; 5.1 44; 144 145; 455 12; 64 Infective-50 0.8; 3.8 68; 138 192; 392 12; 42 Table 1: Qualifying (covering 80% of the total number of compounds in that class) and Preferred (covering 50% of the total number of compounds) ranges of the various physicochemical properties in different drug classes The concept of drug-likeness plays an important role when selecting compounds during the early stages of drug discovery, therefore in addition to the previously described indices, alvaDesc includes the calculation of the quantitative estimate of drug-likeness (QED)14. The QED is based on the concept of desirability and its values can range from 0 (all properties unfavourable) to 1 (all properties favourable). QED is calculated considering eight molecular properties selected on the basis of their relevance in determining drug-likeness: molecular weight (MW) octanol u2013water partition coefficient (LOGP99) number of hydrogen bond donors (Hydrogen-bond donor (D)) number of hydrogen bond acceptors (Hydrogen-bond acceptors (A)) molecular polar surface area (TPSA(tot)) number of rotatable bonds (RBN) number of aromatic rings number of hits identified in a set of 116 structural alerts All these properties are transformed using asymmetric double sigmoidal (ADS) functions to obtain a set of desirability functions (di). alvaDesc calculates also the unweighted quantitative estimate of drug-likeness (QEDu). Unweighted quantitative estimate of drug-likeness (QEDu) is defined as: where n is the number of desirability functions and di is the desirability of the considered property. Weighted quantitative estimate of drug-likeness (QED) is defined as: where di is the individual desirability function, wi is the weight applied to each function and n is the number of desirability functions. Drug-like indices cannot be calculated for disconnected structures Lipinski, C. A., Lombardo, F., Dominy, B. W., &amp; Feeney, P. J. (1997). Experimental and computational approaches to estimate solubility and permeability in drug discovery and development settings. Advanced Drug Delivery Reviews, 23(1 u20133), 3 u201325. https://doi.org/10.1016/S0169-409X(96)00423-1&#160;&#x21a9;&#xfe0e; Lipinski, C. A. (2004). Lead- and drug-like compounds: the rule-of-five revolution. Drug Discovery Today: Technologies, 1(4), 337 u2013341. https://doi.org/10.1016/j.ddtec.2004.11.007&#160;&#x21a9;&#xfe0e; Oprea, T. I., Gottfries, J., Sherbukhin, V., Svensson, P., &amp; K u00fchler, T. C. (2000). Chemical information management in drug discovery: optimizing the computational and combinatorial chemistry interfaces11Color Plates for this article are on page 541. Journal of Molecular Graphics and Modelling, 18(4 u20135), 512 u2013524. https://doi.org/10.1016/S1093-3263(00)00066-8&#160;&#x21a9;&#xfe0e; Oprea, T. I. (2000). Property distribution of drug-related chemical databases. Journal of Computer-Aided Molecular Design, 14(3), 251-64. https://doi.org/10.1023/a:1008130001697&#160;&#x21a9;&#xfe0e; Walters, W. P., &amp; Murcko, M. A. (2002). Prediction of u2018drug-likeness u2019. Advanced Drug Delivery Reviews, 54(3), 255 u2013271. https://doi.org/10.1016/S0169-409X(02)00003-0&#160;&#x21a9;&#xfe0e; Chen, G., Zheng, S., Luo, X., Shen, J., Zhu, W., Liu, H., Gui, C., Zhang, J., Zheng, M., Puah, C. M., Chen, K., &amp; Jiang, H. (2005). Focused Combinatorial Library Design Based on Structural Diversity, Druglikeness and Binding Affinity Score. Journal of Combinatorial Chemistry, 7(3), 398 u2013406. https://doi.org/10.1021/cc049866h&#160;&#x21a9;&#xfe0e; Zheng, S., Luo, X., Chen, G., Zhu, W., Shen, J., Chen, K., &amp; Jiang, H. (2005). A New Rapid and Effective Chemistry Space Filter in Recognizing a Druglike Database. Journal of Chemical Information and Modeling, 45(4), 856 u2013862. https://doi.org/10.1021/ci050031j&#160;&#x21a9;&#xfe0e; Rishton, G. M. (2003). Nonleadlikeness and leadlikeness in biochemical screening. Drug Discovery Today, 8(2), 86 u201396. https://doi.org/10.1016/S1359644602025722&#160;&#x21a9;&#xfe0e; Veber, D. F., Johnson, S. R., Cheng, H.-Y., Smith, B. R., Ward, K. W., &amp; Kopple, K. D. (2002). Molecular Properties That Influence the Oral Bioavailability of Drug Candidates. Journal of Medicinal Chemistry, 45(12), 2615 u20132623. https://doi.org/10.1021/jm020017n&#160;&#x21a9;&#xfe0e; Hann, M. M., Leach, A. R., &amp; Harper, G. (2001). Molecular Complexity and Its Impact on the Probability of Finding Leads for Drug Discovery. Journal of Chemical Information and Computer Sciences, 41(3), 856 u2013864. https://doi.org/10.1021/ci000403i&#160;&#x21a9;&#xfe0e; Congreve, M., Carr, R., Murray, C., &amp; Jhoti, H. (2003). A u2018Rule of Three u2019 for fragment-based lead discovery? Drug Discovery Today, 8(19), 876 u2013877. https://doi.org/10.1016/S1359-6446(03)02831-9&#160;&#x21a9;&#xfe0e; Monge, A., Arrault, A., Marot, C., &amp; Morin-Allory, L. (2006). Managing, profiling and analyzing a library of 2.6 million compounds gathered from 32 chemical providers. Molecular Diversity, 10(3), 389 u2013403. https://doi.org/10.1007/s11030-006-9033-5&#160;&#x21a9;&#xfe0e; Ghose, A. K., Viswanadhan, V. N., &amp; Wendoloski, J. J. (1999). A Knowledge-Based Approach in Designing Combinatorial or Medicinal Chemistry Libraries for Drug Discovery. 1. A Qualitative and Quantitative Characterization of Known Drug Databases. Journal of Combinatorial Chemistry, 1(1), 55 u201368. https://doi.org/10.1021/cc9800071&#160;&#x21a9;&#xfe0e; Bickerton, G. R., Paolini, G. V., Besnard, J., Muresan, S., &amp; Hopkins, A. L. (2012). Quantifying the chemical beauty of drugs. Nature Chemistry, 4(2), 90 u201398. https://doi.org/10.1038/nchem.1243&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/cats3d.html", "title": "CATS 3D descriptors", "tags": [], "description": "", "content": "CATS3D descriptors1 are a special type of three-dimensional autocorrelation descriptors. They can be considered the three-dimensional counterpart of the CATS 2D descriptors. These descriptors can only be calculated for molecules having 3D coordinates for all atoms, including hydrogen atoms. CATS 3D descriptors are calculated considering the Euclidean interatomic distance split in 20 distance bins (from 0 to 20 A u030a in steps of 1 A u030a), since any atom of the molecule can be assigned to none, one, or two potential pharmacophore points (PPP) the CATS 3D descriptors result into a vector of 300 values. The formula for CATS 3D descriptors descriptors is the following: where k is the considered distance bin, u and v represent two different atom types, and u03b4(i;u), u03b4(j;v), and u03b4(rij;k) are three Kronecker delta functions equal to one respectively if atom i is of type u, atom j is of type v, and their interatomic distance rij is comprised in the k-th bin, zero otherwise. Fechner, U., Franke, L., Renner, S., Schneider, P., &amp; Schneider, G. (2003). Comparison of correlation vector methods for ligand-based similarity searching. Journal of Computer-Aided Molecular Design, 17(10), 687 u2013698. https://doi.org/10.1023/B:JCAM.0000017375.61558.ad&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/whales.html", "title": "WHALES descriptors", "tags": [], "description": "", "content": "The Weighted Holistic Atom Localization and Entity Shape (WHALES)1 descriptors have been presented as a way to facilitate scaffold hopping from natural products to isofunctional synthetic compounds. They also have been used for more general virtual compound screening and scaffold hopping2. Their holistic approach allows them to capture many molecular properties at once: geometric interatomic distances molecular shape atomic properties (e.g., partial charge distribution) These descriptors can only be calculated for molecules having 3D coordinates. The WHALES descriptors are based on the calculation of the Atom-centered Mahalanobis (ACM) matrix: where ACM(i, j) is an item of the ACM matrix, i is the row number, j is the column number, xi and xj are the 3D coordinates of the i and j atom and Sw(j)-1 is the Atom-centered covariance matrix which is calculated as follow: where u03b4i is the partial charge of the i-th atom and nSK is the number of atoms in the H-depleted molecular graph. All the WHALES matrices are calculated on the H-depleted molecular graph. These descriptors are divided in three groups: Remoteness (Rem) Isolation degree (Isol) Isolation-Remoteness ratio (IR) Each group contains 11 descriptors calculated as deciles (from 0 to 100) on the specific group matrix. Therefore, the total number of WHALES descriptors is always 33 regardless of the number of atoms in the molecules. Remoteness (Rem) The Remoteness (Rem) is a matrix calculated with the following formula: The Remoteness is high for atoms with large ACM distances from many atomic centres. Isolation degree (Isol) The Isolation degree (Isol) is the ACM matrix column minimum (atomic centre excluded): The isolation degree is high for atoms surrounded by a few close atoms. Isolation-Remoteness ratio (IR) The Isolation-Remoteness ratio (IR) is simply the ratio between the Isolation degree and the Remoteness. Therefore, it contains both global (Remoteness) and local (Isolation degree) information of each atom. The IR, as indicated by the authors, assumes high values for atoms residing off the main molecular axis (i.e., high isolation degree) and a small relative distance from most of the atomic centres (i.e., low remoteness). The values of the Remoteness, Isolation degree and Isolation-Remoteness ratio are multiplied by -1 when calculated for an atom with negative partial charge. Grisoni, F., Merk, D., Consonni, V. et al. (2018). Scaffold hopping from natural products to synthetic mimetics by holistic molecular similarity. Commun Chem 1, 44&#160;&#x21a9;&#xfe0e; Grisoni, F., &amp; Schneider, G. (2021). Molecular Scaffold Hopping via Holistic Molecular Representation. In F. Ballante (Ed.), Protein-Ligand Interactions and Drug Design (Vol. 2266, pp. 11 u201335). Springer US. https://doi.org/10.1007/978-1-0716-1209-5_2&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/mde.html", "title": "MDE descriptors", "tags": [], "description": "", "content": "Molecular Distance Edge (MDE) descriptors1 are based on two fundamental structural variables, the atom types and the distance between atoms in the molecular graph. alvaDesc calculates 19 MDE descriptors, considering carbon (MDEC-), oxygen (MDEO-) and nitrogen atoms (MDEN-). The carbon atoms are classified by four basic types: type 1 for primary (CH3-), type 2 for secondary (CH2&lt;), type 3 for ternary (-CH&lt;) and type 4 for quaternary (&gt;C&lt;) carbon atoms. The oxygen atoms are classified by two basic types: type 1 for primary (OH-) and type 2 for secondary (-O-) oxygen atoms. The nitrogen atoms are classified by three basic types: type 1 for primary (NH2-), type 2 for secondary (-NH-) and type 3 for ternary (-N&lt;) nitrogen atoms. The MDE values are calculated as follows: where nkm is the number of atom pairs belonging to class k and m and mdekm is defined as follows: where dij is the topological distance between atom i and atom j, u03b4(k;m) is a Kronecker delta function equal to one for pairs of atoms belonging to type k and m respectively and nkm the number of atom pairs belonging to class k and m considered in the multiplication. MDE descriptors have been proposed to estimate and predict the normal boiling point (NBP) of alkanes Liu, S., Cao, C., &amp; Li, Z. (1998). Approach to Estimation and Prediction for Normal Boiling Point (NBP) of Alkanes Based on a Novel Molecular Distance-Edge (MDE) Vector, u03bb. Journal of Chemical Information and Computer Sciences, 38(3), 387 u2013394. https://doi.org/10.1021/ci970109z&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/chirality_descriptors.html", "title": "Chirality descriptors", "tags": [], "description": "", "content": "The chirality descriptors are a group of descriptors calculated on the chiral centres of a molecule. These descriptors can only be calculated for molecules with at least one chiral centre. Only the chiral centers on carbon atoms will be considered. The first two descriptors are the ChiralCenter which represents the number of chiral centres in the molecule and the ringChiralCenter which represents the number of chiral centres that are also ring atoms. The following descriptors have been proposed1 to collect information on each chiral centre of the molecule: ID Descriptor name Additional explanation 3 meanDistFromCC mean pairwise distance considering all paths starting at the chiral centre 4 maxDistfromCC longest path starting from the chiral centre 5-14 nLevel1..nLevel10 number of neighbouring atoms of the chiral centre (level1 u2013level10) 15-17 phLevel1..phLevel3 number of neighbouring N/O atoms of the chiral centre (level1 u2013level3) 18-20 arLevel1..arLevel3 number of neighbouring aromatic atoms of the chiral centre (level1 u2013level3) 21-24 sx_size number of heavy atoms of the substituent normalized by the number of atoms shared between substituents 25-28 sx_relSize sx_size normalized by the number of heavy atoms 29-32 sx_phSize number of N/O atoms of the substituent normalized by the number of N/O atoms shared between substituents 33-36 sx_phRelSize sx_phSize normalized by the number of heavy atoms 37-40 sx_phRelSize_2 sx_phSize normalized by sx_size 41-44 sx_pathLength maximum path length of the substituent x 45-48 sx_relPathLength maximum path length of the substituent x normed by number of heavy atoms 49-52 sx_relPathLength_2 maximum path length of the substituent x normed by sx_size 53-56 sx_numSharedNeighbors number of shared neighbours with other substituents 57-60 sx_numRotBonds number of rotatable bonds of the substituent x 61-64 sx_numAroBonds number of aromatic bonds of the substituent x 65 s34_size sum of s3_size and s4_size 66 s34_relSize sum of s3_relSize and s4_relSize 67 s34_phSize sum of s3_phSize and s4_phSize 68 s34_phRelSize sum of s3_phRelSize and s4_phRelSize 69 chiralMoment similar to the moment of inertia, but centred at the chiral centre 70 chiralPhMoment similar to chiralMoment, but only taking the N/O atoms into account In the event that the number of chiral centers is greater than one, alvaDesc calculates the descriptors shown in the table for each chiral center and then returns the mean value. The x in the descriptors starting with sx_ (e.g., sx_size) refers to one of the four possible substituents (1..4) of the chiral atom. In the following image you can see the chiral centre (yellow) and the four substituent highlighted in different colours: The descriptors starting with sx_ assume value of 0 if they refer to the substituent containing only an hydrogen atom. N. Schneider (2018). Chiral Cliffs: Investigating the Influence of Chirality on Binding Affinity. ChemMedChem 2018, 13, 1315 u20131324&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/weighting_scheme.html", "title": "Weighting schemes", "tags": [], "description": "", "content": "alvaDesc makes use of some atomic properties as the atomic weightings for molecular descriptor calculation. Atomic properties used as the atomic weightings for descriptor calculation in alvaDesc are: unitary weight (u) atomic number (Z) atomic mass (m) van der Waals volume (v) atom electronegativity (e) atom polarizability (p) ionization potential (i) intrinsic state (s) covalent radius (r) Atomic masses include the isotopic mass data from G. Audi, A. H. Wapstra Nucl. Phys A. 1993, 565, 1-65 and G. Audi, A. H. Wapstra Nucl. Phys A. 1995, 595, 409-480 Covalent radius (r) parameter is used only for the calculation of 3D autocorrelations Conversely to the other atomic parameters used by alvaDesc the intrinsic state (I-state) is a local vertex invariant derived from the molecular graph1. The I-state of the i-th atom is calculated as: where L is the principal quantum number, u03b4v is the number of valence electrons (valence vertex degree) and u03b4 is the number of sigma electrons (simple vertex degree) of the i-th atom in the hydrogen depleted molecular structure. alvaDesc sets to 1 the value of I-state for hydrogen atoms. Atomic properties are not always defined for the whole periodic table. Molecular descriptors based on a property that is not available for all molecule atoms will result to &lsquo;na&rsquo; (e.g., Sv descriptor that is the sum of atomic van der Waals volumes cannot be calculated on molecules including calcium atom) Atomic properties are typically used in carbon-scaled form. i.e., the ratio between the atomic property and the carbon atom property L.B. Kier, L.H. Hall. (1990) An Electrotopological-State Index for Atoms in Molecules. Pharm. Res. 7, 801&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/potential_pharmacophore_points.html", "title": "Potential Pharmacophore Points (PPP)", "tags": [], "description": "", "content": "alvaDesc implements five Potential Pharmacophore Points (PPP)1 to be used for the calculation of several descriptors. A Potential Pharmacophore Point (PPP) is a generalised atom type defined considering some physicochemical aspects. The five Potential Pharmacophore Points (PPP) implemented in alvaDesc are the following: hydrogen-bond donor (D) hydrogen-bond acceptor (A) positive (P) negative (N) lipophilic (L) alvaDesc Potential Pharmacophore Points (PPP) implementation follows the rules originally proposed for the calculation of CATS Hydrogen-bond donor (D) An atom is considered as hydrogen-bond donor (D) if it is: oxygen atom of an OH-group nitrogen atom of an NH or NH2 group Hydrogen-bond acceptor (A) An atom is considered as hydrogen-bond acceptor (A) if it is: oxygen atom nitrogen atom not adjacent to a hydrogen atom Positive (P) An atom is considered as positive (P) if it is: atom with a positive charge nitrogen atom of an NH2-group Negative (N) An atom is considered as negative (N) if it is: atom with a negative charge carbon, sulphur or phosphorus atom of a COOH, SOOH or POOH group Lipophilic (L) An atom is considered as lipophilic (L) if it is: chlorine, bromine or iodine atom sulphur atom adjacent to exactly two carbon atoms (C u2013S u2013C) carbon atom adjacent only to carbon atoms Any atom of the molecule can be assigned to none, one, or two atom types Renner, S., Fechner, U., &amp; Schneider, G. (2006). Alignment-free pharmacophore patterns u2013 A correlation vector approach. In T. Langer &amp; R. D. Hoffmann (Eds.), Pharmacophores and pharmacophore searches (pp. 49 u201379). Weinheim: Wiley-VCH&#160;&#x21a9;&#xfe0e; " }, { "uri": "./descriptors/matrix-based.html", "title": "Matrix-based descriptors", "tags": [], "description": "", "content": "The Matrix-based descriptors are those calculated using one of the matrices detailed in the 2D matrix-based descriptors or the 3D matrix-based descriptors chapters. Here&rsquo;s the list of the matrix-based descriptors: ID Descriptor name Formula 1 Wi_M - Wiener-like index 2 WiA_M - Average Wiener-like index 3 AVS_M - Average vertex sum 4 H_M - Harary-like index 5 Chi_M - Randic-like index 6 ChiA_M - Average Randic-like index 7 J_M - Balaban-like index 8 HyWi_M - Hyper-Wiener-like index 9 SpAbs_M - Graph energy 10 SpPos_M - Spectral positive sum 11 SpPosA_M - Normalized spectral positive sum 12 SpPosLog_M - Logarithmic spectral positive sum 13 SpMax_M - Leading eigenvalue 14 SpMaxA_M - Normalized leading eigenvalue 15 SpDiam_M - Spectral diameter 16 SpAD_M - Spectral absolute deviation1 17 SpMAD_M - Spectral mean absolute deviation1 18 Ho_M - Hosoya-like index 19 EE_M - Estrada-like index2,3,4 20 SM1_M - Spectral moment of order 1 21 SM2_M - Spectral moment of order 2 22 SM3_M - Spectral moment of order 3 23 SM4_M - Spectral moment of order 4 24 SM5_M - Spectral moment of order 5 25 SM6_M - Spectral moment of order 6 26 VE1_M - Coefficient sum of the last eigenvector (absolute values)5 27 VE2_M - Average coefficient of the last eigenvector (absolute value)5 28 VE3_M - Logarithmic coefficient sum of the last eigenvector (absolute value)5 29 VE1sign_M - Coefficient sum of the last eigenvector 30 VE2sign_M - Average coefficient of the last eigenvector 31 VE3sign_M - Logarithmic coefficient sum of the last eigenvector 32 VR1_M - Randic-like eigenvector-based index 33 VR2_M - Normalized Randic-like eigenvector-based index 34 VR3_M - Logarithmic Randic-like eigenvector-based index Some notes on the notation used: M can be any of the matrix symbols (e.g., VE1_A is the Coefficient sum of the last eigenvector (absolute values) from adjacency matrix) n is nSK for 2D-matrix based descriptors and is nAT for 3D-matrix based descriptors nBO is the number of bonds (H excluded) nCIC is the number of circuits w is the weighting scheme l (also called last eigenvector) is the eigenvector, calculated on M, associated with the largest negative eigenvalue 5 VSi(M) is the i-th matrix row sum aij is an element of the Adjacency matrix (when i and j have the same value aij is equal to 0) ci is the i-th coefficient of the characteristic polynomial of M n+ the number of positive eigenvalues u03bb+ is a positive eigenvalue More notes on the formulas: descriptors based on the last eigenvector (VE1_M, VE2_M, VE3_M, VE1sign_M, VE2sign_M, VE3sign_M, VR1_M, VR2_M, VR3_M) cannot be calculated if the largest negative eigenvalue is not unique (e.g., when calculating these using the Detour/Distance matrix on a molecule with an acyclic graph) some of these descriptors, for specific M, are not present in alvaDesc when an analogous descriptor already exists. For example: SpAbs_A would give the same result as SpAD_A Wi_A would give the same result as nBO Chi_A would give the same result as the Kier-Hall connectivity index of order 1 (X1) the Lovasz-Pelikan index (SpMax_A), also called the leading eigenvalue, is the largest eigenvalue of the adjacency matrix6 the Wiener index (Wi_D) is calculated as the half-sum of all topological distances collected in the distance matrix7. The mean Wiener index (WiA_D) is calculated by dividing the Wiener index by the total number of topological distances in the molecular graph since the smallest eigenvalue obtained from a Laplace matrix is always 0, the Spectral diameter (SpDiam_L) is calculated as the difference between the largest eigenvalue and the second smallest eigenvalue (Acon) the 3D-Wiener index (Wi_G) is calculated as the sum of all the interatomic geometric distances in an H-filled molecule8. Wi_G has different values for different conformations: the larger values correspond to the most extended conformations and the smaller values to the more compact conformations. It decreases with increasing sphericity of the molecular structure9 the folding degree index (SpMaxA_G/D) is the largest eigenvalue of the distance/distance matrix (G/D), normalised dividing it by the number of atoms nAT. SpMaxA_G/D values approaches 1 for linear molecules (of infinite length) and 0 as nAT increases in a hypothetical zigzag structure that overlaps itself10 the Estrada-like index (EE_M) cannot be calculated for some matrices (e.g., detour matrix (Dt)) when working with molecules with many atoms and/or with large rings or fused rings Consonni, V., &amp; Todeschini, R. (2008). New Spectral Indices for Molecule Description. Match, 60, 3 u201314. http://match.pmf.kg.ac.rs/electronic_versions/Match60/n1/match60n1_3-14.pdf&#160;&#x21a9;&#xfe0e; Consonni, V., &amp; Todeschini, R. (2012). Multivariate Analysis of Molecular Descriptors. Statistical Modelling of Molecular Descriptors in QSAR/QSPR, 2, 111 u2013147. https://doi.org/10.1002/9783527645121.ch4&#160;&#x21a9;&#xfe0e; Estrada, E., &amp; Rodr u00edguez-Vel u00e1zquez, J. A. (2005). Subgraph centrality in complex networks. Physical Review E - Statistical, Nonlinear, and Soft Matter Physics, 71(5), 1 u20139. https://doi.org/10.1103/PhysRevE.71.056103&#160;&#x21a9;&#xfe0e; Estrada, E., &amp; Hatano, N. (2007). Statistical-mechanical approach to subgraph centrality in complex networks. Chemical Physics Letters, 439(1 u20133), 247 u2013251. https://doi.org/10.1016/j.cplett.2007.03.098&#160;&#x21a9;&#xfe0e; Balaban, A. T., Ciubotariu, D., &amp; Medeleanu, M. (1991). Topological Indices and Real Number Vertex Invariants Based on Graph Eigenvalues or Eigenvectors. Journal of Chemical Information and Computer Sciences, 31(4), 517 u2013523. https://doi.org/10.1021/ci00004a014&#160;&#x21a9;&#xfe0e; Lov u00e1sz, L., &amp; Pelik u00e1n, J. (1973). On the eigenvalues of trees. Periodica Mathematica Hungarica, 3(1 u20132), 175 u2013182. https://doi.org/10.1007/BF02018473&#160;&#x21a9;&#xfe0e; Wiener, H. (1947). Structural Determination of Paraffin Boiling Points. Journal of American Chemical Society, 69(1), 17 u201320. https://doi.org/10.1021/ja01193a005&#160;&#x21a9;&#xfe0e; Bogdanov, B., Nikoli u0107, S., &amp; Trinajsti u0107, N. (1989). On the three-dimensional wiener number. Journal of Mathematical Chemistry, 3(3), 299 u2013309. https://doi.org/10.1007/BF01169597&#160;&#x21a9;&#xfe0e; Nikoli u0107, S., Trinajsti u0107, N., Mihali u0107, Z., &amp; Carter, S. (1991). On the geometric-distance matrix and the corresponding structural invariants of molecular systems. Chemical Physics Letters, 179(1 u20132), 21 u201328. https://doi.org/10.1016/0009-2614(91)90285-H&#160;&#x21a9;&#xfe0e; Randi u0107, M., &amp; Krilov, G. (1999). On a characterization of the folding of proteins. International Journal of Quantum Chemistry, 75(6), 1017 u20131026. https://doi.org/10.1002/(SICI)1097-461X(1999)75:6&lt;1017::AID-QUA6&gt;3.0.CO;2-C&#160;&#x21a9;&#xfe0e; " }, { "uri": "./categories.html", "title": "Categories", "tags": [], "description": "", "content": "" }, { "uri": "./", "title": "Main page", "tags": [], "description": "", "content": "alvaDesc 2.0 - User manual alvaDesc is an Alvascience software. " }, { "uri": "./tags.html", "title": "Tags", "tags": [], "description": "", "content": "" } ]';

    //MTB [09/01/2019]: parseJSON instead of getJSON
    var objJson = $.parseJSON(jsonContent);
    myLunrParse(objJson);
 
    //MTB [09/01/2019]: this was the original code
    //// First retrieve the index file
    //$.getJSON(baseurl +"index.json")
    //    .done(function(index) {
    //        pagesIndex =   index;
    //        // Set up lunrjs by declaring the fields we use
    //        // Also provide their boost level for the ranking
    //        lunrIndex = new lunr.Index
    //        lunrIndex.ref("uri");
    //        lunrIndex.field('title', {
    //            boost: 15
    //        });
    //        lunrIndex.field('tags', {
    //            boost: 10
    //        });
    //        lunrIndex.field("content", {
    //            boost: 5
    //        });
    //
    //        // Feed lunr with each file and let lunr actually index them
    //        pagesIndex.forEach(function(page) {
    //            lunrIndex.add(page);
    //        });
    //        lunrIndex.pipeline.remove(lunrIndex.stemmer)
    //    })
    //    .fail(function(jqxhr, textStatus, error) {
    //        var err = textStatus + ", " + error;
    //        console.error("Error getting Hugo index flie:", err);
    //    });
}

/**
 * Trigger a search in lunr and transform the result
 *
 * @param  {String} query
 * @return {Array}  results
 */
function search(query) {
    // Find the item in our index corresponding to the lunr one to have more info
    return lunrIndex.search(query).map(function(result) {
            return pagesIndex.filter(function(page) {
                return page.uri === result.ref;
            })[0];
        });
}

// Let's get started
initLunr();

//MTB [09/01/2019]: extract last folder name from current document location
var curPath = window.location.href;
var arrPath = curPath.split('/');
var lastFolder = arrPath[arrPath.length-2];

$( document ).ready(function() {
    var searchList = new autoComplete({
        /* selector for the search box element */
        selector: $("#search-by").get(0),
        /* source is the callback to perform the search */
        source: function(term, response) {
            response(search(term));
        },
        /* renderItem displays individual search results */
        renderItem: function(item, term) {
            var numContextWords = 2;
            var text = item.content.match(
                "(?:\\s?(?:[\\w]+)\\s?){0,"+numContextWords+"}" +
                    term+"(?:\\s?(?:[\\w]+)\\s?){0,"+numContextWords+"}");
            item.context = text;
            return '<div class="autocomplete-suggestion" ' +
                'data-term="' + term + '" ' +
                'data-title="' + item.title + '" ' +
                
                //MTB [09/01/2019]: make it work even if we are in a subdirectory
                //'data-uri="'+ item.uri + '" ' +
                ((lastFolder == 'public' || lastFolder == 'help') ? ('data-uri="'+ item.uri + '" ') : ('data-uri="../'+ item.uri.substring(2) + '" ')) +

                'data-context="' + item.context + '">' +
                '» ' + item.title +
                '<div class="context">' +
                (item.context || '') +'</div>' +
                '</div>';
        },
        /* onSelect callback fires when a search suggestion is chosen */
        onSelect: function(e, term, item) {
            var pro_debug = item.getAttribute('data-uri');
            location.href = pro_debug;
        }
    });
});
